We collected and analyzed DHS data from the ENCODE project for 3 different cell lines (A549, HCT116, and K562). Links to the selected files can be found in this folder. We used bed narrowPeak files aligned to the GRCh38 reference genome for all cell lines, with peaks called using an FDR cutoff of 5%. We manually combined 2 replicates for each cell line and used bedtools to merge them into one bed file, as detailed in 08_DHSdataProcess.ipynb. We then classified guides with cut sites positioned between the ends of at least one peak as overlapping with a peak. Mann-Whitney U tests were performed on the gene level to assess the significance of phenotypic differences displayed by guides within versus outside of peaks.
