{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0b2b6d5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f52cef42",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/fzheng/opt/anaconda3/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "path = '/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/Comp Data Sets/Tiling Library/Gilbert 2014/'\n",
    "seq_detail = pd.read_table(path+\"CRISPRicin library 20130912_final_draft_CRISPRicin.txt\", header = None)\n",
    "seq_phenotpye = pd.read_excel(path + 'CRISPRricin compiled all sgRNA data.xlsx', header = None)\n",
    "\n",
    "path = '/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2023/k562_tiling/CRISPRko/'\n",
    "new_tiling = pd.read_csv(path+'cleaned_horlbeck_tiling.csv')\n",
    "new_col_seq_phenotpye = ['guide id'] + seq_phenotpye[[1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,17, 18, 19, 20, 21, 22, 23, 24]][0:3].apply('_'.join,  axis=0).tolist()\n",
    "seq_phenotpye = seq_phenotpye.drop([0, 1,2,3], axis=0)\n",
    "seq_phenotpye.columns = new_col_seq_phenotpye"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d293a5c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IGF2R_w_160385167.25</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGGAATAAAGAAAAACTACTTTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IGF2R_w_160385190.28</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCCTTTTGCTTAAGCAAAAGTCAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IGF2R_w_160385278.28</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgggaaatttagggaaaggacattg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IGF2R_w_160385278.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGggaaatttagggaaaggacattgt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IGF2R_w_160385278.26</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgaaatttagggaaaggacattgtG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      0                                                  1\n",
       "0  IGF2R_w_160385167.25  cccttggagaaCCAcctTGTTGGGGAATAAAGAAAAACTACTTTGT...\n",
       "1  IGF2R_w_160385190.28  cccttggagaaCCAcctTGTTGGCCTTTTGCTTAAGCAAAAGTCAC...\n",
       "2  IGF2R_w_160385278.28  cccttggagaaCCAcctTGTTGgggaaatttagggaaaggacattg...\n",
       "3  IGF2R_w_160385278.27  cccttggagaaCCAcctTGTTGggaaatttagggaaaggacattgt...\n",
       "4  IGF2R_w_160385278.26  cccttggagaaCCAcctTGTTGgaaatttagggaaaggacattgtG..."
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_detail.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2d4c8ddf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>Cas9-nuclease_gamma_ave_LG_JEV</th>\n",
       "      <th>Cas9-nuclease_rho_ave_LG_JEV</th>\n",
       "      <th>Cas9-nuclease_tau_ave_LG_JEV</th>\n",
       "      <th>dCas9_gamma_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>dCas9_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>dCas9_tau_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>dCas9-KRAB_gamma_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>dCas9-KRAB_tau_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>...</th>\n",
       "      <th>scFV-VP64_tau_ave_LG_JEV</th>\n",
       "      <th>tet-Cas9-nuclease_gamma_ave_LG_JEV</th>\n",
       "      <th>tet-Cas9-nuclease_rho_ave_LG_JEV</th>\n",
       "      <th>tet-Cas9-nuclease_tau_ave_LG_JEV</th>\n",
       "      <th>tet-dCas9_gamma_ave_LG_JEV</th>\n",
       "      <th>tet-dCas9_rho_ave_LG_JEV</th>\n",
       "      <th>tet-dCas9_tau_ave_LG_JEV</th>\n",
       "      <th>tet-dCas9-KRAB_gamma_ave_LG_JEV</th>\n",
       "      <th>tet-dCas9-KRAB_rho_ave_LG_JEV</th>\n",
       "      <th>tet-dCas9-KRAB_tau_ave_LG_JEV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1_c_118438373.25</td>\n",
       "      <td>-0.042085</td>\n",
       "      <td>0.067389</td>\n",
       "      <td>-0.009748</td>\n",
       "      <td>-0.008239</td>\n",
       "      <td>-0.11281</td>\n",
       "      <td>-0.145932</td>\n",
       "      <td>-0.003073</td>\n",
       "      <td>-0.120761</td>\n",
       "      <td>-0.136305</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.000032</td>\n",
       "      <td>-0.011297</td>\n",
       "      <td>-0.097876</td>\n",
       "      <td>-0.124901</td>\n",
       "      <td>-0.006776</td>\n",
       "      <td>0.089315</td>\n",
       "      <td>0.102563</td>\n",
       "      <td>0.000996</td>\n",
       "      <td>-0.005057</td>\n",
       "      <td>-0.00068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ARCN1_c_118438478.24</td>\n",
       "      <td>-0.028864</td>\n",
       "      <td>-0.029428</td>\n",
       "      <td>-0.075971</td>\n",
       "      <td>-0.009053</td>\n",
       "      <td>0.027236</td>\n",
       "      <td>0.007994</td>\n",
       "      <td>-0.003389</td>\n",
       "      <td>0.012341</td>\n",
       "      <td>0.002547</td>\n",
       "      <td>...</td>\n",
       "      <td>0.02593</td>\n",
       "      <td>-0.000848</td>\n",
       "      <td>-0.007781</td>\n",
       "      <td>-0.011579</td>\n",
       "      <td>0.008855</td>\n",
       "      <td>-0.018122</td>\n",
       "      <td>-0.008353</td>\n",
       "      <td>0.009998</td>\n",
       "      <td>-0.012417</td>\n",
       "      <td>0.00571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ARCN1_c_118438478.26</td>\n",
       "      <td>-0.010585</td>\n",
       "      <td>0.082849</td>\n",
       "      <td>0.055826</td>\n",
       "      <td>0.000392</td>\n",
       "      <td>-0.058185</td>\n",
       "      <td>-0.05975</td>\n",
       "      <td>-0.01074</td>\n",
       "      <td>0.029811</td>\n",
       "      <td>0.009294</td>\n",
       "      <td>...</td>\n",
       "      <td>0.051061</td>\n",
       "      <td>0.021742</td>\n",
       "      <td>0.061692</td>\n",
       "      <td>0.098399</td>\n",
       "      <td>0.001522</td>\n",
       "      <td>-0.003338</td>\n",
       "      <td>-0.006786</td>\n",
       "      <td>-0.001491</td>\n",
       "      <td>-0.028662</td>\n",
       "      <td>-0.030491</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ARCN1_c_118438508.26</td>\n",
       "      <td>-0.002614</td>\n",
       "      <td>-0.114052</td>\n",
       "      <td>-0.10155</td>\n",
       "      <td>-0.015898</td>\n",
       "      <td>0.02013</td>\n",
       "      <td>-0.009754</td>\n",
       "      <td>0.00132</td>\n",
       "      <td>0.022175</td>\n",
       "      <td>0.023339</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.016855</td>\n",
       "      <td>0.011423</td>\n",
       "      <td>0.000828</td>\n",
       "      <td>0.017933</td>\n",
       "      <td>0.020234</td>\n",
       "      <td>-0.005401</td>\n",
       "      <td>0.03163</td>\n",
       "      <td>-0.008477</td>\n",
       "      <td>0.005852</td>\n",
       "      <td>-0.01496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ARCN1_c_118438682.27</td>\n",
       "      <td>0.027815</td>\n",
       "      <td>-0.101679</td>\n",
       "      <td>-0.03159</td>\n",
       "      <td>0.032612</td>\n",
       "      <td>0.013335</td>\n",
       "      <td>0.080821</td>\n",
       "      <td>0.02712</td>\n",
       "      <td>-0.031622</td>\n",
       "      <td>0.017937</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024278</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.042612</td>\n",
       "      <td>0.08502</td>\n",
       "      <td>0.037326</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               guide id Cas9-nuclease_gamma_ave_LG_JEV  \\\n",
       "4  ARCN1_c_118438373.25                      -0.042085   \n",
       "5  ARCN1_c_118438478.24                      -0.028864   \n",
       "6  ARCN1_c_118438478.26                      -0.010585   \n",
       "7  ARCN1_c_118438508.26                      -0.002614   \n",
       "8  ARCN1_c_118438682.27                       0.027815   \n",
       "\n",
       "  Cas9-nuclease_rho_ave_LG_JEV Cas9-nuclease_tau_ave_LG_JEV  \\\n",
       "4                     0.067389                    -0.009748   \n",
       "5                    -0.029428                    -0.075971   \n",
       "6                     0.082849                     0.055826   \n",
       "7                    -0.114052                     -0.10155   \n",
       "8                    -0.101679                     -0.03159   \n",
       "\n",
       "  dCas9_gamma_ave_LG2_LG_MH2_MH dCas9_rho_ave_LG2_LG_MH2_MH  \\\n",
       "4                     -0.008239                    -0.11281   \n",
       "5                     -0.009053                    0.027236   \n",
       "6                      0.000392                   -0.058185   \n",
       "7                     -0.015898                     0.02013   \n",
       "8                      0.032612                    0.013335   \n",
       "\n",
       "  dCas9_tau_ave_LG2_LG_MH2_MH dCas9-KRAB_gamma_ave_LG2_LG_MH2_MH  \\\n",
       "4                   -0.145932                          -0.003073   \n",
       "5                    0.007994                          -0.003389   \n",
       "6                    -0.05975                           -0.01074   \n",
       "7                   -0.009754                            0.00132   \n",
       "8                    0.080821                            0.02712   \n",
       "\n",
       "  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH dCas9-KRAB_tau_ave_LG2_LG_MH2_MH  ...  \\\n",
       "4                        -0.120761                        -0.136305  ...   \n",
       "5                         0.012341                         0.002547  ...   \n",
       "6                         0.029811                         0.009294  ...   \n",
       "7                         0.022175                         0.023339  ...   \n",
       "8                        -0.031622                         0.017937  ...   \n",
       "\n",
       "  scFV-VP64_tau_ave_LG_JEV tet-Cas9-nuclease_gamma_ave_LG_JEV  \\\n",
       "4                -0.000032                          -0.011297   \n",
       "5                  0.02593                          -0.000848   \n",
       "6                 0.051061                           0.021742   \n",
       "7                -0.016855                           0.011423   \n",
       "8                 0.024278                                NaN   \n",
       "\n",
       "  tet-Cas9-nuclease_rho_ave_LG_JEV tet-Cas9-nuclease_tau_ave_LG_JEV  \\\n",
       "4                        -0.097876                        -0.124901   \n",
       "5                        -0.007781                        -0.011579   \n",
       "6                         0.061692                         0.098399   \n",
       "7                         0.000828                         0.017933   \n",
       "8                              NaN                              NaN   \n",
       "\n",
       "  tet-dCas9_gamma_ave_LG_JEV tet-dCas9_rho_ave_LG_JEV  \\\n",
       "4                  -0.006776                 0.089315   \n",
       "5                   0.008855                -0.018122   \n",
       "6                   0.001522                -0.003338   \n",
       "7                   0.020234                -0.005401   \n",
       "8                  -0.042612                  0.08502   \n",
       "\n",
       "  tet-dCas9_tau_ave_LG_JEV tet-dCas9-KRAB_gamma_ave_LG_JEV  \\\n",
       "4                 0.102563                        0.000996   \n",
       "5                -0.008353                        0.009998   \n",
       "6                -0.006786                       -0.001491   \n",
       "7                  0.03163                       -0.008477   \n",
       "8                 0.037326                             NaN   \n",
       "\n",
       "  tet-dCas9-KRAB_rho_ave_LG_JEV tet-dCas9-KRAB_tau_ave_LG_JEV  \n",
       "4                     -0.005057                      -0.00068  \n",
       "5                     -0.012417                       0.00571  \n",
       "6                     -0.028662                     -0.030491  \n",
       "7                      0.005852                      -0.01496  \n",
       "8                           NaN                           NaN  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_phenotpye.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "511deed1",
   "metadata": {},
   "source": [
    "### check corrleation among different dcas methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "78f98e6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dCas9_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>tet-dCas9_rho_ave_LG_JEV</th>\n",
       "      <th>tet-dCas9-KRAB_rho_ave_LG_JEV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>dCas9_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.488875</td>\n",
       "      <td>0.357679</td>\n",
       "      <td>0.306273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <td>0.488875</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.268658</td>\n",
       "      <td>0.518444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tet-dCas9_rho_ave_LG_JEV</th>\n",
       "      <td>0.357679</td>\n",
       "      <td>0.268658</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.193303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tet-dCas9-KRAB_rho_ave_LG_JEV</th>\n",
       "      <td>0.306273</td>\n",
       "      <td>0.518444</td>\n",
       "      <td>0.193303</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  dCas9_rho_ave_LG2_LG_MH2_MH  \\\n",
       "dCas9_rho_ave_LG2_LG_MH2_MH                          1.000000   \n",
       "dCas9-KRAB_rho_ave_LG2_LG_MH2_MH                     0.488875   \n",
       "tet-dCas9_rho_ave_LG_JEV                             0.357679   \n",
       "tet-dCas9-KRAB_rho_ave_LG_JEV                        0.306273   \n",
       "\n",
       "                                  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "dCas9_rho_ave_LG2_LG_MH2_MH                               0.488875   \n",
       "dCas9-KRAB_rho_ave_LG2_LG_MH2_MH                          1.000000   \n",
       "tet-dCas9_rho_ave_LG_JEV                                  0.268658   \n",
       "tet-dCas9-KRAB_rho_ave_LG_JEV                             0.518444   \n",
       "\n",
       "                                  tet-dCas9_rho_ave_LG_JEV  \\\n",
       "dCas9_rho_ave_LG2_LG_MH2_MH                       0.357679   \n",
       "dCas9-KRAB_rho_ave_LG2_LG_MH2_MH                  0.268658   \n",
       "tet-dCas9_rho_ave_LG_JEV                          1.000000   \n",
       "tet-dCas9-KRAB_rho_ave_LG_JEV                     0.193303   \n",
       "\n",
       "                                  tet-dCas9-KRAB_rho_ave_LG_JEV  \n",
       "dCas9_rho_ave_LG2_LG_MH2_MH                            0.306273  \n",
       "dCas9-KRAB_rho_ave_LG2_LG_MH2_MH                       0.518444  \n",
       "tet-dCas9_rho_ave_LG_JEV                               0.193303  \n",
       "tet-dCas9-KRAB_rho_ave_LG_JEV                          1.000000  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rho_dcas = seq_phenotpye[['dCas9_rho_ave_LG2_LG_MH2_MH', 'dCas9-KRAB_rho_ave_LG2_LG_MH2_MH', \n",
    "                       'tet-dCas9_rho_ave_LG_JEV', 'tet-dCas9-KRAB_rho_ave_LG_JEV']].dropna().astype(float)\n",
    "rho_dcas.corr()\n",
    "# those three methods don't correlate well so just gonna move forward with dCas9-KRAB\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae8eabb2",
   "metadata": {},
   "source": [
    "## analysis using dCas9-KRAB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d7c03f42",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_detail.columns = ['guide id', 'oligo']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c529d245",
   "metadata": {},
   "outputs": [],
   "source": [
    "dcas_krab = pd.merge(seq_phenotpye[['guide id', 'dCas9-KRAB_rho_ave_LG2_LG_MH2_MH']],seq_detail, on = 'guide id' ).dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "323c2551",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1_c_118438373.25</td>\n",
       "      <td>-0.120761</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGAAAACCTACCAGAGGTAAACAGT...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>c_118438373.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1_c_118438478.24</td>\n",
       "      <td>0.012341</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGATCTGTTTTGCTCGtaatagGTT...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>c_118438478.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1_c_118438478.26</td>\n",
       "      <td>0.029811</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGTGATCTGTTTTGCTCGtaatagG...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>c_118438478.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1_c_118438508.26</td>\n",
       "      <td>0.022175</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgtaatgtgcagtacttactatgtG...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>c_118438508.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1_c_118438682.27</td>\n",
       "      <td>-0.031622</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgcgctgaaaatagcagccaaacac...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>c_118438682.27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               guide id dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "0  ARCN1_c_118438373.25                        -0.120761   \n",
       "1  ARCN1_c_118438478.24                         0.012341   \n",
       "2  ARCN1_c_118438478.26                         0.029811   \n",
       "3  ARCN1_c_118438508.26                         0.022175   \n",
       "4  ARCN1_c_118438682.27                        -0.031622   \n",
       "\n",
       "                                               oligo   Gene        Position  \n",
       "0  cccttggagaaCCAcctTGTTGGAAAACCTACCAGAGGTAAACAGT...  ARCN1  c_118438373.25  \n",
       "1  cccttggagaaCCAcctTGTTGGATCTGTTTTGCTCGtaatagGTT...  ARCN1  c_118438478.24  \n",
       "2  cccttggagaaCCAcctTGTTGGTGATCTGTTTTGCTCGtaatagG...  ARCN1  c_118438478.26  \n",
       "3  cccttggagaaCCAcctTGTTGgtaatgtgcagtacttactatgtG...  ARCN1  c_118438508.26  \n",
       "4  cccttggagaaCCAcctTGTTGgcgctgaaaatagcagccaaacac...  ARCN1  c_118438682.27  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dcas_krab[['Gene', 'Position']] = dcas_krab['guide id'].str.split('_', 1, expand=True)\n",
    "dcas_krab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0bceb8d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#dcas_krab.to_csv('/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2023/k562_tiling/CRISPRi/dcas_krab.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ef4b9ab6",
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_49 = new_tiling['gene symbol'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "72b01f7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import difflib\n",
    "# # https://stackoverflow.com/questions/14128763/how-to-find-the-overlap-between-2-sequences-and-return-it\n",
    "# def get_overlap(s1, s2):\n",
    "#     s = difflib.SequenceMatcher(None, s1, s2)\n",
    "#     pos_a, pos_b, size = s.find_longest_match(0, len(s1), 0, len(s2)) \n",
    "#     return s1[pos_a:pos_a+size]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c078a645",
   "metadata": {},
   "outputs": [],
   "source": [
    "# match_seq_list = []\n",
    "# for gene in genes_49:\n",
    "#     gene_tiling = new_tiling[new_tiling[\"gene symbol\"] == gene].sort_values('sgRNA length (including PAM)', ascending=False)\n",
    "#     gene_pheno = dcas_krab[dcas_krab.Gene == gene]\n",
    "\n",
    "#     for s1 in gene_tiling['sgRNA sequence']:\n",
    "\n",
    "#         for index, row in gene_pheno.iterrows():\n",
    "#             s2 = row['oligo']\n",
    "#             overlap_seq = get_overlap(s1, s2)\n",
    "#             if overlap_seq == s1:\n",
    "#                 row['guide seq'] = s1\n",
    "#                 match_seq_list.append(dict(row))\n",
    "#                 gene_pheno = gene_pheno.drop(index=index)\n",
    "    \n",
    "#     #match_seq_list_genes_49.append(match_seq_list)\n",
    "#             #print(overlap_seq, s2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "21ba293d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'C4orf41', 'RPS25', 'TRAPPC11'}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(new_tiling['gene symbol'].unique()) ^ set(['ARCN1', 'ARF1', 'ARL1', 'C17orf75', 'C4orf41', 'CARM1', 'CCT7',\n",
    "       'COG1', 'COG3', 'COPA', 'COPB1', 'COPZ1', 'CREBBP', 'EIF6', 'GBF1',\n",
    "       'GOLGA1', 'HEATR1', 'IGF2R', 'ILF2', 'ILF3', 'MCL1', 'PRKDC',\n",
    "       'PSMC6', 'QSOX2', 'RAB1A', 'RAB1B', 'RAB2A', 'RAB5C', 'RAD21',\n",
    "       'RPS25', 'SAR1A', 'SCFD1', 'SEC16A', 'SEC22B', 'SEC23B', 'SEC23IP',\n",
    "       'SEC24A', 'SEC24C', 'SMC3', 'ST3GAL4', 'SURF4', 'SYNCRIP',\n",
    "       'TM9SF2', 'TRAPPC8', 'TUBA1B', 'UGP2', 'VPS53', 'VPS54', 'WDR11','YIPF5'])\n",
    "\n",
    "#RPS25 and C4orf41 only in CRISPRi\n",
    "#TRAPPC11 only in CRISPRko\n",
    "#50 genes in CRISPRi instead of 49\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "70c1f976",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "50"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# there are 50 genes in CRISPRi \n",
    "len(['ARCN1', 'ARF1', 'ARL1', 'C17orf75', 'C4orf41', 'CARM1', 'CCT7',\n",
    "       'COG1', 'COG3', 'COPA', 'COPB1', 'COPZ1', 'CREBBP', 'EIF6', 'GBF1',\n",
    "       'GOLGA1', 'HEATR1', 'IGF2R', 'ILF2', 'ILF3', 'MCL1', 'PRKDC',\n",
    "       'PSMC6', 'QSOX2', 'RAB1A', 'RAB1B', 'RAB2A', 'RAB5C', 'RAD21',\n",
    "       'RPS25', 'SAR1A', 'SCFD1', 'SEC16A', 'SEC22B', 'SEC23B', 'SEC23IP',\n",
    "       'SEC24A', 'SEC24C', 'SMC3', 'ST3GAL4', 'SURF4', 'SYNCRIP',\n",
    "       'TM9SF2', 'TRAPPC8', 'TUBA1B', 'UGP2', 'VPS53', 'VPS54', 'WDR11',\n",
    "       'YIPF5'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4743eaf1",
   "metadata": {},
   "source": [
    "## collect all the file in CRISPRi path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "fd8a1733",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "#read the path\n",
    "file_path = '/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2023/k562_tiling/CRISPRi/'\n",
    "file_list = os.listdir(file_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b5c01bf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "52"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(file_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "56bd13ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PRKDC.csv',\n",
       " 'QSOX2.csv',\n",
       " 'SEC23B.csv',\n",
       " 'CREBBP.csv',\n",
       " 'RAD21.csv',\n",
       " 'dcas_krab.csv',\n",
       " 'UGP2.csv',\n",
       " 'RAB2A.csv',\n",
       " 'ARF1.csv',\n",
       " 'ILF3.csv',\n",
       " 'HEATR1.csv',\n",
       " 'ILF2.csv',\n",
       " 'COPZ1.csv',\n",
       " 'COG1.csv',\n",
       " 'TUBA1B.csv',\n",
       " 'RPS25.csv',\n",
       " 'COG3.csv',\n",
       " 'COPB1.csv',\n",
       " 'SCFD1.csv',\n",
       " 'SEC22B.csv',\n",
       " 'SEC23IP.csv',\n",
       " 'SAR1A.csv',\n",
       " 'SURF4.csv',\n",
       " 'EIF6.csv',\n",
       " 'ARL1.csv',\n",
       " 'ST3GAL4.csv',\n",
       " 'TRAPPC8.csv',\n",
       " 'C4orf41.csv',\n",
       " 'WDR11.csv',\n",
       " 'SEC24C.csv',\n",
       " 'ARCN1.csv',\n",
       " 'YIPF5.csv',\n",
       " 'SEC24A.csv',\n",
       " 'RAB5C.csv',\n",
       " 'MCL1.csv',\n",
       " 'GOLGA1.csv',\n",
       " 'VPS54.csv',\n",
       " 'TM9SF2.csv',\n",
       " 'CCT7.csv',\n",
       " 'sgrna_map_crispri.csv',\n",
       " 'VPS53.csv',\n",
       " 'COPA.csv',\n",
       " 'PSMC6.csv',\n",
       " 'SYNCRIP.csv',\n",
       " 'C17orf75.csv',\n",
       " 'SMC3.csv',\n",
       " 'GBF1.csv',\n",
       " 'IGF2R.csv',\n",
       " 'RAB1B.csv',\n",
       " 'CARM1.csv',\n",
       " 'SEC16A.csv',\n",
       " 'RAB1A.csv']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2a4afdc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_list.remove('dcas_krab.csv')\n",
    "file_list.remove('sgrna_map_crispri.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "af7192d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "50"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(file_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "40a20897",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_append = pd.DataFrame()\n",
    "#append all files together\n",
    "for file in file_list:\n",
    "    try:\n",
    "        df_temp = pd.read_csv(file_path+file)\n",
    "        df_append = pd.concat([df_append,df_temp])\n",
    "    except:\n",
    "        pass\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "da84e9fb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "51448\n"
     ]
    }
   ],
   "source": [
    "# Drop negative controls and mismatches \n",
    "df_append = df_append[df_append['guide id'].str[:3] != \"neg\"]\n",
    "df_append = df_append[df_append['guide id'].str[:3] != \"mis\"]\n",
    "print(len(df_append))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6d7d36e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Position</th>\n",
       "      <th>guide seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRKDC_c_48868158.28</td>\n",
       "      <td>-0.016839</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCATTAGGTTTTGCCAAAGCCAGT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>c_48868158.28</td>\n",
       "      <td>GCATTAGGTTTTGCCAAAGCCAGTA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRKDC_w_48873388.28</td>\n",
       "      <td>0.093430</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCCTCCGGGAGGCGCCACTGCGCA...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873388.28</td>\n",
       "      <td>GCCTCCGGGAGGCGCCACTGCGCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKDC_w_48873192.27</td>\n",
       "      <td>-0.022558</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGAGCGTGTGCTTCTTTATCCCTT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873192.27</td>\n",
       "      <td>GGGAGCGTGTGCTTCTTTATCCCTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRKDC_w_48873192.28</td>\n",
       "      <td>0.003080</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGGAGCGTGTGCTTCTTTATCCCT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873192.28</td>\n",
       "      <td>GGGAGCGTGTGCTTCTTTATCCCTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRKDC_w_48873217.27</td>\n",
       "      <td>0.096445</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGTGTGTTCCTCTAGCGTTGGCTA...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873217.27</td>\n",
       "      <td>GGGTGTGTTCCTCTAGCGTTGGCTA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>962</th>\n",
       "      <td>RAB1A_c_65357389.22</td>\n",
       "      <td>0.005347</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGACAGTGCGACGCAAGGCCGTTTA...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>c_65357389.22</td>\n",
       "      <td>GACAGTGCGACGCAAGGCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>963</th>\n",
       "      <td>RAB1A_c_65356889.22</td>\n",
       "      <td>0.316833</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGACAATGGGGCCCGACTCCGTTTA...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>c_65356889.22</td>\n",
       "      <td>GACAATGGGGCCCGACTCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>964</th>\n",
       "      <td>RAB1A_c_65356389.22</td>\n",
       "      <td>0.034136</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGTTTTTATAGGAGGGCAAGTTTA...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>c_65356389.22</td>\n",
       "      <td>GGTTTTTATAGGAGGGCAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>RAB1A_w_65357234.22</td>\n",
       "      <td>0.029229</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGTTCGCCGGCGGCCATTTGTTTA...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>w_65357234.22</td>\n",
       "      <td>GGTTCGCCGGCGGCCATTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966</th>\n",
       "      <td>RAB1A_w_65357270.22</td>\n",
       "      <td>0.356901</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGTAAAGGCCCCTCGGTCTAGTTTA...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>w_65357270.22</td>\n",
       "      <td>GTAAAGGCCCCTCGGTCTA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51448 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                guide id  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "0    PRKDC_c_48868158.28                         -0.016839   \n",
       "1    PRKDC_w_48873388.28                          0.093430   \n",
       "2    PRKDC_w_48873192.27                         -0.022558   \n",
       "3    PRKDC_w_48873192.28                          0.003080   \n",
       "4    PRKDC_w_48873217.27                          0.096445   \n",
       "..                   ...                               ...   \n",
       "962  RAB1A_c_65357389.22                          0.005347   \n",
       "963  RAB1A_c_65356889.22                          0.316833   \n",
       "964  RAB1A_c_65356389.22                          0.034136   \n",
       "965  RAB1A_w_65357234.22                          0.029229   \n",
       "966  RAB1A_w_65357270.22                          0.356901   \n",
       "\n",
       "                                                 oligo   Gene       Position  \\\n",
       "0    cccttggagaaCCAcctTGTTGGCATTAGGTTTTGCCAAAGCCAGT...  PRKDC  c_48868158.28   \n",
       "1    cccttggagaaCCAcctTGTTGGCCTCCGGGAGGCGCCACTGCGCA...  PRKDC  w_48873388.28   \n",
       "2    cccttggagaaCCAcctTGTTGGGAGCGTGTGCTTCTTTATCCCTT...  PRKDC  w_48873192.27   \n",
       "3    cccttggagaaCCAcctTGTTGGGGAGCGTGTGCTTCTTTATCCCT...  PRKDC  w_48873192.28   \n",
       "4    cccttggagaaCCAcctTGTTGGGTGTGTTCCTCTAGCGTTGGCTA...  PRKDC  w_48873217.27   \n",
       "..                                                 ...    ...            ...   \n",
       "962  cccttggagaaCCAcctTGTTGGACAGTGCGACGCAAGGCCGTTTA...  RAB1A  c_65357389.22   \n",
       "963  cccttggagaaCCAcctTGTTGGACAATGGGGCCCGACTCCGTTTA...  RAB1A  c_65356889.22   \n",
       "964  cccttggagaaCCAcctTGTTGGGTTTTTATAGGAGGGCAAGTTTA...  RAB1A  c_65356389.22   \n",
       "965  cccttggagaaCCAcctTGTTGGGTTCGCCGGCGGCCATTTGTTTA...  RAB1A  w_65357234.22   \n",
       "966  cccttggagaaCCAcctTGTTGGTAAAGGCCCCTCGGTCTAGTTTA...  RAB1A  w_65357270.22   \n",
       "\n",
       "                     guide seq  \n",
       "0    GCATTAGGTTTTGCCAAAGCCAGTA  \n",
       "1    GCCTCCGGGAGGCGCCACTGCGCAT  \n",
       "2    GGGAGCGTGTGCTTCTTTATCCCTT  \n",
       "3    GGGAGCGTGTGCTTCTTTATCCCTT  \n",
       "4    GGGTGTGTTCCTCTAGCGTTGGCTA  \n",
       "..                         ...  \n",
       "962        GACAGTGCGACGCAAGGCC  \n",
       "963        GACAATGGGGCCCGACTCC  \n",
       "964        GGTTTTTATAGGAGGGCAA  \n",
       "965        GGTTCGCCGGCGGCCATTT  \n",
       "966        GTAAAGGCCCCTCGGTCTA  \n",
       "\n",
       "[51448 rows x 6 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_append"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d2ee75d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "51448\n"
     ]
    }
   ],
   "source": [
    "# Drop rows where rho values are null\n",
    "df_append = df_append[df_append[\"dCas9-KRAB_rho_ave_LG2_LG_MH2_MH\"].notnull()]\n",
    "print(len(df_append))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "1ee086e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_append[\"gene symbol\"] = df_append['guide id'].str.split(\"_\").str[0]\n",
    "df_append[\"temp\"] = df_append['guide id'].str.split(\"_\").str[2]\n",
    "df_append[\"pam coordinate (hg19)\"] = df_append.temp.str.split(\".\").str[0]\n",
    "df_append[\"length (including pam)\"] = df_append.temp.str.split(\".\").str[1]\n",
    "df_append.drop('temp', axis=1, inplace=True)\n",
    "\n",
    "df_append[\"pam coordinate (hg19)\"] = df_append[\"pam coordinate (hg19)\"].astype(np.int64)\n",
    "df_append[\"length (including pam)\"] = df_append[\"length (including pam)\"].astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e2d4d4a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Position</th>\n",
       "      <th>guide seq</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>pam coordinate (hg19)</th>\n",
       "      <th>length (including pam)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>698</th>\n",
       "      <td>PRKDC_w_48873382.23</td>\n",
       "      <td>0.024931</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCGCCACTGCGCATCGGCAGGTTT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873382.23</td>\n",
       "      <td>GGCGCCACTGCGCATCGGCAGG</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>48873382</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>734</th>\n",
       "      <td>PRKDC_w_48873783.23</td>\n",
       "      <td>-0.085888</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGCTGTAGTGGGCCCGGCTCGTTT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873783.23</td>\n",
       "      <td>GGGCTGTAGTGGGCCCGGCTCG</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>48873783</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>PRKDC_w_48873767.23</td>\n",
       "      <td>-0.017819</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCTCGGGGGACTCACGCGTCGTTT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873767.23</td>\n",
       "      <td>GGCTCGGGGGACTCACGCGTC</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>48873767</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>843</th>\n",
       "      <td>PRKDC_w_48873760.23</td>\n",
       "      <td>0.003155</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGACTCACGCGTCTGGGCGGGTTT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>w_48873760.23</td>\n",
       "      <td>GGGACTCACGCGTCTGGGCGG</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>48873760</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>847</th>\n",
       "      <td>PRKDC_c_48872520.23</td>\n",
       "      <td>-0.014110</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGACCCACCCGCGGCCCAGCTGTTT...</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>c_48872520.23</td>\n",
       "      <td>GGACCCACCCGCGGCCCAGCT</td>\n",
       "      <td>PRKDC</td>\n",
       "      <td>48872520</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>RAB1A_c_65361227.23</td>\n",
       "      <td>0.084612</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgtcccccagagtgatatttgGTTT...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>c_65361227.23</td>\n",
       "      <td>gtcccccagagtgatatttg</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>65361227</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>947</th>\n",
       "      <td>RAB1A_c_65356630.23</td>\n",
       "      <td>-0.088529</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCTCCGCAGAGTGCGGGCGCGTTT...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>c_65356630.23</td>\n",
       "      <td>GCTCCGCAGAGTGCGGGCGC</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>65356630</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>RAB1A_w_65357894.23</td>\n",
       "      <td>0.105124</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGAACCGCTCCGTGCCGTCCAGTTT...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>w_65357894.23</td>\n",
       "      <td>GAACCGCTCCGTGCCGTCCA</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>65357894</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949</th>\n",
       "      <td>RAB1A_c_65361376.23</td>\n",
       "      <td>0.025322</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGTCAGAGGCAGCTTAGCCTGTTT...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>c_65361376.23</td>\n",
       "      <td>GGTCAGAGGCAGCTTAGCCT</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>65361376</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>950</th>\n",
       "      <td>RAB1A_w_65358634.23</td>\n",
       "      <td>-0.072236</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGCAGAATTTGAGGGCTAAGGTTT...</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>w_65358634.23</td>\n",
       "      <td>GGCAGAATTTGAGGGCTAAG</td>\n",
       "      <td>RAB1A</td>\n",
       "      <td>65358634</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2886 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                guide id  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "698  PRKDC_w_48873382.23                          0.024931   \n",
       "734  PRKDC_w_48873783.23                         -0.085888   \n",
       "836  PRKDC_w_48873767.23                         -0.017819   \n",
       "843  PRKDC_w_48873760.23                          0.003155   \n",
       "847  PRKDC_c_48872520.23                         -0.014110   \n",
       "..                   ...                               ...   \n",
       "946  RAB1A_c_65361227.23                          0.084612   \n",
       "947  RAB1A_c_65356630.23                         -0.088529   \n",
       "948  RAB1A_w_65357894.23                          0.105124   \n",
       "949  RAB1A_c_65361376.23                          0.025322   \n",
       "950  RAB1A_w_65358634.23                         -0.072236   \n",
       "\n",
       "                                                 oligo   Gene       Position  \\\n",
       "698  cccttggagaaCCAcctTGTTGGCGCCACTGCGCATCGGCAGGTTT...  PRKDC  w_48873382.23   \n",
       "734  cccttggagaaCCAcctTGTTGGGCTGTAGTGGGCCCGGCTCGTTT...  PRKDC  w_48873783.23   \n",
       "836  cccttggagaaCCAcctTGTTGGCTCGGGGGACTCACGCGTCGTTT...  PRKDC  w_48873767.23   \n",
       "843  cccttggagaaCCAcctTGTTGGGACTCACGCGTCTGGGCGGGTTT...  PRKDC  w_48873760.23   \n",
       "847  cccttggagaaCCAcctTGTTGGACCCACCCGCGGCCCAGCTGTTT...  PRKDC  c_48872520.23   \n",
       "..                                                 ...    ...            ...   \n",
       "946  cccttggagaaCCAcctTGTTGgtcccccagagtgatatttgGTTT...  RAB1A  c_65361227.23   \n",
       "947  cccttggagaaCCAcctTGTTGGCTCCGCAGAGTGCGGGCGCGTTT...  RAB1A  c_65356630.23   \n",
       "948  cccttggagaaCCAcctTGTTGGAACCGCTCCGTGCCGTCCAGTTT...  RAB1A  w_65357894.23   \n",
       "949  cccttggagaaCCAcctTGTTGGGTCAGAGGCAGCTTAGCCTGTTT...  RAB1A  c_65361376.23   \n",
       "950  cccttggagaaCCAcctTGTTGGGCAGAATTTGAGGGCTAAGGTTT...  RAB1A  w_65358634.23   \n",
       "\n",
       "                  guide seq gene symbol  pam coordinate (hg19)  \\\n",
       "698  GGCGCCACTGCGCATCGGCAGG       PRKDC               48873382   \n",
       "734  GGGCTGTAGTGGGCCCGGCTCG       PRKDC               48873783   \n",
       "836   GGCTCGGGGGACTCACGCGTC       PRKDC               48873767   \n",
       "843   GGGACTCACGCGTCTGGGCGG       PRKDC               48873760   \n",
       "847   GGACCCACCCGCGGCCCAGCT       PRKDC               48872520   \n",
       "..                      ...         ...                    ...   \n",
       "946    gtcccccagagtgatatttg       RAB1A               65361227   \n",
       "947    GCTCCGCAGAGTGCGGGCGC       RAB1A               65356630   \n",
       "948    GAACCGCTCCGTGCCGTCCA       RAB1A               65357894   \n",
       "949    GGTCAGAGGCAGCTTAGCCT       RAB1A               65361376   \n",
       "950    GGCAGAATTTGAGGGCTAAG       RAB1A               65358634   \n",
       "\n",
       "     length (including pam)  \n",
       "698                      23  \n",
       "734                      23  \n",
       "836                      23  \n",
       "843                      23  \n",
       "847                      23  \n",
       "..                      ...  \n",
       "946                      23  \n",
       "947                      23  \n",
       "948                      23  \n",
       "949                      23  \n",
       "950                      23  \n",
       "\n",
       "[2886 rows x 9 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_append[df_append['length (including pam)'] ==23]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "f4733d8d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len('GACAGTGCGACGCAAGGCC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "00883ba1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>Gene dcas krab</th>\n",
       "      <th>Gene mapped</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ST3GAL4</th>\n",
       "      <td>3998</td>\n",
       "      <td>3997.0</td>\n",
       "      <td>3997.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB1B</th>\n",
       "      <td>2245</td>\n",
       "      <td>2247.0</td>\n",
       "      <td>2247.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>QSOX2</th>\n",
       "      <td>2135</td>\n",
       "      <td>2136.0</td>\n",
       "      <td>2134.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARF1</th>\n",
       "      <td>2105</td>\n",
       "      <td>2104.0</td>\n",
       "      <td>2104.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC16A</th>\n",
       "      <td>1991</td>\n",
       "      <td>1995.0</td>\n",
       "      <td>1992.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EIF6</th>\n",
       "      <td>1768</td>\n",
       "      <td>1768.0</td>\n",
       "      <td>1768.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SURF4</th>\n",
       "      <td>1656</td>\n",
       "      <td>1659.0</td>\n",
       "      <td>1658.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CREBBP</th>\n",
       "      <td>1632</td>\n",
       "      <td>1638.0</td>\n",
       "      <td>1634.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GBF1</th>\n",
       "      <td>1515</td>\n",
       "      <td>1519.0</td>\n",
       "      <td>1517.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CARM1</th>\n",
       "      <td>1473</td>\n",
       "      <td>1475.0</td>\n",
       "      <td>1475.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCT7</th>\n",
       "      <td>1424</td>\n",
       "      <td>1424.0</td>\n",
       "      <td>1424.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SYNCRIP</th>\n",
       "      <td>1270</td>\n",
       "      <td>1270.0</td>\n",
       "      <td>1269.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB5C</th>\n",
       "      <td>1208</td>\n",
       "      <td>1211.0</td>\n",
       "      <td>1211.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COG1</th>\n",
       "      <td>1198</td>\n",
       "      <td>1199.0</td>\n",
       "      <td>1199.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGF2R</th>\n",
       "      <td>1152</td>\n",
       "      <td>1153.0</td>\n",
       "      <td>1153.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UGP2</th>\n",
       "      <td>1123</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>1123.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ILF3</th>\n",
       "      <td>997</td>\n",
       "      <td>999.0</td>\n",
       "      <td>997.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRKDC</th>\n",
       "      <td>979</td>\n",
       "      <td>980.0</td>\n",
       "      <td>979.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB2A</th>\n",
       "      <td>971</td>\n",
       "      <td>972.0</td>\n",
       "      <td>971.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB1A</th>\n",
       "      <td>967</td>\n",
       "      <td>967.0</td>\n",
       "      <td>967.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COPA</th>\n",
       "      <td>929</td>\n",
       "      <td>933.0</td>\n",
       "      <td>930.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TM9SF2</th>\n",
       "      <td>921</td>\n",
       "      <td>922.0</td>\n",
       "      <td>921.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMC3</th>\n",
       "      <td>904</td>\n",
       "      <td>904.0</td>\n",
       "      <td>904.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MCL1</th>\n",
       "      <td>903</td>\n",
       "      <td>904.0</td>\n",
       "      <td>903.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAR1A</th>\n",
       "      <td>901</td>\n",
       "      <td>901.0</td>\n",
       "      <td>901.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VPS53</th>\n",
       "      <td>841</td>\n",
       "      <td>842.0</td>\n",
       "      <td>841.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VPS54</th>\n",
       "      <td>834</td>\n",
       "      <td>834.0</td>\n",
       "      <td>833.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC23B</th>\n",
       "      <td>799</td>\n",
       "      <td>800.0</td>\n",
       "      <td>800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TUBA1B</th>\n",
       "      <td>775</td>\n",
       "      <td>778.0</td>\n",
       "      <td>776.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRAPPC8</th>\n",
       "      <td>755</td>\n",
       "      <td>757.0</td>\n",
       "      <td>756.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GOLGA1</th>\n",
       "      <td>747</td>\n",
       "      <td>747.0</td>\n",
       "      <td>747.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC23IP</th>\n",
       "      <td>728</td>\n",
       "      <td>730.0</td>\n",
       "      <td>728.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COG3</th>\n",
       "      <td>726</td>\n",
       "      <td>727.0</td>\n",
       "      <td>726.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAD21</th>\n",
       "      <td>694</td>\n",
       "      <td>694.0</td>\n",
       "      <td>694.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC22B</th>\n",
       "      <td>692</td>\n",
       "      <td>703.0</td>\n",
       "      <td>693.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRAPPC11</th>\n",
       "      <td>683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C17orf75</th>\n",
       "      <td>639</td>\n",
       "      <td>640.0</td>\n",
       "      <td>639.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YIPF5</th>\n",
       "      <td>614</td>\n",
       "      <td>614.0</td>\n",
       "      <td>614.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ILF2</th>\n",
       "      <td>610</td>\n",
       "      <td>612.0</td>\n",
       "      <td>612.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC24C</th>\n",
       "      <td>609</td>\n",
       "      <td>610.0</td>\n",
       "      <td>609.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COPZ1</th>\n",
       "      <td>606</td>\n",
       "      <td>607.0</td>\n",
       "      <td>606.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARL1</th>\n",
       "      <td>596</td>\n",
       "      <td>596.0</td>\n",
       "      <td>596.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSMC6</th>\n",
       "      <td>589</td>\n",
       "      <td>589.0</td>\n",
       "      <td>589.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HEATR1</th>\n",
       "      <td>568</td>\n",
       "      <td>571.0</td>\n",
       "      <td>568.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COPB1</th>\n",
       "      <td>557</td>\n",
       "      <td>558.0</td>\n",
       "      <td>558.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC24A</th>\n",
       "      <td>556</td>\n",
       "      <td>556.0</td>\n",
       "      <td>556.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARCN1</th>\n",
       "      <td>552</td>\n",
       "      <td>553.0</td>\n",
       "      <td>552.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SCFD1</th>\n",
       "      <td>525</td>\n",
       "      <td>525.0</td>\n",
       "      <td>525.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WDR11</th>\n",
       "      <td>452</td>\n",
       "      <td>452.0</td>\n",
       "      <td>452.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene symbol  Gene dcas krab  Gene mapped\n",
       "ST3GAL4          3998          3997.0       3997.0\n",
       "RAB1B            2245          2247.0       2247.0\n",
       "QSOX2            2135          2136.0       2134.0\n",
       "ARF1             2105          2104.0       2104.0\n",
       "SEC16A           1991          1995.0       1992.0\n",
       "EIF6             1768          1768.0       1768.0\n",
       "SURF4            1656          1659.0       1658.0\n",
       "CREBBP           1632          1638.0       1634.0\n",
       "GBF1             1515          1519.0       1517.0\n",
       "CARM1            1473          1475.0       1475.0\n",
       "CCT7             1424          1424.0       1424.0\n",
       "SYNCRIP          1270          1270.0       1269.0\n",
       "RAB5C            1208          1211.0       1211.0\n",
       "COG1             1198          1199.0       1199.0\n",
       "IGF2R            1152          1153.0       1153.0\n",
       "UGP2             1123          1123.0       1123.0\n",
       "ILF3              997           999.0        997.0\n",
       "PRKDC             979           980.0        979.0\n",
       "RAB2A             971           972.0        971.0\n",
       "RAB1A             967           967.0        967.0\n",
       "COPA              929           933.0        930.0\n",
       "TM9SF2            921           922.0        921.0\n",
       "SMC3              904           904.0        904.0\n",
       "MCL1              903           904.0        903.0\n",
       "SAR1A             901           901.0        901.0\n",
       "VPS53             841           842.0        841.0\n",
       "VPS54             834           834.0        833.0\n",
       "SEC23B            799           800.0        800.0\n",
       "TUBA1B            775           778.0        776.0\n",
       "TRAPPC8           755           757.0        756.0\n",
       "GOLGA1            747           747.0        747.0\n",
       "SEC23IP           728           730.0        728.0\n",
       "COG3              726           727.0        726.0\n",
       "RAD21             694           694.0        694.0\n",
       "SEC22B            692           703.0        693.0\n",
       "TRAPPC11          683             NaN          NaN\n",
       "C17orf75          639           640.0        639.0\n",
       "YIPF5             614           614.0        614.0\n",
       "ILF2              610           612.0        612.0\n",
       "SEC24C            609           610.0        609.0\n",
       "COPZ1             606           607.0        606.0\n",
       "ARL1              596           596.0        596.0\n",
       "PSMC6             589           589.0        589.0\n",
       "HEATR1            568           571.0        568.0\n",
       "COPB1             557           558.0        558.0\n",
       "SEC24A            556           556.0        556.0\n",
       "ARCN1             552           553.0        552.0\n",
       "SCFD1             525           525.0        525.0\n",
       "WDR11             452           452.0        452.0"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_df_krab = pd.DataFrame(dcas_krab[dcas_krab.Gene.isin(['ARCN1', 'ARF1', 'ARL1', 'C17orf75', 'C4orf41', 'CARM1', 'CCT7',\n",
    "       'COG1', 'COG3', 'COPA', 'COPB1', 'COPZ1', 'CREBBP', 'EIF6', 'GBF1',\n",
    "       'GOLGA1', 'HEATR1', 'IGF2R', 'ILF2', 'ILF3', 'MCL1', 'PRKDC',\n",
    "       'PSMC6', 'QSOX2', 'RAB1A', 'RAB1B', 'RAB2A', 'RAB5C', 'RAD21',\n",
    "       'RPS25', 'SAR1A', 'SCFD1', 'SEC16A', 'SEC22B', 'SEC23B', 'SEC23IP',\n",
    "       'SEC24A', 'SEC24C', 'SMC3', 'ST3GAL4', 'SURF4', 'SYNCRIP',\n",
    "       'TM9SF2', 'TRAPPC8', 'TUBA1B', 'UGP2', 'VPS53', 'VPS54', 'WDR11',\n",
    "       'YIPF5'])].Gene.value_counts()).rename(columns = {'Gene': 'Gene dcas krab'})\n",
    "count_df_append = pd.DataFrame(df_append.Gene.value_counts()).rename(columns = {'Gene': 'Gene mapped'} )\n",
    "pd.DataFrame(new_tiling['gene symbol'].value_counts()).join(count_df_krab.join(count_df_append))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "0489f86d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# tiling_together = pd.merge(df_append.rename(columns={'guide seq':'sgRNA sequence', \n",
    "#                                                      'pam coordinate (hg19)':'PAM genomic coordinate [hg19]',\n",
    "#                                                     'length (including pam)': 'sgRNA length (including PAM)'}),\n",
    "#                            new_tiling, on = ['sgRNA sequence', 'gene symbol', \n",
    "#                                              'PAM genomic coordinate [hg19]', 'sgRNA length (including PAM)'])\n",
    "\n",
    "tiling_together = pd.merge(df_append.rename(columns={'guide seq':'sgRNA sequence'}),\n",
    "                           new_tiling, on = ['sgRNA sequence', 'gene symbol'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "5f417bfc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GGAGCTCGGGGACGAACGGGGGCCG    6\n",
       "GGGAGGTTTGTAACCACCTGTGGTG    6\n",
       "GGATGGGGCGGGCGGGAGCTCGGTG    6\n",
       "GGCCAGGCATACAAGTCCACGACGG    5\n",
       "GGGGAGGTTGAGGATCAGAGATCAG    5\n",
       "                            ..\n",
       "GGGCAGAGAGGCGTAACGCTGTG      1\n",
       "GCGTAAATTTTTAAATTGGCTAA      1\n",
       "gttagagatgagggtttaagccc      1\n",
       "GTTAACAggaagagcttaggctt      1\n",
       "GTAAAGGCCCCTCGGTCTA          1\n",
       "Name: sgRNA sequence, Length: 40553, dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# there are nonunique sgRNA sequence after merging ko and crispri\n",
    "tiling_together['sgRNA sequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "ed034a6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GAGAAACGTCATACTCAAAA         1\n",
       "gcgtctgggcctggcagaagggacc    1\n",
       "gcctggcagaagggaccgggtg       1\n",
       "ggcctggcagaagggaccgggtg      1\n",
       "gggcctggcagaagggaccgggtg     1\n",
       "                            ..\n",
       "gtggaagagagatctgcctaagcaa    1\n",
       "ggaagagagatctgcctaagcaa      1\n",
       "gaagagagatctgcctaagcaa       1\n",
       "gcaaagacagaggctcacccaa       1\n",
       "GTAGCTTCAACTCTATATAGTAGTC    1\n",
       "Name: sgRNA sequence, Length: 557, dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# all the sgrna seq from ko are unique\n",
    "new_tiling[new_tiling['gene symbol'] == 'COPB1']['sgRNA sequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "e318b884",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "cccttggagaaCCAcctTGTTGGTAGCTTCAACTCTATATAGTAGTCGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTaCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGGACTTGCCCCAAATCATGTAAGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtcCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGCATGGAATAAAAGGCAGTggcGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtcCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGGCATGGAATAAAAGGCAGTggcGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGCATACGTTCTCAGCCGCCGTCAGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtCtcgagcgtcgaggatccattagg    1\n",
       "                                                                                                                                                                        ..\n",
       "cccttggagaaCCAcctTGTTGGTGTGGACAAGAAAAAGGATTACGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGgaacaaactgttgtacacacacccGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGGCTGATTATTTTCTCTAGGCAACGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGCTGATTATTTTCTCTAGGCAACGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGgactaggccgcttgcgtctGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtcaggCtcgagcgtcgaggatccattagg    1\n",
       "Name: oligo, Length: 558, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check just one gene\n",
    "COPB1 = tiling_together[tiling_together.Gene == 'COPB1']\n",
    "# all the oligos are unique\n",
    "COPB1.oligo.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "4e870bc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GGCTTTTGAAGATTTGGGAAAGTTG    5\n",
       "GCAGCCTAAAGATCATGCTGGAGTG    3\n",
       "GGAGGAAGGGGAGGAGGAAGCGGGT    3\n",
       "GGGTGTGGACAAGAAAAAGGATTAC    3\n",
       "GGCCTAGGGCTGCTTTGCTGGAGGA    3\n",
       "                            ..\n",
       "gggcctggcagaagggaccgggtg     1\n",
       "gcgtctgggcctggcagaagggac     1\n",
       "GAAAAAGAGCTCTGGACTAGAAAG     1\n",
       "GCTTGGAGAGAGGGAAACTTTTAC     1\n",
       "gactaggccgcttgcgtct          1\n",
       "Name: sgRNA sequence, Length: 464, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# there are nonunique sgRNA sequence\n",
    "COPB1['sgRNA sequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "b5c986c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pam coordinate (hg19)</th>\n",
       "      <th>PAM genomic coordinate [hg19]</th>\n",
       "      <th>length (including pam)</th>\n",
       "      <th>sgRNA length (including PAM)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16141</th>\n",
       "      <td>14521155</td>\n",
       "      <td>14521154</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       pam coordinate (hg19)  PAM genomic coordinate [hg19]  \\\n",
       "16141               14521155                       14521154   \n",
       "\n",
       "       length (including pam)  sgRNA length (including PAM)  \n",
       "16141                      27                            27  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check unique sgRNA\n",
    "COPB1[COPB1['sgRNA sequence'] == 'gcgtctgggcctggcagaagggac'][['pam coordinate (hg19)', 'PAM genomic coordinate [hg19]','length (including pam)', 'sgRNA length (including PAM)']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3bcfff62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Position</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>pam coordinate (hg19)</th>\n",
       "      <th>length (including pam)</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>...</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>sgRNA length (including PAM)</th>\n",
       "      <th>FANTOM 5 TSS [Where P1 and P2 are &gt;1kb apart, TSSs were treated separately and sgRNAs assigned to the closest TSS; otherwise, P1P2 indicates all coordinates are related to the P1 TSS]</th>\n",
       "      <th>Position relative to FANTOM5 TSS</th>\n",
       "      <th>Cas9 nuclease phenotype-signed z-score</th>\n",
       "      <th>Target site MNase signal</th>\n",
       "      <th>Doench et al. Nat. Biotech 2014 sgRNA sequence score</th>\n",
       "      <th>is targeting CDS?</th>\n",
       "      <th>liftover</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16004</th>\n",
       "      <td>COPB1_c_14526027.27</td>\n",
       "      <td>-0.003517</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCAGCCTAAAGATCATGCTGGAGT...</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>c_14526027.27</td>\n",
       "      <td>GCAGCCTAAAGATCATGCTGGAGTG</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>14526027</td>\n",
       "      <td>27</td>\n",
       "      <td>chr11</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>28</td>\n",
       "      <td>P1P2</td>\n",
       "      <td>-4648</td>\n",
       "      <td>0.385861</td>\n",
       "      <td>2.021429</td>\n",
       "      <td>0.513381</td>\n",
       "      <td>False</td>\n",
       "      <td>chr11:14526027-14526027</td>\n",
       "      <td>14504481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16005</th>\n",
       "      <td>COPB1_c_14526027.28</td>\n",
       "      <td>0.007302</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGCAGCCTAAAGATCATGCTGGAG...</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>c_14526027.28</td>\n",
       "      <td>GCAGCCTAAAGATCATGCTGGAGTG</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>14526027</td>\n",
       "      <td>28</td>\n",
       "      <td>chr11</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>28</td>\n",
       "      <td>P1P2</td>\n",
       "      <td>-4648</td>\n",
       "      <td>0.385861</td>\n",
       "      <td>2.021429</td>\n",
       "      <td>0.513381</td>\n",
       "      <td>False</td>\n",
       "      <td>chr11:14526027-14526027</td>\n",
       "      <td>14504481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16006</th>\n",
       "      <td>COPB1_c_14526028.28</td>\n",
       "      <td>0.007732</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCAGCCTAAAGATCATGCTGGAGT...</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>c_14526028.28</td>\n",
       "      <td>GCAGCCTAAAGATCATGCTGGAGTG</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>14526028</td>\n",
       "      <td>28</td>\n",
       "      <td>chr11</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>28</td>\n",
       "      <td>P1P2</td>\n",
       "      <td>-4648</td>\n",
       "      <td>0.385861</td>\n",
       "      <td>2.021429</td>\n",
       "      <td>0.513381</td>\n",
       "      <td>False</td>\n",
       "      <td>chr11:14526027-14526027</td>\n",
       "      <td>14504481</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  guide id  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "16004  COPB1_c_14526027.27                         -0.003517   \n",
       "16005  COPB1_c_14526027.28                          0.007302   \n",
       "16006  COPB1_c_14526028.28                          0.007732   \n",
       "\n",
       "                                                   oligo   Gene  \\\n",
       "16004  cccttggagaaCCAcctTGTTGGCAGCCTAAAGATCATGCTGGAGT...  COPB1   \n",
       "16005  cccttggagaaCCAcctTGTTGGGCAGCCTAAAGATCATGCTGGAG...  COPB1   \n",
       "16006  cccttggagaaCCAcctTGTTGGCAGCCTAAAGATCATGCTGGAGT...  COPB1   \n",
       "\n",
       "            Position             sgRNA sequence gene symbol  \\\n",
       "16004  c_14526027.27  GCAGCCTAAAGATCATGCTGGAGTG       COPB1   \n",
       "16005  c_14526027.28  GCAGCCTAAAGATCATGCTGGAGTG       COPB1   \n",
       "16006  c_14526028.28  GCAGCCTAAAGATCATGCTGGAGTG       COPB1   \n",
       "\n",
       "       pam coordinate (hg19)  length (including pam) chromosome  ...  \\\n",
       "16004               14526027                      27      chr11  ...   \n",
       "16005               14526027                      28      chr11  ...   \n",
       "16006               14526028                      28      chr11  ...   \n",
       "\n",
       "       strand targeted sgRNA length (including PAM)  \\\n",
       "16004                -                           28   \n",
       "16005                -                           28   \n",
       "16006                -                           28   \n",
       "\n",
       "       FANTOM 5 TSS [Where P1 and P2 are >1kb apart, TSSs were treated separately and sgRNAs assigned to the closest TSS; otherwise, P1P2 indicates all coordinates are related to the P1 TSS]  \\\n",
       "16004                                               P1P2                                                                                                                                         \n",
       "16005                                               P1P2                                                                                                                                         \n",
       "16006                                               P1P2                                                                                                                                         \n",
       "\n",
       "      Position relative to FANTOM5 TSS  \\\n",
       "16004                            -4648   \n",
       "16005                            -4648   \n",
       "16006                            -4648   \n",
       "\n",
       "       Cas9 nuclease phenotype-signed z-score  Target site MNase signal  \\\n",
       "16004                                0.385861                  2.021429   \n",
       "16005                                0.385861                  2.021429   \n",
       "16006                                0.385861                  2.021429   \n",
       "\n",
       "       Doench et al. Nat. Biotech 2014 sgRNA sequence score  \\\n",
       "16004                                           0.513381      \n",
       "16005                                           0.513381      \n",
       "16006                                           0.513381      \n",
       "\n",
       "       is targeting CDS?                 liftover  \\\n",
       "16004              False  chr11:14526027-14526027   \n",
       "16005              False  chr11:14526027-14526027   \n",
       "16006              False  chr11:14526027-14526027   \n",
       "\n",
       "      PAM genomic coordinate [hg38]  \n",
       "16004                      14504481  \n",
       "16005                      14504481  \n",
       "16006                      14504481  \n",
       "\n",
       "[3 rows x 21 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check nonunique sgRNA\n",
    "COPB1[COPB1['sgRNA sequence'] == 'GCAGCCTAAAGATCATGCTGGAGTG']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "3c9cd781",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Position</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>pam coordinate (hg19)</th>\n",
       "      <th>length (including pam)</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>...</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>sgRNA length (including PAM)</th>\n",
       "      <th>FANTOM 5 TSS [Where P1 and P2 are &gt;1kb apart, TSSs were treated separately and sgRNAs assigned to the closest TSS; otherwise, P1P2 indicates all coordinates are related to the P1 TSS]</th>\n",
       "      <th>Position relative to FANTOM5 TSS</th>\n",
       "      <th>Cas9 nuclease phenotype-signed z-score</th>\n",
       "      <th>Target site MNase signal</th>\n",
       "      <th>Doench et al. Nat. Biotech 2014 sgRNA sequence score</th>\n",
       "      <th>is targeting CDS?</th>\n",
       "      <th>liftover</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16480</th>\n",
       "      <td>COPB1_w_14521173.22</td>\n",
       "      <td>-0.060035</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgactaggccgcttgcgtctGTTTA...</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>w_14521173.22</td>\n",
       "      <td>gactaggccgcttgcgtct</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>14521173</td>\n",
       "      <td>22</td>\n",
       "      <td>chr11</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>22</td>\n",
       "      <td>P1P2</td>\n",
       "      <td>207</td>\n",
       "      <td>-0.192195</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.193313</td>\n",
       "      <td>False</td>\n",
       "      <td>chr11:14521172-14521172</td>\n",
       "      <td>14499626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  guide id  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "16480  COPB1_w_14521173.22                         -0.060035   \n",
       "\n",
       "                                                   oligo   Gene  \\\n",
       "16480  cccttggagaaCCAcctTGTTGgactaggccgcttgcgtctGTTTA...  COPB1   \n",
       "\n",
       "            Position       sgRNA sequence gene symbol  pam coordinate (hg19)  \\\n",
       "16480  w_14521173.22  gactaggccgcttgcgtct       COPB1               14521173   \n",
       "\n",
       "       length (including pam) chromosome  ...  strand targeted  \\\n",
       "16480                      22      chr11  ...                +   \n",
       "\n",
       "      sgRNA length (including PAM)  \\\n",
       "16480                           22   \n",
       "\n",
       "       FANTOM 5 TSS [Where P1 and P2 are >1kb apart, TSSs were treated separately and sgRNAs assigned to the closest TSS; otherwise, P1P2 indicates all coordinates are related to the P1 TSS]  \\\n",
       "16480                                               P1P2                                                                                                                                         \n",
       "\n",
       "      Position relative to FANTOM5 TSS  \\\n",
       "16480                              207   \n",
       "\n",
       "       Cas9 nuclease phenotype-signed z-score  Target site MNase signal  \\\n",
       "16480                               -0.192195                       0.0   \n",
       "\n",
       "       Doench et al. Nat. Biotech 2014 sgRNA sequence score  \\\n",
       "16480                                           0.193313      \n",
       "\n",
       "       is targeting CDS?                 liftover  \\\n",
       "16480              False  chr11:14521172-14521172   \n",
       "\n",
       "      PAM genomic coordinate [hg38]  \n",
       "16480                      14499626  \n",
       "\n",
       "[1 rows x 21 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check unique sgRNA\n",
    "COPB1[COPB1['sgRNA sequence'] == 'gactaggccgcttgcgtct']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d79035ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Position</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>pam coordinate (hg19)</th>\n",
       "      <th>length (including pam)</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>...</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>sgRNA length (including PAM)</th>\n",
       "      <th>FANTOM 5 TSS [Where P1 and P2 are &gt;1kb apart, TSSs were treated separately and sgRNAs assigned to the closest TSS; otherwise, P1P2 indicates all coordinates are related to the P1 TSS]</th>\n",
       "      <th>Position relative to FANTOM5 TSS</th>\n",
       "      <th>Cas9 nuclease phenotype-signed z-score</th>\n",
       "      <th>Target site MNase signal</th>\n",
       "      <th>Doench et al. Nat. Biotech 2014 sgRNA sequence score</th>\n",
       "      <th>is targeting CDS?</th>\n",
       "      <th>liftover</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16017</th>\n",
       "      <td>COPB1_w_14520974.28</td>\n",
       "      <td>-0.002363</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGAGGAAGGGGAGGAGGAAGCGGGT...</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>w_14520974.28</td>\n",
       "      <td>GGAGGAAGGGGAGGAGGAAGCGGGT</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>14520974</td>\n",
       "      <td>28</td>\n",
       "      <td>chr11</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>28</td>\n",
       "      <td>P1P2</td>\n",
       "      <td>405</td>\n",
       "      <td>0.919909</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.031136</td>\n",
       "      <td>False</td>\n",
       "      <td>chr11:14520974-14520974</td>\n",
       "      <td>14499428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16018</th>\n",
       "      <td>COPB1_w_14520975.27</td>\n",
       "      <td>-0.144419</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGAGGAAGGGGAGGAGGAAGCGGGT...</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>w_14520975.27</td>\n",
       "      <td>GGAGGAAGGGGAGGAGGAAGCGGGT</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>14520975</td>\n",
       "      <td>27</td>\n",
       "      <td>chr11</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>28</td>\n",
       "      <td>P1P2</td>\n",
       "      <td>405</td>\n",
       "      <td>0.919909</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.031136</td>\n",
       "      <td>False</td>\n",
       "      <td>chr11:14520974-14520974</td>\n",
       "      <td>14499428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16019</th>\n",
       "      <td>COPB1_w_14520975.28</td>\n",
       "      <td>-0.065629</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGGAGGAAGGGGAGGAGGAAGCGGG...</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>w_14520975.28</td>\n",
       "      <td>GGAGGAAGGGGAGGAGGAAGCGGGT</td>\n",
       "      <td>COPB1</td>\n",
       "      <td>14520975</td>\n",
       "      <td>28</td>\n",
       "      <td>chr11</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>28</td>\n",
       "      <td>P1P2</td>\n",
       "      <td>405</td>\n",
       "      <td>0.919909</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.031136</td>\n",
       "      <td>False</td>\n",
       "      <td>chr11:14520974-14520974</td>\n",
       "      <td>14499428</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  guide id  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "16017  COPB1_w_14520974.28                         -0.002363   \n",
       "16018  COPB1_w_14520975.27                         -0.144419   \n",
       "16019  COPB1_w_14520975.28                         -0.065629   \n",
       "\n",
       "                                                   oligo   Gene  \\\n",
       "16017  cccttggagaaCCAcctTGTTGGAGGAAGGGGAGGAGGAAGCGGGT...  COPB1   \n",
       "16018  cccttggagaaCCAcctTGTTGGAGGAAGGGGAGGAGGAAGCGGGT...  COPB1   \n",
       "16019  cccttggagaaCCAcctTGTTGGGAGGAAGGGGAGGAGGAAGCGGG...  COPB1   \n",
       "\n",
       "            Position             sgRNA sequence gene symbol  \\\n",
       "16017  w_14520974.28  GGAGGAAGGGGAGGAGGAAGCGGGT       COPB1   \n",
       "16018  w_14520975.27  GGAGGAAGGGGAGGAGGAAGCGGGT       COPB1   \n",
       "16019  w_14520975.28  GGAGGAAGGGGAGGAGGAAGCGGGT       COPB1   \n",
       "\n",
       "       pam coordinate (hg19)  length (including pam) chromosome  ...  \\\n",
       "16017               14520974                      28      chr11  ...   \n",
       "16018               14520975                      27      chr11  ...   \n",
       "16019               14520975                      28      chr11  ...   \n",
       "\n",
       "       strand targeted sgRNA length (including PAM)  \\\n",
       "16017                +                           28   \n",
       "16018                +                           28   \n",
       "16019                +                           28   \n",
       "\n",
       "       FANTOM 5 TSS [Where P1 and P2 are >1kb apart, TSSs were treated separately and sgRNAs assigned to the closest TSS; otherwise, P1P2 indicates all coordinates are related to the P1 TSS]  \\\n",
       "16017                                               P1P2                                                                                                                                         \n",
       "16018                                               P1P2                                                                                                                                         \n",
       "16019                                               P1P2                                                                                                                                         \n",
       "\n",
       "      Position relative to FANTOM5 TSS  \\\n",
       "16017                              405   \n",
       "16018                              405   \n",
       "16019                              405   \n",
       "\n",
       "       Cas9 nuclease phenotype-signed z-score  Target site MNase signal  \\\n",
       "16017                                0.919909                       0.1   \n",
       "16018                                0.919909                       0.1   \n",
       "16019                                0.919909                       0.1   \n",
       "\n",
       "       Doench et al. Nat. Biotech 2014 sgRNA sequence score  \\\n",
       "16017                                           0.031136      \n",
       "16018                                           0.031136      \n",
       "16019                                           0.031136      \n",
       "\n",
       "       is targeting CDS?                 liftover  \\\n",
       "16017              False  chr11:14520974-14520974   \n",
       "16018              False  chr11:14520974-14520974   \n",
       "16019              False  chr11:14520974-14520974   \n",
       "\n",
       "      PAM genomic coordinate [hg38]  \n",
       "16017                      14499428  \n",
       "16018                      14499428  \n",
       "16019                      14499428  \n",
       "\n",
       "[3 rows x 21 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check nonunique sgRNA\n",
    "COPB1[COPB1['sgRNA sequence'] == 'GGAGGAAGGGGAGGAGGAAGCGGGT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "c3221020",
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop duplicate to see approximately the difference between pam coord from ko and crispri\n",
    "tiling_together = tiling_together.drop_duplicates('sgRNA sequence')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ef866102",
   "metadata": {},
   "outputs": [],
   "source": [
    "tiling_together['difference pam coordinate'] = tiling_together['pam coordinate (hg19)'] - tiling_together['PAM genomic coordinate [hg19]']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "38e7518f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       " 1    39064\n",
       " 0     1313\n",
       " 2      171\n",
       "-1        3\n",
       "-2        2\n",
       "Name: difference pam coordinate, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tiling_together['difference pam coordinate'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a487abee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9632826178087934"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# percent of guide off by one\n",
    "39064/40553"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "d218a370",
   "metadata": {},
   "outputs": [],
   "source": [
    "#tiling_together.to_csv('/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2023/k562_tiling/CRISPRi/sgrna_map_crispri.csv', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f8a7d1f",
   "metadata": {},
   "source": [
    "# using above information and Dany's code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3b2ea784",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "52926\n"
     ]
    }
   ],
   "source": [
    "# Drop negative controls and mismatches \n",
    "tiling_crispri = dcas_krab[dcas_krab['guide id'].str[:3] != \"neg\"]\n",
    "tiling_crispri = tiling_crispri[tiling_crispri['guide id'].str[:3] != \"mis\"]\n",
    "print(len(tiling_crispri))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e04b7b25",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "52926\n"
     ]
    }
   ],
   "source": [
    "# Drop rows where rho values are null\n",
    "tiling_crispri = tiling_crispri[tiling_crispri[\"dCas9-KRAB_rho_ave_LG2_LG_MH2_MH\"].notnull()]\n",
    "print(len(tiling_crispri))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "a9bd2a1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "tiling_crispri[\"gene symbol\"] = tiling_crispri['guide id'].str.split(\"_\").str[0]\n",
    "tiling_crispri[\"temp\"] = tiling_crispri['guide id'].str.split(\"_\").str[2]\n",
    "tiling_crispri[\"pam coordinate (hg19)\"] = tiling_crispri.temp.str.split(\".\").str[0]\n",
    "tiling_crispri[\"length (including pam)\"] = tiling_crispri.temp.str.split(\".\").str[1]\n",
    "tiling_crispri.drop('temp', axis=1, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "4ada63bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "tiling_crispri.drop('Position', axis=1, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "5327356f",
   "metadata": {},
   "outputs": [],
   "source": [
    "tiling_crispri.drop('Gene', axis=1, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "58fdb2ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>pam coordinate (hg19)</th>\n",
       "      <th>length (including pam)</th>\n",
       "      <th>pam coordinate (hg19) - 1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1_c_118438373.25</td>\n",
       "      <td>-0.120761</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGAAAACCTACCAGAGGTAAACAGT...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>118438373</td>\n",
       "      <td>25</td>\n",
       "      <td>118438372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1_c_118438478.24</td>\n",
       "      <td>0.012341</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGATCTGTTTTGCTCGtaatagGTT...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>118438478</td>\n",
       "      <td>24</td>\n",
       "      <td>118438477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1_c_118438478.26</td>\n",
       "      <td>0.029811</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGTGATCTGTTTTGCTCGtaatagG...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>118438478</td>\n",
       "      <td>26</td>\n",
       "      <td>118438477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1_c_118438508.26</td>\n",
       "      <td>0.022175</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgtaatgtgcagtacttactatgtG...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>118438508</td>\n",
       "      <td>26</td>\n",
       "      <td>118438507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1_c_118438682.27</td>\n",
       "      <td>-0.031622</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgcgctgaaaatagcagccaaacac...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>118438682</td>\n",
       "      <td>27</td>\n",
       "      <td>118438681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53668</th>\n",
       "      <td>YIPF5_w_143554761.28</td>\n",
       "      <td>-0.001537</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgggcattacatggcacctacctca...</td>\n",
       "      <td>YIPF5</td>\n",
       "      <td>143554761</td>\n",
       "      <td>28</td>\n",
       "      <td>143554760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53669</th>\n",
       "      <td>YIPF5_w_143554879.28</td>\n",
       "      <td>0.024536</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgcctactgctttagcataaggact...</td>\n",
       "      <td>YIPF5</td>\n",
       "      <td>143554879</td>\n",
       "      <td>28</td>\n",
       "      <td>143554878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53670</th>\n",
       "      <td>YIPF5_w_143554997.27</td>\n",
       "      <td>-0.050816</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgtctagttgcaattttaatttata...</td>\n",
       "      <td>YIPF5</td>\n",
       "      <td>143554997</td>\n",
       "      <td>27</td>\n",
       "      <td>143554996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53671</th>\n",
       "      <td>YIPF5_w_143555197.28</td>\n",
       "      <td>-0.062784</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgtacagaatgtaaatccatgcagt...</td>\n",
       "      <td>YIPF5</td>\n",
       "      <td>143555197</td>\n",
       "      <td>28</td>\n",
       "      <td>143555196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53672</th>\n",
       "      <td>YIPF5_w_143555235.28</td>\n",
       "      <td>-0.004076</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGgtttaaataaattactgaagctcc...</td>\n",
       "      <td>YIPF5</td>\n",
       "      <td>143555235</td>\n",
       "      <td>28</td>\n",
       "      <td>143555234</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52926 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   guide id dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "0      ARCN1_c_118438373.25                        -0.120761   \n",
       "1      ARCN1_c_118438478.24                         0.012341   \n",
       "2      ARCN1_c_118438478.26                         0.029811   \n",
       "3      ARCN1_c_118438508.26                         0.022175   \n",
       "4      ARCN1_c_118438682.27                        -0.031622   \n",
       "...                     ...                              ...   \n",
       "53668  YIPF5_w_143554761.28                        -0.001537   \n",
       "53669  YIPF5_w_143554879.28                         0.024536   \n",
       "53670  YIPF5_w_143554997.27                        -0.050816   \n",
       "53671  YIPF5_w_143555197.28                        -0.062784   \n",
       "53672  YIPF5_w_143555235.28                        -0.004076   \n",
       "\n",
       "                                                   oligo gene symbol  \\\n",
       "0      cccttggagaaCCAcctTGTTGGAAAACCTACCAGAGGTAAACAGT...       ARCN1   \n",
       "1      cccttggagaaCCAcctTGTTGGATCTGTTTTGCTCGtaatagGTT...       ARCN1   \n",
       "2      cccttggagaaCCAcctTGTTGGTGATCTGTTTTGCTCGtaatagG...       ARCN1   \n",
       "3      cccttggagaaCCAcctTGTTGgtaatgtgcagtacttactatgtG...       ARCN1   \n",
       "4      cccttggagaaCCAcctTGTTGgcgctgaaaatagcagccaaacac...       ARCN1   \n",
       "...                                                  ...         ...   \n",
       "53668  cccttggagaaCCAcctTGTTGgggcattacatggcacctacctca...       YIPF5   \n",
       "53669  cccttggagaaCCAcctTGTTGgcctactgctttagcataaggact...       YIPF5   \n",
       "53670  cccttggagaaCCAcctTGTTGgtctagttgcaattttaatttata...       YIPF5   \n",
       "53671  cccttggagaaCCAcctTGTTGgtacagaatgtaaatccatgcagt...       YIPF5   \n",
       "53672  cccttggagaaCCAcctTGTTGgtttaaataaattactgaagctcc...       YIPF5   \n",
       "\n",
       "       pam coordinate (hg19)  length (including pam)  \\\n",
       "0                  118438373                      25   \n",
       "1                  118438478                      24   \n",
       "2                  118438478                      26   \n",
       "3                  118438508                      26   \n",
       "4                  118438682                      27   \n",
       "...                      ...                     ...   \n",
       "53668              143554761                      28   \n",
       "53669              143554879                      28   \n",
       "53670              143554997                      27   \n",
       "53671              143555197                      28   \n",
       "53672              143555235                      28   \n",
       "\n",
       "       pam coordinate (hg19) - 1  \n",
       "0                      118438372  \n",
       "1                      118438477  \n",
       "2                      118438477  \n",
       "3                      118438507  \n",
       "4                      118438681  \n",
       "...                          ...  \n",
       "53668                  143554760  \n",
       "53669                  143554878  \n",
       "53670                  143554996  \n",
       "53671                  143555196  \n",
       "53672                  143555234  \n",
       "\n",
       "[52926 rows x 7 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tiling_crispri[\"pam coordinate (hg19)\"] = tiling_crispri[\"pam coordinate (hg19)\"].astype(np.int64)\n",
    "tiling_crispri[\"length (including pam)\"] = tiling_crispri[\"length (including pam)\"].astype(np.int64)\n",
    "\n",
    "# The PAM site from the KO data seems to be off by 1 from the CRISPRi data\n",
    "tiling_crispri[\"pam coordinate (hg19) - 1\"] = tiling_crispri[\"pam coordinate (hg19)\"] - 1\n",
    "tiling_crispri\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "12f6c3b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_tiling = pd.merge(tiling_crispri, new_tiling, \n",
    "         left_on = ['gene symbol', 'length (including pam)', 'pam coordinate (hg19) - 1'],\n",
    "         right_on = ['gene symbol', 'sgRNA length (including PAM)','PAM genomic coordinate [hg19]'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "258e5e22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "cccttggagaaCCAcctTGTTGGAAAACCTACCAGAGGTAAACAGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtcCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGATATCATATTTTAGGTATTTACTGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGgtggcaTAATGAAATTTTTCATcaGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGggactgttctgtatattgtgtGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtcaCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGgggactgttctgtatattgtgtGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtcCtcgagcgtcgaggatccattagg    1\n",
       "                                                                                                                                                                        ..\n",
       "cccttggagaaCCAcctTGTTGGTAAACCCCCATGATTTGACTCCACGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTaCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGCTGCCACTAGGCTGTGCGAGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagtcagCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGATCCGCTGCCACTAGGCTGTGCGAGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTaCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGGCTGTGCGAAGGCATTTACCTCCCGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTagCtcgagcgtcgaggatccattagg    1\n",
       "cccttggagaaCCAcctTGTTGgtttaaataaattactgaagctccaGTTTAAGAGCTATGCTGGAAACAGCATAGCAAGTTTAAATAAGGCTAGTCCGTTATCAACTTGAAAAAGTGGCACCGAGTCGGTGCTTTTTTTaCtcgagcgtcgaggatccattagg    1\n",
       "Name: oligo, Length: 51420, dtype: int64"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check unique oligo; all oligos are unique\n",
    "merged_tiling['oligo'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "6efbf92e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GAAAACCTACCAGAGGTAAACA       1\n",
       "GATATCATATTTTAGGTATTTACT     1\n",
       "gtggcaTAATGAAATTTTTCATca     1\n",
       "ggactgttctgtatattgtgt        1\n",
       "gggactgttctgtatattgtgt       1\n",
       "                            ..\n",
       "GTAAACCCCCATGATTTGACTCCAC    1\n",
       "GCTGCCACTAGGCTGTGCGA         1\n",
       "GATCCGCTGCCACTAGGCTGTGCGA    1\n",
       "GCTGTGCGAAGGCATTTACCTCCC     1\n",
       "gtttaaataaattactgaagctcca    1\n",
       "Name: sgRNA sequence, Length: 51420, dtype: int64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check unique sgRNA sequence; all sgRNA sequence are unique\n",
    "merged_tiling['sgRNA sequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "2ea709cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51420"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# number of overlap between merged ko and crispri\n",
    "sum(merged_tiling.apply(lambda row: row['sgRNA sequence'] in row['oligo'],\n",
    "                     axis=1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0104fbcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51420"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(merged_tiling)\n",
    "# looks like all the rows have ko and crispri overlapped yayy!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "1ca62765",
   "metadata": {},
   "outputs": [],
   "source": [
    "#import dany's annotation of context seq for CRSIPRko\n",
    "dany_context_seq = pd.read_csv('/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2023/cleaned_crispri_datasets/temp_data_clean/Gilbert 2014/Dany-horlbeck-tiling-with-rs3.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "d7ec6744",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>hg38_coord</th>\n",
       "      <th>context seq</th>\n",
       "      <th>Position relative to FANTOM5 TSS</th>\n",
       "      <th>absolute z-score</th>\n",
       "      <th>rs3 score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567638.0</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGTAGGTTT</td>\n",
       "      <td>-4793.0</td>\n",
       "      <td>0.409907</td>\n",
       "      <td>-0.067121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567850.0</td>\n",
       "      <td>ATTGGCAGTAACATCTATCTCATAAGGTTG</td>\n",
       "      <td>-4581.0</td>\n",
       "      <td>0.694579</td>\n",
       "      <td>-0.554137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567997.0</td>\n",
       "      <td>TCCCATAGTGGCAGTATCTTATAATGGTTA</td>\n",
       "      <td>-4434.0</td>\n",
       "      <td>0.566546</td>\n",
       "      <td>-0.090399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568013.0</td>\n",
       "      <td>CATGCTTCTTGAGGAGTCCCATAGTGGCAG</td>\n",
       "      <td>-4418.0</td>\n",
       "      <td>0.997859</td>\n",
       "      <td>0.220064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568026.0</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTGAGGAGT</td>\n",
       "      <td>-4405.0</td>\n",
       "      <td>0.386895</td>\n",
       "      <td>-0.563347</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene symbol chromosome strand targeted   hg38_coord  \\\n",
       "0       ARCN1      chr11               +  118567638.0   \n",
       "1       ARCN1      chr11               +  118567850.0   \n",
       "2       ARCN1      chr11               +  118567997.0   \n",
       "3       ARCN1      chr11               +  118568013.0   \n",
       "4       ARCN1      chr11               +  118568026.0   \n",
       "\n",
       "                      context seq  Position relative to FANTOM5 TSS  \\\n",
       "0  GCACTTTCCTTGTTTACCTCTGGTAGGTTT                           -4793.0   \n",
       "1  ATTGGCAGTAACATCTATCTCATAAGGTTG                           -4581.0   \n",
       "2  TCCCATAGTGGCAGTATCTTATAATGGTTA                           -4434.0   \n",
       "3  CATGCTTCTTGAGGAGTCCCATAGTGGCAG                           -4418.0   \n",
       "4  GCAACCCTGTAAGCATGCTTCTTGAGGAGT                           -4405.0   \n",
       "\n",
       "   absolute z-score  rs3 score  \n",
       "0          0.409907  -0.067121  \n",
       "1          0.694579  -0.554137  \n",
       "2          0.566546  -0.090399  \n",
       "3          0.997859   0.220064  \n",
       "4          0.386895  -0.563347  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dany_context_seq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "a18c1ab4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29339"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dany_context_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "23d7d8ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51420"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(merged_tiling)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "162a558c",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleaned_df = pd.merge(merged_tiling[['guide id', 'dCas9-KRAB_rho_ave_LG2_LG_MH2_MH', 'oligo', 'gene symbol', 'chromosome',\n",
    "       'strand targeted', 'sgRNA sequence','Target site MNase signal',\n",
    "       'Doench et al. Nat. Biotech 2014 sgRNA sequence score',\n",
    "       'is targeting CDS?','PAM genomic coordinate [hg38]']],\n",
    "         dany_context_seq.rename(columns={'hg38_coord':'PAM genomic coordinate [hg38]'}),\n",
    "         on = ['gene symbol', 'chromosome', 'strand targeted','PAM genomic coordinate [hg38]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "9ca3e8f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GAAAACCTACCAGAGGTAAACA       1\n",
       "GGACAGCCAGCAGTTCCTCCTG       1\n",
       "gcagacatctgtGCATGGTTT        1\n",
       "gactcagcagacatctgtGCA        1\n",
       "ggactcagcagacatctgtGCA       1\n",
       "                            ..\n",
       "gtgctgcTGGCACCTAGTAAAGGC     1\n",
       "ggtgctgcTGGCACCTAGTAAAGGC    1\n",
       "ggtgctgcTGGCACCTAGTAA        1\n",
       "gcggtgctgcTGGCACCTAGTAA      1\n",
       "gtttaaataaattactgaagctcca    1\n",
       "Name: sgRNA sequence, Length: 50728, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_df['sgRNA sequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "88fe5962",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CGGGGGCGGGCGCGGTCCTGCGCGGGGTTT    6\n",
       "GGGTGGGGTGCAGTCATGTGAGTCGGGCCC    6\n",
       "GAGGGGGCCGGGGTACGAGCGTGTCGGCTC    6\n",
       "GGAGGGGTCCTTTCAACCGGGCTCAGGTTG    6\n",
       "TGGGGGCGCCGCACAGTCCTGACCCGGGCC    6\n",
       "                                 ..\n",
       "AAGAAATTCCAGGCCCTGACGGCAAGGGAG    1\n",
       "CCAAGGAGACCCCAGGGCAACCTGAGGAAA    1\n",
       "CCTCATGGGCACAGCTTGGCTCTCAGGACT    1\n",
       "GCTCCGGGGACTCCTCAGTGCCAAGGGCCT    1\n",
       "TTTAAATAAATTACTGAAGCTCCATGGCAA    1\n",
       "Name: context seq, Length: 28947, dtype: int64"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_df['context seq'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "d8d59543",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['guide id', 'dCas9-KRAB_rho_ave_LG2_LG_MH2_MH', 'oligo', 'gene symbol',\n",
       "       'chromosome', 'strand targeted', 'sgRNA sequence',\n",
       "       'Target site MNase signal',\n",
       "       'Doench et al. Nat. Biotech 2014 sgRNA sequence score',\n",
       "       'is targeting CDS?', 'PAM genomic coordinate [hg38]', 'context seq',\n",
       "       'Position relative to FANTOM5 TSS', 'absolute z-score', 'rs3 score'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "5b89e11f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>guide id</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>oligo</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>Target site MNase signal</th>\n",
       "      <th>Doench et al. Nat. Biotech 2014 sgRNA sequence score</th>\n",
       "      <th>is targeting CDS?</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "      <th>context seq</th>\n",
       "      <th>Position relative to FANTOM5 TSS</th>\n",
       "      <th>absolute z-score</th>\n",
       "      <th>rs3 score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1_c_118438373.25</td>\n",
       "      <td>-0.120761</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGAAAACCTACCAGAGGTAAACAGT...</td>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>-</td>\n",
       "      <td>GAAAACCTACCAGAGGTAAACA</td>\n",
       "      <td>0.164</td>\n",
       "      <td>0.127736</td>\n",
       "      <td>False</td>\n",
       "      <td>118567657</td>\n",
       "      <td>CTGAAAACCTACCAGAGGTAAACAAGGAAA</td>\n",
       "      <td>-4774.0</td>\n",
       "      <td>0.817426</td>\n",
       "      <td>0.587021</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               guide id dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "0  ARCN1_c_118438373.25                        -0.120761   \n",
       "\n",
       "                                               oligo gene symbol chromosome  \\\n",
       "0  cccttggagaaCCAcctTGTTGGAAAACCTACCAGAGGTAAACAGT...       ARCN1      chr11   \n",
       "\n",
       "  strand targeted          sgRNA sequence  Target site MNase signal  \\\n",
       "0               -  GAAAACCTACCAGAGGTAAACA                     0.164   \n",
       "\n",
       "   Doench et al. Nat. Biotech 2014 sgRNA sequence score  is targeting CDS?  \\\n",
       "0                                           0.127736                 False   \n",
       "\n",
       "   PAM genomic coordinate [hg38]                     context seq  \\\n",
       "0                      118567657  CTGAAAACCTACCAGAGGTAAACAAGGAAA   \n",
       "\n",
       "   Position relative to FANTOM5 TSS  absolute z-score  rs3 score  \n",
       "0                           -4774.0          0.817426   0.587021  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_df.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "18984575",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleaned_df = cleaned_df[['guide id','gene symbol','chromosome','strand targeted','PAM genomic coordinate [hg38]', 'sgRNA sequence', \n",
    "            'context seq','dCas9-KRAB_rho_ave_LG2_LG_MH2_MH','Position relative to FANTOM5 TSS',\n",
    "            'Target site MNase signal','is targeting CDS?','Doench et al. Nat. Biotech 2014 sgRNA sequence score',\n",
    "        'rs3 score']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "a070e80d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CGGGGGCGGGCGCGGTCCTGCGCGGGGTTT    6\n",
       "GGGTGGGGTGCAGTCATGTGAGTCGGGCCC    6\n",
       "GAGGGGGCCGGGGTACGAGCGTGTCGGCTC    6\n",
       "GGAGGGGTCCTTTCAACCGGGCTCAGGTTG    6\n",
       "TGGGGGCGCCGCACAGTCCTGACCCGGGCC    6\n",
       "                                 ..\n",
       "AAGAAATTCCAGGCCCTGACGGCAAGGGAG    1\n",
       "CCAAGGAGACCCCAGGGCAACCTGAGGAAA    1\n",
       "CCTCATGGGCACAGCTTGGCTCTCAGGACT    1\n",
       "GCTCCGGGGACTCCTCAGTGCCAAGGGCCT    1\n",
       "TTTAAATAAATTACTGAAGCTCCATGGCAA    1\n",
       "Name: context seq, Length: 28947, dtype: int64"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_df['context seq'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "e8b56845",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupliated_df_same_gene = cleaned_df[cleaned_df[['context seq','gene symbol']].duplicated(keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "3cd95d53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cleaned_df[cleaned_df[['context seq']].duplicated(keep = False)]) == len(dupliated_df_same_gene)\n",
    "# all the duplicated seq are within gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "5f1b953a",
   "metadata": {},
   "outputs": [],
   "source": [
    "for seq in dupliated_df_same_gene['context seq'].unique():\n",
    "    seq_df = cleaned_df[cleaned_df['context seq'] == seq]\n",
    "    if len(set(seq_df['gene symbol'])) == 1:\n",
    "        cleaned_df.loc[seq_df.index,'dCas9-KRAB_rho_ave_LG2_LG_MH2_MH'] = seq_df['dCas9-KRAB_rho_ave_LG2_LG_MH2_MH'].mean()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "f8fbc6e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleaned_df_no_dup = cleaned_df.drop_duplicates(['gene symbol','context seq'], keep = 'first').copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "e66bfe0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28947"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cleaned_df_no_dup)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "1eed99a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_path = '/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2023/cleaned_crispri_datasets/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "115cb2e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleaned_df_no_dup.to_csv(out_path + 'gilbert_2014_cleaned_v2.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "837ce489",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleaned_df_no_dup/len(dcas_krab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3ce0f12",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(cleaned_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96b04136",
   "metadata": {},
   "outputs": [],
   "source": [
    "sum(cleaned_df['context seq'].str[25:27] != 'GG')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a941a53d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
