{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "40a33804",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import pyBigWig\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e15d6a0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataSet\n",
       "InHouse    344011\n",
       "Nunez       72499\n",
       "Gilbert     18595\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ATAC seq is run on this file [old cleaned dataset ]\n",
    "\n",
    "inpath = '../../Data/CleanedExternalData/archive/'\n",
    "DatasetCombine = pd.read_csv(inpath + 'DatasetCombine.csv')\n",
    "DatasetCombine['DataSet'] = DatasetCombine['DataSet'].str.replace('Nunze', 'Nunez')\n",
    "DatasetCombine.DataSet.value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "dcb74dba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataSet\n",
       "InHouse    344011\n",
       "Nunez       68575\n",
       "Gilbert     18509\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# new cleaned file\n",
    "inpath = '../../Data/CleanedExternalData/'\n",
    "\n",
    "NewDatasetCombine = pd.read_csv(inpath + 'DatasetCombineCleaned_v3.csv', low_memory=False)\n",
    "NewDatasetCombine.DataSet.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5184e359",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check if the new cleaned file is contained in the old file that ATACseq is run on\n",
    "checkNewDatasetCombine = pd.merge(DatasetCombine[['Target Gene Symbol', 'chromosome',\n",
    "       'sgRNA Context Sequence','condition', 'Domain', 'DataSet']],\n",
    "         NewDatasetCombine, on = ['Target Gene Symbol',\n",
    "       'sgRNA Context Sequence','condition', 'Domain', 'DataSet','chromosome'])\n",
    "len(checkNewDatasetCombine) == len(NewDatasetCombine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "16429b6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataSet\n",
       "InHouse    344011\n",
       "Nunez       68575\n",
       "Gilbert     18509\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkNewDatasetCombine.DataSet.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "643804c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "checkNewDatasetCombine['orginal phenotype'] = checkNewDatasetCombine['Phenotype scores-ave_Rep1_Rep2'].combine_first(checkNewDatasetCombine['z_scored_avg_lfc']).combine_first(checkNewDatasetCombine['rhoSignChangedAndAvged'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8b4bbb58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "condition  Domain  DataSet\n",
       "HCT116     Kox1    InHouse    86047\n",
       "A549       Zim3    InHouse    86026\n",
       "           Kox1    InHouse    85993\n",
       "HCT116     Zim3    InHouse    85945\n",
       "K562       Kox1    Nunez      68575\n",
       "                   Gilbert    18509\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkNewDatasetCombine[['condition', 'Domain', 'DataSet']].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "0ec79808",
   "metadata": {},
   "outputs": [],
   "source": [
    "# the code for those processed data can be found in the previousATAC folder\n",
    "atac_out = '../../Data/ATACseq/'\n",
    "chrtest_k562 = pd.read_csv(atac_out + 'ATAC_K562.csv')\n",
    "chrtest_k562['ATACseq'] = 'K562'\n",
    "\n",
    "chrtest_hct116 = pd.read_csv(atac_out + 'ATAC_HCT116.csv')\n",
    "chrtest_hct116['ATACseq'] = 'HCT116'\n",
    "\n",
    "chrtest_a549 = pd.read_csv(atac_out + 'ATAC_A549.csv')\n",
    "chrtest_a549['ATACseq'] = 'A549'\n",
    "\n",
    "chrtest_MCF7 = pd.read_csv(atac_out + 'ATAC_MCF7.csv' )\n",
    "chrtest_MCF7['ATACseq'] = 'MCF7'\n",
    "\n",
    "chrtest_Panc1 = pd.read_csv(atac_out + 'ATAC_Panc1.csv' )\n",
    "chrtest_Panc1['ATACseq'] = 'Panc1'\n",
    "\n",
    "chrtest_HepG2 = pd.read_csv(atac_out + 'chrtest_HepG2.csv' )\n",
    "chrtest_HepG2['ATACseq'] = 'HepG2'\n",
    "\n",
    "chrtest_GM12878 = pd.read_csv(atac_out + 'ATAC_GM12878.csv' )\n",
    "chrtest_GM12878['ATACseq'] = 'GM12878'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "1576287c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>scaled_phenotype</th>\n",
       "      <th>orginal phenotype</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>overlap with peak</th>\n",
       "      <th>ATACseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BPY2</td>\n",
       "      <td>chrY</td>\n",
       "      <td>AAAAAATAAGGCAGGTGCCA</td>\n",
       "      <td>TACAAAAAAATAAGGCAGGTGCCATGGCAG</td>\n",
       "      <td>22983942.0</td>\n",
       "      <td>+</td>\n",
       "      <td>22984264.0</td>\n",
       "      <td>-322.0</td>\n",
       "      <td>0.376492</td>\n",
       "      <td>-0.125426</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BPY2</td>\n",
       "      <td>chrY</td>\n",
       "      <td>AAAAAATAAGGCAGGTGCCA</td>\n",
       "      <td>TACAAAAAAATAAGGCAGGTGCCATGGCAG</td>\n",
       "      <td>22983942.0</td>\n",
       "      <td>+</td>\n",
       "      <td>22984264.0</td>\n",
       "      <td>-322.0</td>\n",
       "      <td>0.336803</td>\n",
       "      <td>-0.113912</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BPY2</td>\n",
       "      <td>chrY</td>\n",
       "      <td>AAAAAATAAGGCAGGTGCCA</td>\n",
       "      <td>TACAAAAAAATAAGGCAGGTGCCATGGCAG</td>\n",
       "      <td>22983942.0</td>\n",
       "      <td>+</td>\n",
       "      <td>22984264.0</td>\n",
       "      <td>-322.0</td>\n",
       "      <td>0.096359</td>\n",
       "      <td>0.539853</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BPY2</td>\n",
       "      <td>chrY</td>\n",
       "      <td>AAAAAATAAGGCAGGTGCCA</td>\n",
       "      <td>TACAAAAAAATAAGGCAGGTGCCATGGCAG</td>\n",
       "      <td>22983942.0</td>\n",
       "      <td>+</td>\n",
       "      <td>22984264.0</td>\n",
       "      <td>-322.0</td>\n",
       "      <td>0.302406</td>\n",
       "      <td>0.070993</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DAZ1</td>\n",
       "      <td>chrY</td>\n",
       "      <td>AAAAATTCTAAAATTCTCAA</td>\n",
       "      <td>GTTTAAAAATTCTAAAATTCTCAATGGTAA</td>\n",
       "      <td>23198021.0</td>\n",
       "      <td>-</td>\n",
       "      <td>23199008.0</td>\n",
       "      <td>988.0</td>\n",
       "      <td>0.233765</td>\n",
       "      <td>0.371842</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435100</th>\n",
       "      <td>LUZP4</td>\n",
       "      <td>chrX</td>\n",
       "      <td>TCCTTTCAGGTGGGGAACTG</td>\n",
       "      <td>CCTCTCCTTTCAGGTGGGGAACTGAGGGAG</td>\n",
       "      <td>115290090.0</td>\n",
       "      <td>+</td>\n",
       "      <td>115289716.0</td>\n",
       "      <td>374.0</td>\n",
       "      <td>0.109000</td>\n",
       "      <td>0.614498</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>GM12878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435101</th>\n",
       "      <td>LUZP4</td>\n",
       "      <td>chrX</td>\n",
       "      <td>TCCTTTCAGGTGGGGAACTG</td>\n",
       "      <td>CCTCTCCTTTCAGGTGGGGAACTGAGGGAG</td>\n",
       "      <td>115290090.0</td>\n",
       "      <td>+</td>\n",
       "      <td>115289716.0</td>\n",
       "      <td>374.0</td>\n",
       "      <td>0.157225</td>\n",
       "      <td>0.312526</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>GM12878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435102</th>\n",
       "      <td>LUZP4</td>\n",
       "      <td>chrX</td>\n",
       "      <td>TCCTTTCAGGTGGGGAACTG</td>\n",
       "      <td>CCTCTCCTTTCAGGTGGGGAACTGAGGGAG</td>\n",
       "      <td>115290090.0</td>\n",
       "      <td>+</td>\n",
       "      <td>115289716.0</td>\n",
       "      <td>374.0</td>\n",
       "      <td>0.147145</td>\n",
       "      <td>0.473902</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>GM12878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435103</th>\n",
       "      <td>SAGE1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>ACTGACCAAGAAAAATAGAA</td>\n",
       "      <td>CTTGACTGACCAAGAAAAATAGAAAGGACT</td>\n",
       "      <td>135894542.0</td>\n",
       "      <td>+</td>\n",
       "      <td>135893717.0</td>\n",
       "      <td>825.0</td>\n",
       "      <td>0.220117</td>\n",
       "      <td>0.359468</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>GM12878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435104</th>\n",
       "      <td>SAGE1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>ACTGACCAAGAAAAATAGAA</td>\n",
       "      <td>CTTGACTGACCAAGAAAAATAGAAAGGACT</td>\n",
       "      <td>135894542.0</td>\n",
       "      <td>+</td>\n",
       "      <td>135893717.0</td>\n",
       "      <td>825.0</td>\n",
       "      <td>0.293135</td>\n",
       "      <td>-0.100868</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "      <td>GM12878</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3045735 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome        sgRNA Sequence  \\\n",
       "0                    BPY2       chrY  AAAAAATAAGGCAGGTGCCA   \n",
       "1                    BPY2       chrY  AAAAAATAAGGCAGGTGCCA   \n",
       "2                    BPY2       chrY  AAAAAATAAGGCAGGTGCCA   \n",
       "3                    BPY2       chrY  AAAAAATAAGGCAGGTGCCA   \n",
       "4                    DAZ1       chrY  AAAAATTCTAAAATTCTCAA   \n",
       "...                   ...        ...                   ...   \n",
       "435100              LUZP4       chrX  TCCTTTCAGGTGGGGAACTG   \n",
       "435101              LUZP4       chrX  TCCTTTCAGGTGGGGAACTG   \n",
       "435102              LUZP4       chrX  TCCTTTCAGGTGGGGAACTG   \n",
       "435103              SAGE1       chrX  ACTGACCAAGAAAAATAGAA   \n",
       "435104              SAGE1       chrX  ACTGACCAAGAAAAATAGAA   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "0       TACAAAAAAATAAGGCAGGTGCCATGGCAG            22983942.0                +   \n",
       "1       TACAAAAAAATAAGGCAGGTGCCATGGCAG            22983942.0                +   \n",
       "2       TACAAAAAAATAAGGCAGGTGCCATGGCAG            22983942.0                +   \n",
       "3       TACAAAAAAATAAGGCAGGTGCCATGGCAG            22983942.0                +   \n",
       "4       GTTTAAAAATTCTAAAATTCTCAATGGTAA            23198021.0                -   \n",
       "...                                ...                   ...              ...   \n",
       "435100  CCTCTCCTTTCAGGTGGGGAACTGAGGGAG           115290090.0                +   \n",
       "435101  CCTCTCCTTTCAGGTGGGGAACTGAGGGAG           115290090.0                +   \n",
       "435102  CCTCTCCTTTCAGGTGGGGAACTGAGGGAG           115290090.0                +   \n",
       "435103  CTTGACTGACCAAGAAAAATAGAAAGGACT           135894542.0                +   \n",
       "435104  CTTGACTGACCAAGAAAAATAGAAAGGACT           135894542.0                +   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  scaled_phenotype  \\\n",
       "0         22984264.0                       -322.0          0.376492   \n",
       "1         22984264.0                       -322.0          0.336803   \n",
       "2         22984264.0                       -322.0          0.096359   \n",
       "3         22984264.0                       -322.0          0.302406   \n",
       "4         23199008.0                        988.0          0.233765   \n",
       "...              ...                          ...               ...   \n",
       "435100   115289716.0                        374.0          0.109000   \n",
       "435101   115289716.0                        374.0          0.157225   \n",
       "435102   115289716.0                        374.0          0.147145   \n",
       "435103   135893717.0                        825.0          0.220117   \n",
       "435104   135893717.0                        825.0          0.293135   \n",
       "\n",
       "        orginal phenotype condition Domain  DataSet  overlap with peak  \\\n",
       "0               -0.125426      A549   Kox1  InHouse                  0   \n",
       "1               -0.113912      A549   Zim3  InHouse                  0   \n",
       "2                0.539853    HCT116   Kox1  InHouse                  0   \n",
       "3                0.070993    HCT116   Zim3  InHouse                  0   \n",
       "4                0.371842      A549   Kox1  InHouse                  0   \n",
       "...                   ...       ...    ...      ...                ...   \n",
       "435100           0.614498      A549   Zim3  InHouse                  0   \n",
       "435101           0.312526    HCT116   Kox1  InHouse                  0   \n",
       "435102           0.473902    HCT116   Zim3  InHouse                  0   \n",
       "435103           0.359468    HCT116   Kox1  InHouse                  0   \n",
       "435104          -0.100868    HCT116   Zim3  InHouse                  0   \n",
       "\n",
       "        ATACseq  \n",
       "0          K562  \n",
       "1          K562  \n",
       "2          K562  \n",
       "3          K562  \n",
       "4          K562  \n",
       "...         ...  \n",
       "435100  GM12878  \n",
       "435101  GM12878  \n",
       "435102  GM12878  \n",
       "435103  GM12878  \n",
       "435104  GM12878  \n",
       "\n",
       "[3045735 rows x 15 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrtest = pd.concat([chrtest_k562,chrtest_hct116,chrtest_a549,chrtest_MCF7,chrtest_Panc1,chrtest_HepG2,chrtest_GM12878])\n",
    "chrtest['DataSet'] = chrtest['DataSet'].str.replace('Nunze', 'Nunez')\n",
    "chrtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4ff6db78",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ATACseq\n",
       "K562       435105\n",
       "HCT116     435105\n",
       "A549       435105\n",
       "MCF7       435105\n",
       "Panc1      435105\n",
       "HepG2      435105\n",
       "GM12878    435105\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrtest.ATACseq.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "66622599",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ATACseq  condition  Domain  DataSet\n",
       "A549     HCT116     Kox1    InHouse    86047\n",
       "K562     HCT116     Kox1    InHouse    86047\n",
       "Panc1    HCT116     Kox1    InHouse    86047\n",
       "GM12878  HCT116     Kox1    InHouse    86047\n",
       "HepG2    HCT116     Kox1    InHouse    86047\n",
       "MCF7     HCT116     Kox1    InHouse    86047\n",
       "HCT116   HCT116     Kox1    InHouse    86047\n",
       "K562     A549       Zim3    InHouse    86026\n",
       "Panc1    A549       Zim3    InHouse    86026\n",
       "A549     A549       Zim3    InHouse    86026\n",
       "GM12878  A549       Zim3    InHouse    86026\n",
       "HepG2    A549       Zim3    InHouse    86026\n",
       "HCT116   A549       Zim3    InHouse    86026\n",
       "MCF7     A549       Zim3    InHouse    86026\n",
       "                    Kox1    InHouse    85993\n",
       "HepG2    A549       Kox1    InHouse    85993\n",
       "A549     A549       Kox1    InHouse    85993\n",
       "K562     A549       Kox1    InHouse    85993\n",
       "Panc1    A549       Kox1    InHouse    85993\n",
       "GM12878  A549       Kox1    InHouse    85993\n",
       "HCT116   A549       Kox1    InHouse    85993\n",
       "K562     HCT116     Zim3    InHouse    85945\n",
       "MCF7     HCT116     Zim3    InHouse    85945\n",
       "Panc1    HCT116     Zim3    InHouse    85945\n",
       "HepG2    HCT116     Zim3    InHouse    85945\n",
       "A549     HCT116     Zim3    InHouse    85945\n",
       "GM12878  HCT116     Zim3    InHouse    85945\n",
       "HCT116   HCT116     Zim3    InHouse    85945\n",
       "HepG2    K562       Kox1    Nunez      72499\n",
       "GM12878  K562       Kox1    Nunez      72499\n",
       "A549     K562       Kox1    Nunez      72499\n",
       "MCF7     K562       Kox1    Nunez      72499\n",
       "Panc1    K562       Kox1    Nunez      72499\n",
       "K562     K562       Kox1    Nunez      72499\n",
       "HCT116   K562       Kox1    Nunez      72499\n",
       "HepG2    K562       Kox1    Gilbert    18595\n",
       "MCF7     K562       Kox1    Gilbert    18595\n",
       "HCT116   K562       Kox1    Gilbert    18595\n",
       "K562     K562       Kox1    Gilbert    18595\n",
       "A549     K562       Kox1    Gilbert    18595\n",
       "Panc1    K562       Kox1    Gilbert    18595\n",
       "GM12878  K562       Kox1    Gilbert    18595\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrtest[['ATACseq','condition', 'Domain', 'DataSet']].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0f6893ea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "      <th>orginal phenotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>GCGAGCACATAGAGGGATCT</td>\n",
       "      <td>100198288.0</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816.0</td>\n",
       "      <td>-2528.0</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>-0.889828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.018881</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Target Gene Symbol chromosome          sgRNA Context Sequence condition  \\\n",
       "0              ACTR6      chr12  TGCTTCGAGCACATAGAGGGATCTGGGCCC      K562   \n",
       "\n",
       "  Domain DataSet        sgRNA Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "0   Kox1   Nunez  GCGAGCACATAGAGGGATCT           100198288.0                +   \n",
       "\n",
       "   TSS Position  sgRNA 'Cut' Site TSS Offset  Phenotype scores-ave_Rep1_Rep2  \\\n",
       "0   100200816.0                      -2528.0                       -0.018881   \n",
       "\n",
       "   Avg_LFC-Tfinal  z-score relative to NC  rhoSignChangedAndAvged  \\\n",
       "0        0.035104               -0.889828                     NaN   \n",
       "\n",
       "   zscoreRelativeNC_signed  Avg_LFC_signed  z_scored_avg_lfc  \\\n",
       "0                      NaN             NaN               NaN   \n",
       "\n",
       "   orginal phenotype  \n",
       "0          -0.018881  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkNewDatasetCombine.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4d4ebf4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chr12', 'chr18', 'chr3', 'chr1', 'chr8', 'chr22', 'chr17', 'chr5',\n",
       "       'chr10', 'chr11', 'chr14', 'chr2', 'chr21', 'chr15', 'chr6',\n",
       "       'chr4', 'chr19', 'chr13', 'chr16', 'chr7', 'chr20', 'chr9',\n",
       "       'chr23', 'chr24'], dtype=object)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkNewDatasetCombine.chromosome.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7c919a57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chrY', 'chr11', 'chr17', 'chr10', 'chr18', 'chr1', 'chr9', 'chr8',\n",
       "       'chr19', 'chr16', 'chr14', 'chr15', 'chr3', 'chr21', 'chr6',\n",
       "       'chr5', 'chr13', 'chr2', 'chr22', 'chr4', 'chr12', 'chr7', 'chrX',\n",
       "       'chr20'], dtype=object)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrtest.chromosome.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a6159535",
   "metadata": {},
   "outputs": [],
   "source": [
    "chrtest['chromosome'] = chrtest['chromosome'].replace('chrY','chr24')\n",
    "chrtest['chromosome'] = chrtest['chromosome'].replace('chrX','chr23')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "76264eae",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_chrtest = pd.merge(checkNewDatasetCombine[['Target Gene Symbol', 'chromosome',\n",
    "       'sgRNA Context Sequence','condition', 'Domain', 'DataSet','orginal phenotype']], \n",
    "                    chrtest[['Target Gene Symbol', 'chromosome',\n",
    "       'sgRNA Context Sequence','condition', 'Domain', 'DataSet','overlap with peak','ATACseq']],\n",
    "                       on = ['Target Gene Symbol','chromosome','sgRNA Context Sequence','condition', 'Domain', 'DataSet'],\n",
    "                                               how = 'inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "d29f42c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count\n",
       "86047    7\n",
       "86026    7\n",
       "85993    7\n",
       "85945    7\n",
       "68575    7\n",
       "18509    7\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_chrtest[['ATACseq','condition', 'Domain', 'DataSet']].value_counts().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "5bab23e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>orginal phenotype</th>\n",
       "      <th>overlap with peak</th>\n",
       "      <th>ATACseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0</td>\n",
       "      <td>HCT116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0</td>\n",
       "      <td>MCF7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0</td>\n",
       "      <td>Panc1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3017660</th>\n",
       "      <td>KRT36</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ATGTTCCTTCCCTTCCTACAGCAAGGGGAG</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>-0.510239</td>\n",
       "      <td>0</td>\n",
       "      <td>A549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3017661</th>\n",
       "      <td>KRT36</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ATGTTCCTTCCCTTCCTACAGCAAGGGGAG</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>-0.510239</td>\n",
       "      <td>0</td>\n",
       "      <td>MCF7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3017662</th>\n",
       "      <td>KRT36</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ATGTTCCTTCCCTTCCTACAGCAAGGGGAG</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>-0.510239</td>\n",
       "      <td>0</td>\n",
       "      <td>Panc1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3017663</th>\n",
       "      <td>KRT36</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ATGTTCCTTCCCTTCCTACAGCAAGGGGAG</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>-0.510239</td>\n",
       "      <td>0</td>\n",
       "      <td>HepG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3017664</th>\n",
       "      <td>KRT36</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ATGTTCCTTCCCTTCCTACAGCAAGGGGAG</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>-0.510239</td>\n",
       "      <td>0</td>\n",
       "      <td>GM12878</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3017665 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Target Gene Symbol chromosome          sgRNA Context Sequence  \\\n",
       "0                    ACTR6      chr12  TGCTTCGAGCACATAGAGGGATCTGGGCCC   \n",
       "1                    ACTR6      chr12  TGCTTCGAGCACATAGAGGGATCTGGGCCC   \n",
       "2                    ACTR6      chr12  TGCTTCGAGCACATAGAGGGATCTGGGCCC   \n",
       "3                    ACTR6      chr12  TGCTTCGAGCACATAGAGGGATCTGGGCCC   \n",
       "4                    ACTR6      chr12  TGCTTCGAGCACATAGAGGGATCTGGGCCC   \n",
       "...                    ...        ...                             ...   \n",
       "3017660              KRT36      chr17  ATGTTCCTTCCCTTCCTACAGCAAGGGGAG   \n",
       "3017661              KRT36      chr17  ATGTTCCTTCCCTTCCTACAGCAAGGGGAG   \n",
       "3017662              KRT36      chr17  ATGTTCCTTCCCTTCCTACAGCAAGGGGAG   \n",
       "3017663              KRT36      chr17  ATGTTCCTTCCCTTCCTACAGCAAGGGGAG   \n",
       "3017664              KRT36      chr17  ATGTTCCTTCCCTTCCTACAGCAAGGGGAG   \n",
       "\n",
       "        condition Domain  DataSet  orginal phenotype  overlap with peak  \\\n",
       "0            K562   Kox1    Nunez          -0.018881                  0   \n",
       "1            K562   Kox1    Nunez          -0.018881                  0   \n",
       "2            K562   Kox1    Nunez          -0.018881                  0   \n",
       "3            K562   Kox1    Nunez          -0.018881                  0   \n",
       "4            K562   Kox1    Nunez          -0.018881                  0   \n",
       "...           ...    ...      ...                ...                ...   \n",
       "3017660    HCT116   Zim3  InHouse          -0.510239                  0   \n",
       "3017661    HCT116   Zim3  InHouse          -0.510239                  0   \n",
       "3017662    HCT116   Zim3  InHouse          -0.510239                  0   \n",
       "3017663    HCT116   Zim3  InHouse          -0.510239                  0   \n",
       "3017664    HCT116   Zim3  InHouse          -0.510239                  0   \n",
       "\n",
       "         ATACseq  \n",
       "0           K562  \n",
       "1         HCT116  \n",
       "2           A549  \n",
       "3           MCF7  \n",
       "4          Panc1  \n",
       "...          ...  \n",
       "3017660     A549  \n",
       "3017661     MCF7  \n",
       "3017662    Panc1  \n",
       "3017663    HepG2  \n",
       "3017664  GM12878  \n",
       "\n",
       "[3017665 rows x 9 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_chrtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "71e4f6d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# new_chrtest['orginal phenotype'] = new_chrtest['Phenotype scores-ave_Rep1_Rep2'].combine_first(new_chrtest['z_scored_avg_lfc']).combine_first(new_chrtest['rhoSignChangedAndAvged'])\n",
    "new_chrtest = new_chrtest[['Target Gene Symbol', 'chromosome', 'sgRNA Context Sequence',\n",
    "       'condition', 'Domain', 'DataSet', 'overlap with peak', \n",
    "       'orginal phenotype','ATACseq']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "520b9704",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    1584080\n",
       "1    1433585\n",
       "Name: overlap with peak, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_chrtest['overlap with peak'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9ba22517",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_path = '../../Data/ATACseq/'\n",
    "new_chrtest.to_csv(out_path+'ATACseqPeakOverlap.csv', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
