{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "40a33804",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pyBigWig\n",
    "# import actc_seq\n",
    "import tqdm\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "from scipy.stats import mannwhitneyu\n",
    "import matplotlib.pyplot as plt\n",
    "from tqdm import tqdm\n",
    "import os\n",
    "from collections import Counter\n",
    "from pylab import rcParams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e15d6a0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "InHouse    344011\n",
       "Nunez       72499\n",
       "Gilbert     18595\n",
       "Name: DataSet, dtype: int64"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ATAC seq is run on this file\n",
    "inpath = '/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2024/DataCleaning/'\n",
    "DatasetCombine = pd.read_csv(inpath + 'DatasetCombine.csv')\n",
    "DatasetCombine['DataSet'] = DatasetCombine['DataSet'].str.replace('Nunze', 'Nunez')\n",
    "DatasetCombine.DataSet.value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "dcb74dba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "InHouse    344011\n",
       "Nunez       68575\n",
       "Gilbert     18509\n",
       "Name: DataSet, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# new cleaned file\n",
    "inpath = '/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2024/Manuscript/CleanedData/'\n",
    "\n",
    "NewDatasetCombine = pd.read_csv(inpath + 'DatasetCombineCleaned_v3.csv', low_memory=False)\n",
    "NewDatasetCombine.DataSet.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "01497eee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "431095"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(NewDatasetCombine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5184e359",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check if the new cleaned file is contained in the old file that ATACseq is run on\n",
    "checkNewDatasetCombine = pd.merge(DatasetCombine[['Target Gene Symbol', 'chromosome',\n",
    "       'sgRNA Context Sequence','condition', 'Domain', 'DataSet']],\n",
    "         NewDatasetCombine, on = ['Target Gene Symbol',\n",
    "       'sgRNA Context Sequence','condition', 'Domain', 'DataSet'])\n",
    "len(checkNewDatasetCombine) == len(NewDatasetCombine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "16429b6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "InHouse    344011\n",
       "Nunez       68575\n",
       "Gilbert     18509\n",
       "Name: DataSet, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkNewDatasetCombine.DataSet.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "643804c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "checkNewDatasetCombine = checkNewDatasetCombine.drop(['chromosome_y', 'metrics'], axis=1)\n",
    "checkNewDatasetCombine = checkNewDatasetCombine.rename(columns = {'chromosome_x': 'chromosome'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0ec79808",
   "metadata": {},
   "outputs": [],
   "source": [
    "atac_out = '/Users/fzheng/Library/CloudStorage/GoogleDrive-fzheng@broadinstitute.org/Shared drives/GPP Cloud /R&D/People/Fengyi/rule_set/crispri/data/2024/ATAC/'\n",
    "\n",
    "chrtest_hct116 = pd.read_csv(atac_out + 'ATAC_HCT116.csv')\n",
    "chrtest_hct116['ATACseq'] = 'HCT116'\n",
    "\n",
    "chrtest_a549 = pd.read_csv(atac_out + 'ATAC_A549.csv')\n",
    "chrtest_a549['ATACseq'] = 'A549'\n",
    "\n",
    "chrtest_MCF7 = pd.read_csv(atac_out + 'ATAC_MCF7.csv' )\n",
    "chrtest_MCF7['ATACseq'] = 'MCF7'\n",
    "\n",
    "chrtest_Panc1 = pd.read_csv(atac_out + 'ATAC_Panc1.csv' )\n",
    "chrtest_Panc1['ATACseq'] = 'Panc1'\n",
    "\n",
    "chrtest_HepG2 = pd.read_csv(atac_out + 'chrtest_HepG2.csv' )\n",
    "chrtest_HepG2['ATACseq'] = 'HepG2'\n",
    "\n",
    "chrtest_GM12878 = pd.read_csv(atac_out + 'ATAC_GM12878.csv' )\n",
    "chrtest_GM12878['ATACseq'] = 'GM12878'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3c4e96eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>overlap with peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>gaggcaggacagtgacttaggcccaggagt</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>gaggccgaggcaggacagtgacttaggccc</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>CCCTGAAAGAATTGAGATCGggccaggtgt</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>TGCTGGTGCTTAACACAGGGGCAGTGGTCC</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>GATATATGCTGGTGCTTAACACAGGGGCAG</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435100</th>\n",
       "      <td>TOP2A</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ACACCGTAAATAATTTATAATAATAGGCAC</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435101</th>\n",
       "      <td>TOP2A</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ACACCGTAAATAATTTATAATAATAGGCAC</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435102</th>\n",
       "      <td>COPS3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435103</th>\n",
       "      <td>COPS3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435104</th>\n",
       "      <td>KRT36</td>\n",
       "      <td>chr17</td>\n",
       "      <td>ATGTTCCTTCCCTTCCTACAGCAAGGGGAG</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>435105 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome          sgRNA Context Sequence  \\\n",
       "0                   CDAN1      chr15  gaggcaggacagtgacttaggcccaggagt   \n",
       "1                   CDAN1      chr15  gaggccgaggcaggacagtgacttaggccc   \n",
       "2                   CDAN1      chr15  CCCTGAAAGAATTGAGATCGggccaggtgt   \n",
       "3                   CDAN1      chr15  TGCTGGTGCTTAACACAGGGGCAGTGGTCC   \n",
       "4                   CDAN1      chr15  GATATATGCTGGTGCTTAACACAGGGGCAG   \n",
       "...                   ...        ...                             ...   \n",
       "435100              TOP2A      chr17  ACACCGTAAATAATTTATAATAATAGGCAC   \n",
       "435101              TOP2A      chr17  ACACCGTAAATAATTTATAATAATAGGCAC   \n",
       "435102              COPS3      chr17  GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA   \n",
       "435103              COPS3      chr17  GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA   \n",
       "435104              KRT36      chr17  ATGTTCCTTCCCTTCCTACAGCAAGGGGAG   \n",
       "\n",
       "       condition Domain  DataSet  overlap with peak  \n",
       "0           K562   Kox1    Nunze                  0  \n",
       "1           K562   Kox1    Nunze                  0  \n",
       "2           K562   Kox1    Nunze                  0  \n",
       "3           K562   Kox1    Nunze                  0  \n",
       "4           K562   Kox1    Nunze                  0  \n",
       "...          ...    ...      ...                ...  \n",
       "435100    HCT116   Kox1  InHouse                  0  \n",
       "435101    HCT116   Zim3  InHouse                  0  \n",
       "435102    HCT116   Kox1  InHouse                  1  \n",
       "435103    HCT116   Zim3  InHouse                  1  \n",
       "435104    HCT116   Zim3  InHouse                  0  \n",
       "\n",
       "[435105 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrtest_hct116[['Target Gene Symbol', 'chromosome',\n",
    "       'sgRNA Context Sequence','condition', 'Domain', 'DataSet','overlap with peak']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "71e4f6d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>scaled_phenotype</th>\n",
       "      <th>orginal phenotype</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>overlap with peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>Gaggacagtgacttaggccc</td>\n",
       "      <td>gaggcaggacagtgacttaggcccaggagt</td>\n",
       "      <td>42734850.0</td>\n",
       "      <td>-</td>\n",
       "      <td>42737128.0</td>\n",
       "      <td>2279.0</td>\n",
       "      <td>0.115119</td>\n",
       "      <td>-0.018816</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>Gcgaggcaggacagtgactt</td>\n",
       "      <td>gaggccgaggcaggacagtgacttaggccc</td>\n",
       "      <td>42734856.0</td>\n",
       "      <td>-</td>\n",
       "      <td>42737128.0</td>\n",
       "      <td>2273.0</td>\n",
       "      <td>0.101538</td>\n",
       "      <td>-0.009835</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>GAAAGAATTGAGATCGggcc</td>\n",
       "      <td>CCCTGAAAGAATTGAGATCGggccaggtgt</td>\n",
       "      <td>42734918.0</td>\n",
       "      <td>-</td>\n",
       "      <td>42737128.0</td>\n",
       "      <td>2211.0</td>\n",
       "      <td>0.031007</td>\n",
       "      <td>0.036808</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>GGTGCTTAACACAGGGGCAG</td>\n",
       "      <td>TGCTGGTGCTTAACACAGGGGCAGTGGTCC</td>\n",
       "      <td>42735006.0</td>\n",
       "      <td>-</td>\n",
       "      <td>42737128.0</td>\n",
       "      <td>2123.0</td>\n",
       "      <td>0.084988</td>\n",
       "      <td>0.001110</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CDAN1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>GATGCTGGTGCTTAACACAG</td>\n",
       "      <td>GATATATGCTGGTGCTTAACACAGGGGCAG</td>\n",
       "      <td>42735012.0</td>\n",
       "      <td>-</td>\n",
       "      <td>42737128.0</td>\n",
       "      <td>2117.0</td>\n",
       "      <td>0.111959</td>\n",
       "      <td>-0.016726</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunze</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435100</th>\n",
       "      <td>TOP2A</td>\n",
       "      <td>chr17</td>\n",
       "      <td>CGTAAATAATTTATAATAAT</td>\n",
       "      <td>ACACCGTAAATAATTTATAATAATAGGCAC</td>\n",
       "      <td>40417221.0</td>\n",
       "      <td>-</td>\n",
       "      <td>40417896.0</td>\n",
       "      <td>676.0</td>\n",
       "      <td>0.302011</td>\n",
       "      <td>-0.356402</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435101</th>\n",
       "      <td>TOP2A</td>\n",
       "      <td>chr17</td>\n",
       "      <td>CGTAAATAATTTATAATAAT</td>\n",
       "      <td>ACACCGTAAATAATTTATAATAATAGGCAC</td>\n",
       "      <td>40417221.0</td>\n",
       "      <td>-</td>\n",
       "      <td>40417896.0</td>\n",
       "      <td>676.0</td>\n",
       "      <td>0.334489</td>\n",
       "      <td>-0.589737</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435102</th>\n",
       "      <td>COPS3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>TCGCCGCTCGGGCCGCCCGG</td>\n",
       "      <td>GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA</td>\n",
       "      <td>17281198.0</td>\n",
       "      <td>-</td>\n",
       "      <td>17281246.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.590759</td>\n",
       "      <td>-1.192805</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435103</th>\n",
       "      <td>COPS3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>TCGCCGCTCGGGCCGCCCGG</td>\n",
       "      <td>GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA</td>\n",
       "      <td>17281198.0</td>\n",
       "      <td>-</td>\n",
       "      <td>17281246.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0.544161</td>\n",
       "      <td>-1.157846</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435104</th>\n",
       "      <td>KRT36</td>\n",
       "      <td>chr17</td>\n",
       "      <td>TCCTTCCCTTCCTACAGCAA</td>\n",
       "      <td>ATGTTCCTTCCCTTCCTACAGCAAGGGGAG</td>\n",
       "      <td>41489187.0</td>\n",
       "      <td>-</td>\n",
       "      <td>41489910.0</td>\n",
       "      <td>724.0</td>\n",
       "      <td>0.394624</td>\n",
       "      <td>0.055050</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>435105 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome        sgRNA Sequence  \\\n",
       "0                   CDAN1      chr15  Gaggacagtgacttaggccc   \n",
       "1                   CDAN1      chr15  Gcgaggcaggacagtgactt   \n",
       "2                   CDAN1      chr15  GAAAGAATTGAGATCGggcc   \n",
       "3                   CDAN1      chr15  GGTGCTTAACACAGGGGCAG   \n",
       "4                   CDAN1      chr15  GATGCTGGTGCTTAACACAG   \n",
       "...                   ...        ...                   ...   \n",
       "435100              TOP2A      chr17  CGTAAATAATTTATAATAAT   \n",
       "435101              TOP2A      chr17  CGTAAATAATTTATAATAAT   \n",
       "435102              COPS3      chr17  TCGCCGCTCGGGCCGCCCGG   \n",
       "435103              COPS3      chr17  TCGCCGCTCGGGCCGCCCGG   \n",
       "435104              KRT36      chr17  TCCTTCCCTTCCTACAGCAA   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "0       gaggcaggacagtgacttaggcccaggagt            42734850.0                -   \n",
       "1       gaggccgaggcaggacagtgacttaggccc            42734856.0                -   \n",
       "2       CCCTGAAAGAATTGAGATCGggccaggtgt            42734918.0                -   \n",
       "3       TGCTGGTGCTTAACACAGGGGCAGTGGTCC            42735006.0                -   \n",
       "4       GATATATGCTGGTGCTTAACACAGGGGCAG            42735012.0                -   \n",
       "...                                ...                   ...              ...   \n",
       "435100  ACACCGTAAATAATTTATAATAATAGGCAC            40417221.0                -   \n",
       "435101  ACACCGTAAATAATTTATAATAATAGGCAC            40417221.0                -   \n",
       "435102  GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA            17281198.0                -   \n",
       "435103  GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA            17281198.0                -   \n",
       "435104  ATGTTCCTTCCCTTCCTACAGCAAGGGGAG            41489187.0                -   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  scaled_phenotype  \\\n",
       "0         42737128.0                       2279.0          0.115119   \n",
       "1         42737128.0                       2273.0          0.101538   \n",
       "2         42737128.0                       2211.0          0.031007   \n",
       "3         42737128.0                       2123.0          0.084988   \n",
       "4         42737128.0                       2117.0          0.111959   \n",
       "...              ...                          ...               ...   \n",
       "435100    40417896.0                        676.0          0.302011   \n",
       "435101    40417896.0                        676.0          0.334489   \n",
       "435102    17281246.0                         49.0          0.590759   \n",
       "435103    17281246.0                         49.0          0.544161   \n",
       "435104    41489910.0                        724.0          0.394624   \n",
       "\n",
       "        orginal phenotype condition Domain  DataSet  overlap with peak  \n",
       "0               -0.018816      K562   Kox1    Nunze                  0  \n",
       "1               -0.009835      K562   Kox1    Nunze                  0  \n",
       "2                0.036808      K562   Kox1    Nunze                  0  \n",
       "3                0.001110      K562   Kox1    Nunze                  0  \n",
       "4               -0.016726      K562   Kox1    Nunze                  0  \n",
       "...                   ...       ...    ...      ...                ...  \n",
       "435100          -0.356402    HCT116   Kox1  InHouse                  0  \n",
       "435101          -0.589737    HCT116   Zim3  InHouse                  0  \n",
       "435102          -1.192805    HCT116   Kox1  InHouse                  1  \n",
       "435103          -1.157846    HCT116   Zim3  InHouse                  1  \n",
       "435104           0.055050    HCT116   Zim3  InHouse                  0  \n",
       "\n",
       "[435105 rows x 14 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrtest_hct116"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "520b9704",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
