{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "70925395",
   "metadata": {},
   "source": [
    "Data Cleaning \n",
    "- Remove overlapping genes that have their TSS within 1kb of the TSS the selected genes and has at least overlap in category 2 or 3\n",
    "- TSS position is defined by ‘cut’ site in CRISPick\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "36bb4e87",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from copy import deepcopy\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "faa35ca4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def standardize_chromosome(chromosome):\n",
    "    chromosome = str(chromosome).replace('.0', '')\n",
    "    if not chromosome.startswith('chr'):\n",
    "        chromosome = 'chr' + chromosome\n",
    "    return chromosome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "eec04317",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>condition</th>\n",
       "      <th>avg_lfc</th>\n",
       "      <th>n_obs</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>Target Gene ID</th>\n",
       "      <th>Category</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>On-Target Ruleset</th>\n",
       "      <th>On-Target Efficacy Score</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "      <th>TSS Bucket</th>\n",
       "      <th>Domain</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>Strand of sgRNA</th>\n",
       "      <th>Orientation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAAAAAAAAAGGGCAGAAG</td>\n",
       "      <td>A549</td>\n",
       "      <td>-0.376159</td>\n",
       "      <td>2</td>\n",
       "      <td>APOBEC1</td>\n",
       "      <td>339.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>7665908.0</td>\n",
       "      <td>-999.0</td>\n",
       "      <td>AAAAAAAAAAAAAAAGGGCAGAAGTGGACT</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.7248</td>\n",
       "      <td>-1.389004</td>\n",
       "      <td>[-1000, -500)</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>12.0</td>\n",
       "      <td>7666908.0</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>sense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAAAAAAAAATTAGACCTC</td>\n",
       "      <td>A549</td>\n",
       "      <td>-0.021644</td>\n",
       "      <td>2</td>\n",
       "      <td>MRGPRD</td>\n",
       "      <td>116512.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>68980986.0</td>\n",
       "      <td>-735.0</td>\n",
       "      <td>AAAAAAAAAAAAAAATTAGACCTCAGGGCA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.6309</td>\n",
       "      <td>-0.528140</td>\n",
       "      <td>[-1000, -500)</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>68981722.0</td>\n",
       "      <td>-</td>\n",
       "      <td>+</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAAAAAAAAATTAGACCTC</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>-0.257163</td>\n",
       "      <td>2</td>\n",
       "      <td>MRGPRD</td>\n",
       "      <td>116512.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>68980986.0</td>\n",
       "      <td>-735.0</td>\n",
       "      <td>AAAAAAAAAAAAAAATTAGACCTCAGGGCA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.6309</td>\n",
       "      <td>-1.381448</td>\n",
       "      <td>[-1000, -500)</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>68981722.0</td>\n",
       "      <td>-</td>\n",
       "      <td>+</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAAAAAAAAATTAGACCTC</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>-0.277625</td>\n",
       "      <td>2</td>\n",
       "      <td>MRGPRD</td>\n",
       "      <td>116512.0</td>\n",
       "      <td>Non-essential</td>\n",
       "      <td>68980986.0</td>\n",
       "      <td>-735.0</td>\n",
       "      <td>AAAAAAAAAAAAAAATTAGACCTCAGGGCA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>-0.6309</td>\n",
       "      <td>-1.315569</td>\n",
       "      <td>[-1000, -500)</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>11.0</td>\n",
       "      <td>68981722.0</td>\n",
       "      <td>-</td>\n",
       "      <td>+</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAAAAAAAACAGGACACAG</td>\n",
       "      <td>A549</td>\n",
       "      <td>0.031810</td>\n",
       "      <td>2</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>23395.0</td>\n",
       "      <td>Essential</td>\n",
       "      <td>45388577.0</td>\n",
       "      <td>667.0</td>\n",
       "      <td>CAGAAAAAAAAAAACAGGACACAGGGGAGA</td>\n",
       "      <td>RS3seq-Chen2013</td>\n",
       "      <td>0.0140</td>\n",
       "      <td>-0.398339</td>\n",
       "      <td>[500, 1000)</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>45389244.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>antisense</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sgRNA Sequence condition   avg_lfc  n_obs Target Gene Symbol  \\\n",
       "0  AAAAAAAAAAAGGGCAGAAG      A549 -0.376159      2            APOBEC1   \n",
       "1  AAAAAAAAAAATTAGACCTC      A549 -0.021644      2             MRGPRD   \n",
       "2  AAAAAAAAAAATTAGACCTC    HCT116 -0.257163      2             MRGPRD   \n",
       "3  AAAAAAAAAAATTAGACCTC    HCT116 -0.277625      2             MRGPRD   \n",
       "4  AAAAAAAAAACAGGACACAG      A549  0.031810      2              LARS2   \n",
       "\n",
       "   Target Gene ID       Category  TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "0           339.0  Non-essential     7665908.0                       -999.0   \n",
       "1        116512.0  Non-essential    68980986.0                       -735.0   \n",
       "2        116512.0  Non-essential    68980986.0                       -735.0   \n",
       "3        116512.0  Non-essential    68980986.0                       -735.0   \n",
       "4         23395.0      Essential    45388577.0                        667.0   \n",
       "\n",
       "           sgRNA Context Sequence On-Target Ruleset  On-Target Efficacy Score  \\\n",
       "0  AAAAAAAAAAAAAAAGGGCAGAAGTGGACT   RS3seq-Chen2013                   -0.7248   \n",
       "1  AAAAAAAAAAAAAAATTAGACCTCAGGGCA   RS3seq-Chen2013                   -0.6309   \n",
       "2  AAAAAAAAAAAAAAATTAGACCTCAGGGCA   RS3seq-Chen2013                   -0.6309   \n",
       "3  AAAAAAAAAAAAAAATTAGACCTCAGGGCA   RS3seq-Chen2013                   -0.6309   \n",
       "4  CAGAAAAAAAAAAACAGGACACAGGGGAGA   RS3seq-Chen2013                    0.0140   \n",
       "\n",
       "   z_scored_avg_lfc     TSS Bucket Domain  chromosome  sgRNA 'Cut' Position  \\\n",
       "0         -1.389004  [-1000, -500)   Kox1        12.0             7666908.0   \n",
       "1         -0.528140  [-1000, -500)   Kox1        11.0            68981722.0   \n",
       "2         -1.381448  [-1000, -500)   Kox1        11.0            68981722.0   \n",
       "3         -1.315569  [-1000, -500)   Zim3        11.0            68981722.0   \n",
       "4         -0.398339    [500, 1000)   Kox1         3.0            45389244.0   \n",
       "\n",
       "  Strand of Target Strand of sgRNA Orientation  \n",
       "0                -               -       sense  \n",
       "1                -               +   antisense  \n",
       "2                -               +   antisense  \n",
       "3                -               +   antisense  \n",
       "4                +               -   antisense  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# In house CRISPRi Tiling\n",
    "path_out = '../Data/PrimaryLibrary/processed_z_score/'\n",
    "annot_guide_lfcs_A549_Kox1 = pd.read_csv(path_out + 'A549_Kox1_zscore.csv')\n",
    "annot_guide_lfcs_A549_Kox1 = annot_guide_lfcs_A549_Kox1[annot_guide_lfcs_A549_Kox1.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "annot_guide_lfcs_A549_Zim3 = pd.read_csv(path_out +  'A549_Zim3_zscore.csv')\n",
    "annot_guide_lfcs_A549_Zim3 = annot_guide_lfcs_A549_Zim3[annot_guide_lfcs_A549_Zim3.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "annot_guide_lfcs_HCT116_Kox1 = pd.read_csv(path_out +  'HCT116_Kox1_zscore.csv')\n",
    "annot_guide_lfcs_HCT116_Kox1 = annot_guide_lfcs_HCT116_Kox1[annot_guide_lfcs_HCT116_Kox1.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "annot_guide_lfcs_HCT116_Zim3 = pd.read_csv(path_out +  'HCT116_Zim3_zscore.csv')\n",
    "annot_guide_lfcs_HCT116_Zim3 = annot_guide_lfcs_HCT116_Zim3[annot_guide_lfcs_HCT116_Zim3.Category.isin(['Essential', 'Non-essential'])]\n",
    "\n",
    "\n",
    "\n",
    "annot_guide_lfcs_A549_Kox1['Domain'] = 'Kox1'\n",
    "annot_guide_lfcs_A549_Zim3['Domain'] = 'Zim3'\n",
    "annot_guide_lfcs_HCT116_Kox1['Domain'] = 'Kox1'\n",
    "annot_guide_lfcs_HCT116_Zim3['Domain'] = 'Zim3'\n",
    "\n",
    "InHouseCleanData = pd.concat([annot_guide_lfcs_A549_Kox1, annot_guide_lfcs_A549_Zim3, annot_guide_lfcs_HCT116_Kox1, annot_guide_lfcs_HCT116_Zim3])\n",
    "\n",
    "# Additional information for In house data\n",
    "ess_noness = pd.read_csv('../Data/PrimaryLibrary/design/essentials_nonessentials_CRISPRi_tiling_designs_v2.csv')\n",
    "ess_noness['chromosome'] = ess_noness['Reference Sequence'].apply(lambda x: int(x.split(\"NC_\")[1].split(\".\")[0]) if pd.notnull(x) else x)\n",
    "\n",
    "ess_noness = ess_noness[['Target Gene Symbol','Target Gene ID','Category','chromosome','sgRNA Sequence', \n",
    "           'sgRNA Context Sequence', 'TSS Position', \n",
    "       'sgRNA \\'Cut\\' Site TSS Offset', 'sgRNA \\'Cut\\' Position', \n",
    "            'Strand of Target','Strand of sgRNA', 'Orientation',\n",
    "            'On-Target Ruleset','On-Target Efficacy Score']]\n",
    "InHouseCleanData = pd.merge(InHouseCleanData, ess_noness)\n",
    "\n",
    "InHouseCleanData.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bca23ad4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Target Gene Symbol  strand\n",
       "ACTR6               +         159\n",
       "                    -         146\n",
       "AFG3L2              -         165\n",
       "                    +         119\n",
       "ANAPC13             +         139\n",
       "                             ... \n",
       "ZNF574              +         110\n",
       "ZNF830              +         105\n",
       "                    -          93\n",
       "ZNHIT6              +         127\n",
       "                    -          52\n",
       "Name: strand, Length: 925, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Nunze dataset\n",
    "NunzeCleanData = pd.read_csv('../Data/CleanedExternalData/Nunez2020Cleaned2024_v1.csv')\n",
    "NunzeCleanData['condition'] = 'K562'\n",
    "NunzeCleanData['Domain'] = 'Kox1'\n",
    "NunzeCleanData.groupby('Target Gene Symbol')['strand'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "21408eef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Target Gene Symbol  strand targeted\n",
       "ARCN1               -                  154\n",
       "                    +                  126\n",
       "ARF1                -                  521\n",
       "                    +                  485\n",
       "ARL1                +                  159\n",
       "                                      ... \n",
       "VPS54               -                  165\n",
       "WDR11               -                  129\n",
       "                    +                   99\n",
       "YIPF5               -                  185\n",
       "                    +                  166\n",
       "Name: strand targeted, Length: 94, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Gilbert Ricin Tiling\n",
    "GilbertRicinCleanData = pd.read_csv('../Data/CleanedExternalData/Gilbert2014Cleaned2024_v1.csv')\n",
    "GilbertRicinCleanData['condition'] = 'K562'\n",
    "GilbertRicinCleanData['Domain'] = 'Kox1'\n",
    "GilbertRicinCleanData.groupby('Target Gene Symbol')['strand targeted'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "68b24348",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24791"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(GilbertRicinCleanData)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d5398d9",
   "metadata": {},
   "source": [
    "# Cut Site reassign "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3fdb116a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cut site crispick\n",
    "def CRISPickCutSite(df, strand_col, point_col, context_seq_col):\n",
    "    CutSiteList = []\n",
    "    for index, row in df.iterrows():\n",
    "        if row[strand_col] == '+':\n",
    "            CutSiteList.append(row[point_col] + 7)\n",
    "        elif row[strand_col] == '-':\n",
    "            CutSiteList.append(row[point_col] - 4)\n",
    "        else:\n",
    "            print(row[context_seq_col] + f'strand is neither + nor -')\n",
    "    return CutSiteList\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "56ef71de",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneListGilbertAndNunez = GilbertRicinCleanData['Target Gene Symbol'].unique().tolist() + NunzeCleanData['Target Gene Symbol'].unique().tolist()\n",
    "\n",
    "pd.DataFrame(geneListGilbertAndNunez).to_csv('../Data/ExternalData/CRISPick/geneListGilbertAndNunez.csv',\n",
    "                              index = False)\n",
    "#ran CRISPick https://portals.broadinstitute.org/gppx/crispick/platform/results/1da3859a-902d-471a-b06b-7aa43da31abd\n",
    "# NCBI GRCh38 is used because it has MANE select TSS Position\n",
    "CRISPick_geneListGilbertAndNunez = pd.read_table('../Data/ExternalData/CRISPick/geneListGilbertAndNunez-sgrna-designs.txt')\n",
    "CRISPick_geneListGilbertAndNunez = CRISPick_geneListGilbertAndNunez[[ \n",
    "       'Target Gene Symbol',  'Reference Sequence', 'Strand of Target',\n",
    "       'TSS Position']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92625f35",
   "metadata": {},
   "source": [
    "# Gilbert"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c24742d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "      <th>context seq</th>\n",
       "      <th>guide id</th>\n",
       "      <th>oligo</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>zscoreRelativeNC</th>\n",
       "      <th>...</th>\n",
       "      <th>Target Gene Function</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567638.0</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGTAGGTTT</td>\n",
       "      <td>ARCN1_w_118438354.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGT</td>\n",
       "      <td>0.001556</td>\n",
       "      <td>0.250358</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.001556</td>\n",
       "      <td>-0.250358</td>\n",
       "      <td>0.134928</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118567645.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568026.0</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTGAGGAGT</td>\n",
       "      <td>ARCN1_w_118438742.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTG</td>\n",
       "      <td>-0.011286</td>\n",
       "      <td>0.157641</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>0.011286</td>\n",
       "      <td>-0.157641</td>\n",
       "      <td>0.166789</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568033.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568080.0</td>\n",
       "      <td>GATGCTGCTTCTTCGGAAATATAACGGTAG</td>\n",
       "      <td>ARCN1_w_118438796.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...</td>\n",
       "      <td>GCTGCTTCTTCGGAAATATAA</td>\n",
       "      <td>0.018150</td>\n",
       "      <td>0.216340</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.018150</td>\n",
       "      <td>-0.216340</td>\n",
       "      <td>0.146618</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568087.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568134.0</td>\n",
       "      <td>TAGCACCTTCCAGGGCTTCCAACTTGGATC</td>\n",
       "      <td>ARCN1_w_118438850.25</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...</td>\n",
       "      <td>GCACCTTCCAGGGCTTCCAACT</td>\n",
       "      <td>0.021472</td>\n",
       "      <td>0.320467</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>-0.021472</td>\n",
       "      <td>-0.320467</td>\n",
       "      <td>0.110835</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568141.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568146.0</td>\n",
       "      <td>ATTGCCTTATTGTAGCACCTTCCAGGGCTT</td>\n",
       "      <td>ARCN1_w_118438862.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...</td>\n",
       "      <td>GCCTTATTGTAGCACCTTCCA</td>\n",
       "      <td>-0.018377</td>\n",
       "      <td>-0.708019</td>\n",
       "      <td>...</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>0.018377</td>\n",
       "      <td>0.708019</td>\n",
       "      <td>0.464269</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568153.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Target Gene Symbol chromosome strand targeted  \\\n",
       "0              ARCN1      chr11               +   \n",
       "1              ARCN1      chr11               +   \n",
       "2              ARCN1      chr11               +   \n",
       "3              ARCN1      chr11               +   \n",
       "4              ARCN1      chr11               +   \n",
       "\n",
       "   PAM genomic coordinate [hg38]                     context seq  \\\n",
       "0                    118567638.0  GCACTTTCCTTGTTTACCTCTGGTAGGTTT   \n",
       "1                    118568026.0  GCAACCCTGTAAGCATGCTTCTTGAGGAGT   \n",
       "2                    118568080.0  GATGCTGCTTCTTCGGAAATATAACGGTAG   \n",
       "3                    118568134.0  TAGCACCTTCCAGGGCTTCCAACTTGGATC   \n",
       "4                    118568146.0  ATTGCCTTATTGTAGCACCTTCCAGGGCTT   \n",
       "\n",
       "               guide id                                              oligo  \\\n",
       "0  ARCN1_w_118438354.27  cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...   \n",
       "1  ARCN1_w_118438742.27  cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...   \n",
       "2  ARCN1_w_118438796.24  cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...   \n",
       "3  ARCN1_w_118438850.25  cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...   \n",
       "4  ARCN1_w_118438862.24  cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...   \n",
       "\n",
       "             sgRNA sequence  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "0  GCACTTTCCTTGTTTACCTCTGGT                          0.001556   \n",
       "1  GCAACCCTGTAAGCATGCTTCTTG                         -0.011286   \n",
       "2     GCTGCTTCTTCGGAAATATAA                          0.018150   \n",
       "3    GCACCTTCCAGGGCTTCCAACT                          0.021472   \n",
       "4     GCCTTATTGTAGCACCTTCCA                         -0.018377   \n",
       "\n",
       "   zscoreRelativeNC  ...  Target Gene Function  rhoSignChangedAndAvged  \\\n",
       "0          0.250358  ...            Resistance               -0.001556   \n",
       "1          0.157641  ...            Resistance                0.011286   \n",
       "2          0.216340  ...            Resistance               -0.018150   \n",
       "3          0.320467  ...            Resistance               -0.021472   \n",
       "4         -0.708019  ...            Resistance                0.018377   \n",
       "\n",
       "  zscoreRelativeNC_signed  Avg_LFC_signed  condition  Domain CRISPick CutSite  \\\n",
       "0               -0.250358        0.134928       K562    Kox1      118567645.0   \n",
       "1               -0.157641        0.166789       K562    Kox1      118568033.0   \n",
       "2               -0.216340        0.146618       K562    Kox1      118568087.0   \n",
       "3               -0.320467        0.110835       K562    Kox1      118568141.0   \n",
       "4                0.708019        0.464269       K562    Kox1      118568153.0   \n",
       "\n",
       "  Reference Sequence  Strand of Target TSS Position  \n",
       "0       NC_000011.10                 +    118572410  \n",
       "1       NC_000011.10                 +    118572410  \n",
       "2       NC_000011.10                 +    118572410  \n",
       "3       NC_000011.10                 +    118572410  \n",
       "4       NC_000011.10                 +    118572410  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GilbertRicinCleanDataCutSite = CRISPickCutSite(GilbertRicinCleanData,'strand targeted' ,'PAM genomic coordinate [hg38]', 'context seq')\n",
    "GilbertRicinCleanData['CRISPick CutSite'] = GilbertRicinCleanDataCutSite\n",
    "GilbertRicinCleanData = pd.merge(GilbertRicinCleanData,CRISPick_geneListGilbertAndNunez, \n",
    "                                                 on = 'Target Gene Symbol')\n",
    "GilbertRicinCleanData.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4fbd09cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand targeted</th>\n",
       "      <th>PAM genomic coordinate [hg38]</th>\n",
       "      <th>context seq</th>\n",
       "      <th>guide id</th>\n",
       "      <th>oligo</th>\n",
       "      <th>sgRNA sequence</th>\n",
       "      <th>dCas9-KRAB_rho_ave_LG2_LG_MH2_MH</th>\n",
       "      <th>zscoreRelativeNC</th>\n",
       "      <th>...</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>Distance to TSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118567638.0</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGTAGGTTT</td>\n",
       "      <td>ARCN1_w_118438354.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...</td>\n",
       "      <td>GCACTTTCCTTGTTTACCTCTGGT</td>\n",
       "      <td>0.001556</td>\n",
       "      <td>0.250358</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.001556</td>\n",
       "      <td>-0.250358</td>\n",
       "      <td>0.134928</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118567645.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4765.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568026.0</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTGAGGAGT</td>\n",
       "      <td>ARCN1_w_118438742.27</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...</td>\n",
       "      <td>GCAACCCTGTAAGCATGCTTCTTG</td>\n",
       "      <td>-0.011286</td>\n",
       "      <td>0.157641</td>\n",
       "      <td>...</td>\n",
       "      <td>0.011286</td>\n",
       "      <td>-0.157641</td>\n",
       "      <td>0.166789</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568033.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4377.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568080.0</td>\n",
       "      <td>GATGCTGCTTCTTCGGAAATATAACGGTAG</td>\n",
       "      <td>ARCN1_w_118438796.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...</td>\n",
       "      <td>GCTGCTTCTTCGGAAATATAA</td>\n",
       "      <td>0.018150</td>\n",
       "      <td>0.216340</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.018150</td>\n",
       "      <td>-0.216340</td>\n",
       "      <td>0.146618</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568087.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4323.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568134.0</td>\n",
       "      <td>TAGCACCTTCCAGGGCTTCCAACTTGGATC</td>\n",
       "      <td>ARCN1_w_118438850.25</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...</td>\n",
       "      <td>GCACCTTCCAGGGCTTCCAACT</td>\n",
       "      <td>0.021472</td>\n",
       "      <td>0.320467</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.021472</td>\n",
       "      <td>-0.320467</td>\n",
       "      <td>0.110835</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568141.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4269.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARCN1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>118568146.0</td>\n",
       "      <td>ATTGCCTTATTGTAGCACCTTCCAGGGCTT</td>\n",
       "      <td>ARCN1_w_118438862.24</td>\n",
       "      <td>cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...</td>\n",
       "      <td>GCCTTATTGTAGCACCTTCCA</td>\n",
       "      <td>-0.018377</td>\n",
       "      <td>-0.708019</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018377</td>\n",
       "      <td>0.708019</td>\n",
       "      <td>0.464269</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>118568153.0</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>+</td>\n",
       "      <td>118572410</td>\n",
       "      <td>-4257.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Target Gene Symbol chromosome strand targeted  \\\n",
       "0              ARCN1      chr11               +   \n",
       "1              ARCN1      chr11               +   \n",
       "2              ARCN1      chr11               +   \n",
       "3              ARCN1      chr11               +   \n",
       "4              ARCN1      chr11               +   \n",
       "\n",
       "   PAM genomic coordinate [hg38]                     context seq  \\\n",
       "0                    118567638.0  GCACTTTCCTTGTTTACCTCTGGTAGGTTT   \n",
       "1                    118568026.0  GCAACCCTGTAAGCATGCTTCTTGAGGAGT   \n",
       "2                    118568080.0  GATGCTGCTTCTTCGGAAATATAACGGTAG   \n",
       "3                    118568134.0  TAGCACCTTCCAGGGCTTCCAACTTGGATC   \n",
       "4                    118568146.0  ATTGCCTTATTGTAGCACCTTCCAGGGCTT   \n",
       "\n",
       "               guide id                                              oligo  \\\n",
       "0  ARCN1_w_118438354.27  cccttggagaaCCAcctTGTTGGCACTTTCCTTGTTTACCTCTGGT...   \n",
       "1  ARCN1_w_118438742.27  cccttggagaaCCAcctTGTTGGCAACCCTGTAAGCATGCTTCTTG...   \n",
       "2  ARCN1_w_118438796.24  cccttggagaaCCAcctTGTTGGCTGCTTCTTCGGAAATATAAGTT...   \n",
       "3  ARCN1_w_118438850.25  cccttggagaaCCAcctTGTTGGCACCTTCCAGGGCTTCCAACTGT...   \n",
       "4  ARCN1_w_118438862.24  cccttggagaaCCAcctTGTTGGCCTTATTGTAGCACCTTCCAGTT...   \n",
       "\n",
       "             sgRNA sequence  dCas9-KRAB_rho_ave_LG2_LG_MH2_MH  \\\n",
       "0  GCACTTTCCTTGTTTACCTCTGGT                          0.001556   \n",
       "1  GCAACCCTGTAAGCATGCTTCTTG                         -0.011286   \n",
       "2     GCTGCTTCTTCGGAAATATAA                          0.018150   \n",
       "3    GCACCTTCCAGGGCTTCCAACT                          0.021472   \n",
       "4     GCCTTATTGTAGCACCTTCCA                         -0.018377   \n",
       "\n",
       "   zscoreRelativeNC  ...  rhoSignChangedAndAvged  zscoreRelativeNC_signed  \\\n",
       "0          0.250358  ...               -0.001556                -0.250358   \n",
       "1          0.157641  ...                0.011286                -0.157641   \n",
       "2          0.216340  ...               -0.018150                -0.216340   \n",
       "3          0.320467  ...               -0.021472                -0.320467   \n",
       "4         -0.708019  ...                0.018377                 0.708019   \n",
       "\n",
       "  Avg_LFC_signed  condition  Domain  CRISPick CutSite Reference Sequence  \\\n",
       "0       0.134928       K562    Kox1       118567645.0       NC_000011.10   \n",
       "1       0.166789       K562    Kox1       118568033.0       NC_000011.10   \n",
       "2       0.146618       K562    Kox1       118568087.0       NC_000011.10   \n",
       "3       0.110835       K562    Kox1       118568141.0       NC_000011.10   \n",
       "4       0.464269       K562    Kox1       118568153.0       NC_000011.10   \n",
       "\n",
       "  Strand of Target  TSS Position Distance to TSS  \n",
       "0                +     118572410         -4765.0  \n",
       "1                +     118572410         -4377.0  \n",
       "2                +     118572410         -4323.0  \n",
       "3                +     118572410         -4269.0  \n",
       "4                +     118572410         -4257.0  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#calculate guide distance from MANESelect annotated TSS \n",
    "GilbertRicinCleanData[\"Distance to TSS\"] = GilbertRicinCleanData.apply(\n",
    "    lambda r:(r[\"CRISPick CutSite\"] - r[\"TSS Position\"]) if r[\"Strand of Target\"] == \"+\" \\\n",
    "              else (r[\"TSS Position\"] - r[\"CRISPick CutSite\"]), axis=1)\n",
    "GilbertRicinCleanData = GilbertRicinCleanData[abs(GilbertRicinCleanData['Distance to TSS']) <= 5000]\n",
    "GilbertRicinCleanData.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be600e4b",
   "metadata": {},
   "source": [
    "# Nunze"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2dd65b43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand</th>\n",
       "      <th>sequence</th>\n",
       "      <th>context_seq</th>\n",
       "      <th>hg38_coord</th>\n",
       "      <th>start_coord_hg38</th>\n",
       "      <th>end_coord_hg38</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6_+_100592059</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCGAGCACATAGAGGGATCT</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>100198281</td>\n",
       "      <td>100198284</td>\n",
       "      <td>100198303</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>-0.889828</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198288</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTR6_+_100592060</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GTCGAGCACATAGAGGGATC</td>\n",
       "      <td>CTGCTTCGAGCACATAGAGGGATCTGGGCC</td>\n",
       "      <td>100198282</td>\n",
       "      <td>100198285</td>\n",
       "      <td>100198304</td>\n",
       "      <td>0.009292</td>\n",
       "      <td>0.295731</td>\n",
       "      <td>0.580559</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198289</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTR6_+_100592066</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCCTGCTTCGAGCACATAGA</td>\n",
       "      <td>GTCCACCTGCTTCGAGCACATAGAGGGATC</td>\n",
       "      <td>100198288</td>\n",
       "      <td>100198291</td>\n",
       "      <td>100198310</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.214023</td>\n",
       "      <td>0.119582</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198295</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID Target Gene Symbol chromosome strand  \\\n",
       "0  ACTR6_+_100592059              ACTR6      chr12      +   \n",
       "1  ACTR6_+_100592060              ACTR6      chr12      +   \n",
       "2  ACTR6_+_100592066              ACTR6      chr12      +   \n",
       "\n",
       "               sequence                     context_seq  hg38_coord  \\\n",
       "0  GCGAGCACATAGAGGGATCT  TGCTTCGAGCACATAGAGGGATCTGGGCCC   100198281   \n",
       "1  GTCGAGCACATAGAGGGATC  CTGCTTCGAGCACATAGAGGGATCTGGGCC   100198282   \n",
       "2  GCCTGCTTCGAGCACATAGA  GTCCACCTGCTTCGAGCACATAGAGGGATC   100198288   \n",
       "\n",
       "   start_coord_hg38  end_coord_hg38  Phenotype scores-ave_Rep1_Rep2  \\\n",
       "0         100198284       100198303                       -0.018881   \n",
       "1         100198285       100198304                        0.009292   \n",
       "2         100198291       100198310                        0.000734   \n",
       "\n",
       "   Avg_LFC-Tfinal  z-score relative to NC condition Domain  CRISPick CutSite  \\\n",
       "0        0.035104               -0.889828      K562   Kox1         100198288   \n",
       "1        0.295731                0.580559      K562   Kox1         100198289   \n",
       "2        0.214023                0.119582      K562   Kox1         100198295   \n",
       "\n",
       "  Reference Sequence Strand of Target  TSS Position  \n",
       "0       NC_000012.12                +     100200816  \n",
       "1       NC_000012.12                +     100200816  \n",
       "2       NC_000012.12                +     100200816  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NunzeCleanDataCutSite = CRISPickCutSite(NunzeCleanData,'strand' ,'hg38_coord', 'context_seq')\n",
    "NunzeCleanData['CRISPick CutSite'] = NunzeCleanDataCutSite\n",
    "NunzeCleanData = pd.merge(NunzeCleanData, CRISPick_geneListGilbertAndNunez,\n",
    "                          on= ['Target Gene Symbol'],how = 'inner')\n",
    "NunzeCleanData.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "af89468d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>strand</th>\n",
       "      <th>sequence</th>\n",
       "      <th>context_seq</th>\n",
       "      <th>hg38_coord</th>\n",
       "      <th>start_coord_hg38</th>\n",
       "      <th>end_coord_hg38</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>CRISPick CutSite</th>\n",
       "      <th>Reference Sequence</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>Distance to TSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTR6_+_100592059</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCGAGCACATAGAGGGATCT</td>\n",
       "      <td>TGCTTCGAGCACATAGAGGGATCTGGGCCC</td>\n",
       "      <td>100198281</td>\n",
       "      <td>100198284</td>\n",
       "      <td>100198303</td>\n",
       "      <td>-0.018881</td>\n",
       "      <td>0.035104</td>\n",
       "      <td>-0.889828</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198288</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTR6_+_100592060</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GTCGAGCACATAGAGGGATC</td>\n",
       "      <td>CTGCTTCGAGCACATAGAGGGATCTGGGCC</td>\n",
       "      <td>100198282</td>\n",
       "      <td>100198285</td>\n",
       "      <td>100198304</td>\n",
       "      <td>0.009292</td>\n",
       "      <td>0.295731</td>\n",
       "      <td>0.580559</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198289</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTR6_+_100592066</td>\n",
       "      <td>ACTR6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>GCCTGCTTCGAGCACATAGA</td>\n",
       "      <td>GTCCACCTGCTTCGAGCACATAGAGGGATC</td>\n",
       "      <td>100198288</td>\n",
       "      <td>100198291</td>\n",
       "      <td>100198310</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.214023</td>\n",
       "      <td>0.119582</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>100198295</td>\n",
       "      <td>NC_000012.12</td>\n",
       "      <td>+</td>\n",
       "      <td>100200816</td>\n",
       "      <td>-2521</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID Target Gene Symbol chromosome strand  \\\n",
       "0  ACTR6_+_100592059              ACTR6      chr12      +   \n",
       "1  ACTR6_+_100592060              ACTR6      chr12      +   \n",
       "2  ACTR6_+_100592066              ACTR6      chr12      +   \n",
       "\n",
       "               sequence                     context_seq  hg38_coord  \\\n",
       "0  GCGAGCACATAGAGGGATCT  TGCTTCGAGCACATAGAGGGATCTGGGCCC   100198281   \n",
       "1  GTCGAGCACATAGAGGGATC  CTGCTTCGAGCACATAGAGGGATCTGGGCC   100198282   \n",
       "2  GCCTGCTTCGAGCACATAGA  GTCCACCTGCTTCGAGCACATAGAGGGATC   100198288   \n",
       "\n",
       "   start_coord_hg38  end_coord_hg38  Phenotype scores-ave_Rep1_Rep2  \\\n",
       "0         100198284       100198303                       -0.018881   \n",
       "1         100198285       100198304                        0.009292   \n",
       "2         100198291       100198310                        0.000734   \n",
       "\n",
       "   Avg_LFC-Tfinal  z-score relative to NC condition Domain  CRISPick CutSite  \\\n",
       "0        0.035104               -0.889828      K562   Kox1         100198288   \n",
       "1        0.295731                0.580559      K562   Kox1         100198289   \n",
       "2        0.214023                0.119582      K562   Kox1         100198295   \n",
       "\n",
       "  Reference Sequence Strand of Target  TSS Position  Distance to TSS  \n",
       "0       NC_000012.12                +     100200816            -2528  \n",
       "1       NC_000012.12                +     100200816            -2527  \n",
       "2       NC_000012.12                +     100200816            -2521  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#calculate guide distance from MANESelect annotated TSS \n",
    "NunzeCleanData[\"Distance to TSS\"] = NunzeCleanData.apply(\n",
    "    lambda r:(r[\"CRISPick CutSite\"] - r[\"TSS Position\"]) if r[\"Strand of Target\"] == \"+\" \\\n",
    "              else (r[\"TSS Position\"] - r[\"CRISPick CutSite\"]), axis=1)\n",
    "NunzeCleanData = NunzeCleanData[abs(NunzeCleanData['Distance to TSS'])<=5000]\n",
    "NunzeCleanData.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8884d31b",
   "metadata": {},
   "source": [
    "# Combine datasets together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "debfb94d",
   "metadata": {},
   "outputs": [],
   "source": [
    "GilbertRicinCleanData = deepcopy(GilbertRicinCleanData[['Target Gene Symbol','chromosome','sgRNA sequence', 'context seq', \n",
    "                                'CRISPick CutSite', 'Strand of Target', 'TSS Position', 'Distance to TSS',\n",
    "                                'rhoSignChangedAndAvged','zscoreRelativeNC_signed', 'Avg_LFC_signed', \n",
    "                                'condition', 'Domain']])\n",
    "\n",
    "\n",
    "GilbertRicinCleanData.rename(columns={\n",
    "                                      'sgRNA sequence':'sgRNA Sequence', \n",
    "                                      'context seq':'sgRNA Context Sequence',\n",
    "                                      'CRISPick CutSite': 'sgRNA \\'Cut\\' Position', \n",
    "                                      'Distance to TSS':'sgRNA \\'Cut\\' Site TSS Offset'\n",
    "                                     }, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3986dca5",
   "metadata": {},
   "outputs": [],
   "source": [
    "NunzeCleanData = deepcopy(NunzeCleanData[['Target Gene Symbol','chromosome','sequence', \n",
    "                                               'context_seq', 'CRISPick CutSite',\n",
    "                                                       'Strand of Target', 'TSS Position', 'Distance to TSS',\n",
    "                                                       'Phenotype scores-ave_Rep1_Rep2','Avg_LFC-Tfinal', \n",
    "                         'z-score relative to NC','condition','Domain']])\n",
    "\n",
    "NunzeCleanData.rename(columns={'sequence':'sgRNA Sequence', \n",
    "                                    'context_seq':'sgRNA Context Sequence',\n",
    "                                   'CRISPick CutSite': 'sgRNA \\'Cut\\' Position', \n",
    "                                      'Distance to TSS':'sgRNA \\'Cut\\' Site TSS Offset'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "435bce6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "InHouseCleanData = deepcopy(InHouseCleanData\\\n",
    "                                 [['Target Gene Symbol','chromosome','sgRNA Sequence', \n",
    "                                   'sgRNA Context Sequence','sgRNA \\'Cut\\' Position',\n",
    "                                   'Strand of Target', 'TSS Position', 'sgRNA \\'Cut\\' Site TSS Offset',\n",
    "                                                       'z_scored_avg_lfc',\n",
    "                                                              'condition','Domain']])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "6e55e0dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "NunzeCleanData['DataSet'] = 'Nunez'\n",
    "GilbertRicinCleanData['DataSet'] = 'Gilbert'\n",
    "InHouseCleanData['DataSet'] = 'InHouse'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "352e90f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "DatasetCombine = pd.concat([NunzeCleanData, GilbertRicinCleanData, InHouseCleanData]).reset_index(drop = True)\n",
    "#na in some columns is because the corrsponding dataset doesnt has that phenotype"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49e508b1",
   "metadata": {},
   "source": [
    "# Quality Check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "ba0dba24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GGGACCTCATATTGTGCCCGCCTAGGGACT    6\n",
       "GCCGTCCCCAGGCAGAATTAGAGTCGGGGT    6\n",
       "TCCTGGGCAGAGGCTAGTGAGGCCGGGGTC    6\n",
       "GGAGTGGAGAGATATCCTGGGCAGAGGCTA    6\n",
       "CAGGGAGGAGTGGAGAGATATCCTGGGCAG    6\n",
       "                                 ..\n",
       "GTATCCTGAACTTTGAGGAGTTATGGGAGG    2\n",
       "TGTATCCTGAACTTTGAGGAGTTATGGGAG    2\n",
       "GCGCCTGGTGTATCCTGAACTTTGAGGAGT    2\n",
       "GACTTTATTGTGAATCACAGCGCCTGGTGT    2\n",
       "GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA    2\n",
       "Name: sgRNA Context Sequence, Length: 107393, dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DatasetCombine[DatasetCombine['sgRNA Context Sequence'].duplicated(False)]['sgRNA Context Sequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "de8bcecb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>105792</th>\n",
       "      <td>PRKDC</td>\n",
       "      <td>chr8</td>\n",
       "      <td>GAATATTTATTGCAACCTGCGG</td>\n",
       "      <td>CTGAATATTTATTGCAACCTGCGGTGGCTG</td>\n",
       "      <td>47961937.0</td>\n",
       "      <td>-</td>\n",
       "      <td>47960136.0</td>\n",
       "      <td>-1801.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>0.049736</td>\n",
       "      <td>0.464529</td>\n",
       "      <td>0.380595</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198691</th>\n",
       "      <td>MCM4</td>\n",
       "      <td>8.0</td>\n",
       "      <td>ATATTTATTGCAACCTGCGG</td>\n",
       "      <td>CTGAATATTTATTGCAACCTGCGGTGGCTG</td>\n",
       "      <td>47961937.0</td>\n",
       "      <td>+</td>\n",
       "      <td>47960942.0</td>\n",
       "      <td>995.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.479500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198692</th>\n",
       "      <td>MCM4</td>\n",
       "      <td>8.0</td>\n",
       "      <td>ATATTTATTGCAACCTGCGG</td>\n",
       "      <td>CTGAATATTTATTGCAACCTGCGGTGGCTG</td>\n",
       "      <td>47961937.0</td>\n",
       "      <td>+</td>\n",
       "      <td>47960942.0</td>\n",
       "      <td>995.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.501068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198693</th>\n",
       "      <td>MCM4</td>\n",
       "      <td>8.0</td>\n",
       "      <td>ATATTTATTGCAACCTGCGG</td>\n",
       "      <td>CTGAATATTTATTGCAACCTGCGGTGGCTG</td>\n",
       "      <td>47961937.0</td>\n",
       "      <td>+</td>\n",
       "      <td>47960942.0</td>\n",
       "      <td>995.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.700090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198694</th>\n",
       "      <td>MCM4</td>\n",
       "      <td>8.0</td>\n",
       "      <td>ATATTTATTGCAACCTGCGG</td>\n",
       "      <td>CTGAATATTTATTGCAACCTGCGGTGGCTG</td>\n",
       "      <td>47961937.0</td>\n",
       "      <td>+</td>\n",
       "      <td>47960942.0</td>\n",
       "      <td>995.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.084287</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome          sgRNA Sequence  \\\n",
       "105792              PRKDC       chr8  GAATATTTATTGCAACCTGCGG   \n",
       "198691               MCM4        8.0    ATATTTATTGCAACCTGCGG   \n",
       "198692               MCM4        8.0    ATATTTATTGCAACCTGCGG   \n",
       "198693               MCM4        8.0    ATATTTATTGCAACCTGCGG   \n",
       "198694               MCM4        8.0    ATATTTATTGCAACCTGCGG   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "105792  CTGAATATTTATTGCAACCTGCGGTGGCTG            47961937.0                -   \n",
       "198691  CTGAATATTTATTGCAACCTGCGGTGGCTG            47961937.0                +   \n",
       "198692  CTGAATATTTATTGCAACCTGCGGTGGCTG            47961937.0                +   \n",
       "198693  CTGAATATTTATTGCAACCTGCGGTGGCTG            47961937.0                +   \n",
       "198694  CTGAATATTTATTGCAACCTGCGGTGGCTG            47961937.0                +   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "105792    47960136.0                      -1801.0   \n",
       "198691    47960942.0                        995.0   \n",
       "198692    47960942.0                        995.0   \n",
       "198693    47960942.0                        995.0   \n",
       "198694    47960942.0                        995.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "105792                             NaN             NaN   \n",
       "198691                             NaN             NaN   \n",
       "198692                             NaN             NaN   \n",
       "198693                             NaN             NaN   \n",
       "198694                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "105792                     NaN      K562   Kox1  Gilbert   \n",
       "198691                     NaN      A549   Kox1  InHouse   \n",
       "198692                     NaN      A549   Zim3  InHouse   \n",
       "198693                     NaN    HCT116   Kox1  InHouse   \n",
       "198694                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "105792                0.049736                 0.464529        0.380595   \n",
       "198691                     NaN                      NaN             NaN   \n",
       "198692                     NaN                      NaN             NaN   \n",
       "198693                     NaN                      NaN             NaN   \n",
       "198694                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "105792               NaN  \n",
       "198691         -2.479500  \n",
       "198692         -1.501068  \n",
       "198693         -1.700090  \n",
       "198694         -1.084287  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# the same context is targeting two genes aka gene overlap\n",
    "DatasetCombine[DatasetCombine['sgRNA Context Sequence'] == 'CTGAATATTTATTGCAACCTGCGGTGGCTG']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7a48d56e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25811</th>\n",
       "      <td>EMG1</td>\n",
       "      <td>chr12</td>\n",
       "      <td>GCGCGTGGTGAGGGGAAACC</td>\n",
       "      <td>AAAAACGCGTGGTGAGGGGAAACCGGGCAA</td>\n",
       "      <td>6969946.0</td>\n",
       "      <td>+</td>\n",
       "      <td>6970914.0</td>\n",
       "      <td>-968.0</td>\n",
       "      <td>-0.032545</td>\n",
       "      <td>-0.094882</td>\n",
       "      <td>-1.623168</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155817</th>\n",
       "      <td>PHB2</td>\n",
       "      <td>12.0</td>\n",
       "      <td>ACGCGTGGTGAGGGGAAACC</td>\n",
       "      <td>AAAAACGCGTGGTGAGGGGAAACCGGGCAA</td>\n",
       "      <td>6969946.0</td>\n",
       "      <td>-</td>\n",
       "      <td>6970664.0</td>\n",
       "      <td>719.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.518868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155818</th>\n",
       "      <td>PHB2</td>\n",
       "      <td>12.0</td>\n",
       "      <td>ACGCGTGGTGAGGGGAAACC</td>\n",
       "      <td>AAAAACGCGTGGTGAGGGGAAACCGGGCAA</td>\n",
       "      <td>6969946.0</td>\n",
       "      <td>-</td>\n",
       "      <td>6970664.0</td>\n",
       "      <td>719.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-6.784053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155819</th>\n",
       "      <td>PHB2</td>\n",
       "      <td>12.0</td>\n",
       "      <td>ACGCGTGGTGAGGGGAAACC</td>\n",
       "      <td>AAAAACGCGTGGTGAGGGGAAACCGGGCAA</td>\n",
       "      <td>6969946.0</td>\n",
       "      <td>-</td>\n",
       "      <td>6970664.0</td>\n",
       "      <td>719.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-8.892879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155820</th>\n",
       "      <td>PHB2</td>\n",
       "      <td>12.0</td>\n",
       "      <td>ACGCGTGGTGAGGGGAAACC</td>\n",
       "      <td>AAAAACGCGTGGTGAGGGGAAACCGGGCAA</td>\n",
       "      <td>6969946.0</td>\n",
       "      <td>-</td>\n",
       "      <td>6970664.0</td>\n",
       "      <td>719.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-8.323199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351648</th>\n",
       "      <td>PTCD3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>GATCTGTGATAGTTATAGCT</td>\n",
       "      <td>TTTTGATCTGTGATAGTTATAGCTTGGAAC</td>\n",
       "      <td>86106851.0</td>\n",
       "      <td>+</td>\n",
       "      <td>86106236.0</td>\n",
       "      <td>615.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.048226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351650</th>\n",
       "      <td>PTCD3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>GATCTGTGATAGTTATAGCT</td>\n",
       "      <td>TTTTGATCTGTGATAGTTATAGCTTGGAAC</td>\n",
       "      <td>86106851.0</td>\n",
       "      <td>+</td>\n",
       "      <td>86106236.0</td>\n",
       "      <td>615.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.620939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351651</th>\n",
       "      <td>PTCD3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>GATCTGTGATAGTTATAGCT</td>\n",
       "      <td>TTTTGATCTGTGATAGTTATAGCTTGGAAC</td>\n",
       "      <td>86106851.0</td>\n",
       "      <td>+</td>\n",
       "      <td>86106236.0</td>\n",
       "      <td>615.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.605212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56877</th>\n",
       "      <td>POLR1A</td>\n",
       "      <td>chr2</td>\n",
       "      <td>GATCTGTGATAGTTATAGCT</td>\n",
       "      <td>TTTTGATCTGTGATAGTTATAGCTTGGAAC</td>\n",
       "      <td>86106851.0</td>\n",
       "      <td>-</td>\n",
       "      <td>86105886.0</td>\n",
       "      <td>-965.0</td>\n",
       "      <td>-0.006604</td>\n",
       "      <td>0.115769</td>\n",
       "      <td>-0.434737</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351649</th>\n",
       "      <td>PTCD3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>GATCTGTGATAGTTATAGCT</td>\n",
       "      <td>TTTTGATCTGTGATAGTTATAGCTTGGAAC</td>\n",
       "      <td>86106851.0</td>\n",
       "      <td>+</td>\n",
       "      <td>86106236.0</td>\n",
       "      <td>615.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.434086</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5345 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome        sgRNA Sequence  \\\n",
       "25811                EMG1      chr12  GCGCGTGGTGAGGGGAAACC   \n",
       "155817               PHB2       12.0  ACGCGTGGTGAGGGGAAACC   \n",
       "155818               PHB2       12.0  ACGCGTGGTGAGGGGAAACC   \n",
       "155819               PHB2       12.0  ACGCGTGGTGAGGGGAAACC   \n",
       "155820               PHB2       12.0  ACGCGTGGTGAGGGGAAACC   \n",
       "...                   ...        ...                   ...   \n",
       "351648              PTCD3        2.0  GATCTGTGATAGTTATAGCT   \n",
       "351650              PTCD3        2.0  GATCTGTGATAGTTATAGCT   \n",
       "351651              PTCD3        2.0  GATCTGTGATAGTTATAGCT   \n",
       "56877              POLR1A       chr2  GATCTGTGATAGTTATAGCT   \n",
       "351649              PTCD3        2.0  GATCTGTGATAGTTATAGCT   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "25811   AAAAACGCGTGGTGAGGGGAAACCGGGCAA             6969946.0                +   \n",
       "155817  AAAAACGCGTGGTGAGGGGAAACCGGGCAA             6969946.0                -   \n",
       "155818  AAAAACGCGTGGTGAGGGGAAACCGGGCAA             6969946.0                -   \n",
       "155819  AAAAACGCGTGGTGAGGGGAAACCGGGCAA             6969946.0                -   \n",
       "155820  AAAAACGCGTGGTGAGGGGAAACCGGGCAA             6969946.0                -   \n",
       "...                                ...                   ...              ...   \n",
       "351648  TTTTGATCTGTGATAGTTATAGCTTGGAAC            86106851.0                +   \n",
       "351650  TTTTGATCTGTGATAGTTATAGCTTGGAAC            86106851.0                +   \n",
       "351651  TTTTGATCTGTGATAGTTATAGCTTGGAAC            86106851.0                +   \n",
       "56877   TTTTGATCTGTGATAGTTATAGCTTGGAAC            86106851.0                -   \n",
       "351649  TTTTGATCTGTGATAGTTATAGCTTGGAAC            86106851.0                +   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "25811      6970914.0                       -968.0   \n",
       "155817     6970664.0                        719.0   \n",
       "155818     6970664.0                        719.0   \n",
       "155819     6970664.0                        719.0   \n",
       "155820     6970664.0                        719.0   \n",
       "...              ...                          ...   \n",
       "351648    86106236.0                        615.0   \n",
       "351650    86106236.0                        615.0   \n",
       "351651    86106236.0                        615.0   \n",
       "56877     86105886.0                       -965.0   \n",
       "351649    86106236.0                        615.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "25811                        -0.032545       -0.094882   \n",
       "155817                             NaN             NaN   \n",
       "155818                             NaN             NaN   \n",
       "155819                             NaN             NaN   \n",
       "155820                             NaN             NaN   \n",
       "...                                ...             ...   \n",
       "351648                             NaN             NaN   \n",
       "351650                             NaN             NaN   \n",
       "351651                             NaN             NaN   \n",
       "56877                        -0.006604        0.115769   \n",
       "351649                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "25811                -1.623168      K562   Kox1    Nunez   \n",
       "155817                     NaN      A549   Kox1  InHouse   \n",
       "155818                     NaN      A549   Zim3  InHouse   \n",
       "155819                     NaN    HCT116   Kox1  InHouse   \n",
       "155820                     NaN    HCT116   Zim3  InHouse   \n",
       "...                        ...       ...    ...      ...   \n",
       "351648                     NaN      A549   Kox1  InHouse   \n",
       "351650                     NaN    HCT116   Kox1  InHouse   \n",
       "351651                     NaN    HCT116   Zim3  InHouse   \n",
       "56877                -0.434737      K562   Kox1    Nunez   \n",
       "351649                     NaN      A549   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "25811                      NaN                      NaN             NaN   \n",
       "155817                     NaN                      NaN             NaN   \n",
       "155818                     NaN                      NaN             NaN   \n",
       "155819                     NaN                      NaN             NaN   \n",
       "155820                     NaN                      NaN             NaN   \n",
       "...                        ...                      ...             ...   \n",
       "351648                     NaN                      NaN             NaN   \n",
       "351650                     NaN                      NaN             NaN   \n",
       "351651                     NaN                      NaN             NaN   \n",
       "56877                      NaN                      NaN             NaN   \n",
       "351649                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "25811                NaN  \n",
       "155817         -4.518868  \n",
       "155818         -6.784053  \n",
       "155819         -8.892879  \n",
       "155820         -8.323199  \n",
       "...                  ...  \n",
       "351648         -0.048226  \n",
       "351650         -1.620939  \n",
       "351651         -1.605212  \n",
       "56877                NaN  \n",
       "351649         -2.434086  \n",
       "\n",
       "[5345 rows x 18 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# the scale of the overlap problem in combined dataset\n",
    "conseq_with_varied_gene = DatasetCombine.groupby('sgRNA Context Sequence').filter(lambda DatasetCombine: DatasetCombine['Target Gene Symbol'].nunique() > 1)\n",
    "conseq_with_varied_gene.sort_values(by = 'sgRNA Context Sequence')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "ef62de3a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13674</th>\n",
       "      <td>COG3</td>\n",
       "      <td>chr13</td>\n",
       "      <td>GTAGGGAAAAGCTGGCTCTC</td>\n",
       "      <td>GACGCTAGGGAAAAGCTGGCTCTCTGGGAT</td>\n",
       "      <td>45465106.0</td>\n",
       "      <td>+</td>\n",
       "      <td>45464940.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>-0.143685</td>\n",
       "      <td>-0.927616</td>\n",
       "      <td>-6.321215</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99671</th>\n",
       "      <td>COG3</td>\n",
       "      <td>chr13</td>\n",
       "      <td>GCTAGGGAAAAGCTGGCTCTC</td>\n",
       "      <td>GACGCTAGGGAAAAGCTGGCTCTCTGGGAT</td>\n",
       "      <td>45465106.0</td>\n",
       "      <td>+</td>\n",
       "      <td>45464940.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>0.047545</td>\n",
       "      <td>0.646435</td>\n",
       "      <td>0.443106</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299591</th>\n",
       "      <td>COG3</td>\n",
       "      <td>13.0</td>\n",
       "      <td>CTAGGGAAAAGCTGGCTCTC</td>\n",
       "      <td>GACGCTAGGGAAAAGCTGGCTCTCTGGGAT</td>\n",
       "      <td>45465106.0</td>\n",
       "      <td>+</td>\n",
       "      <td>45464940.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.716820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299592</th>\n",
       "      <td>COG3</td>\n",
       "      <td>13.0</td>\n",
       "      <td>CTAGGGAAAAGCTGGCTCTC</td>\n",
       "      <td>GACGCTAGGGAAAAGCTGGCTCTCTGGGAT</td>\n",
       "      <td>45465106.0</td>\n",
       "      <td>+</td>\n",
       "      <td>45464940.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-7.612488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299593</th>\n",
       "      <td>COG3</td>\n",
       "      <td>13.0</td>\n",
       "      <td>CTAGGGAAAAGCTGGCTCTC</td>\n",
       "      <td>GACGCTAGGGAAAAGCTGGCTCTCTGGGAT</td>\n",
       "      <td>45465106.0</td>\n",
       "      <td>+</td>\n",
       "      <td>45464940.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.602514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299594</th>\n",
       "      <td>COG3</td>\n",
       "      <td>13.0</td>\n",
       "      <td>CTAGGGAAAAGCTGGCTCTC</td>\n",
       "      <td>GACGCTAGGGAAAAGCTGGCTCTCTGGGAT</td>\n",
       "      <td>45465106.0</td>\n",
       "      <td>+</td>\n",
       "      <td>45464940.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.972307</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome         sgRNA Sequence  \\\n",
       "13674                COG3      chr13   GTAGGGAAAAGCTGGCTCTC   \n",
       "99671                COG3      chr13  GCTAGGGAAAAGCTGGCTCTC   \n",
       "299591               COG3       13.0   CTAGGGAAAAGCTGGCTCTC   \n",
       "299592               COG3       13.0   CTAGGGAAAAGCTGGCTCTC   \n",
       "299593               COG3       13.0   CTAGGGAAAAGCTGGCTCTC   \n",
       "299594               COG3       13.0   CTAGGGAAAAGCTGGCTCTC   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "13674   GACGCTAGGGAAAAGCTGGCTCTCTGGGAT            45465106.0                +   \n",
       "99671   GACGCTAGGGAAAAGCTGGCTCTCTGGGAT            45465106.0                +   \n",
       "299591  GACGCTAGGGAAAAGCTGGCTCTCTGGGAT            45465106.0                +   \n",
       "299592  GACGCTAGGGAAAAGCTGGCTCTCTGGGAT            45465106.0                +   \n",
       "299593  GACGCTAGGGAAAAGCTGGCTCTCTGGGAT            45465106.0                +   \n",
       "299594  GACGCTAGGGAAAAGCTGGCTCTCTGGGAT            45465106.0                +   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "13674     45464940.0                        166.0   \n",
       "99671     45464940.0                        166.0   \n",
       "299591    45464940.0                        166.0   \n",
       "299592    45464940.0                        166.0   \n",
       "299593    45464940.0                        166.0   \n",
       "299594    45464940.0                        166.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "13674                        -0.143685       -0.927616   \n",
       "99671                              NaN             NaN   \n",
       "299591                             NaN             NaN   \n",
       "299592                             NaN             NaN   \n",
       "299593                             NaN             NaN   \n",
       "299594                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "13674                -6.321215      K562   Kox1    Nunez   \n",
       "99671                      NaN      K562   Kox1  Gilbert   \n",
       "299591                     NaN      A549   Kox1  InHouse   \n",
       "299592                     NaN      A549   Zim3  InHouse   \n",
       "299593                     NaN    HCT116   Kox1  InHouse   \n",
       "299594                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "13674                      NaN                      NaN             NaN   \n",
       "99671                 0.047545                 0.646435        0.443106   \n",
       "299591                     NaN                      NaN             NaN   \n",
       "299592                     NaN                      NaN             NaN   \n",
       "299593                     NaN                      NaN             NaN   \n",
       "299594                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "13674                NaN  \n",
       "99671                NaN  \n",
       "299591         -3.716820  \n",
       "299592         -7.612488  \n",
       "299593         -3.602514  \n",
       "299594         -3.972307  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# how does my calcualtion of CRISPick cut site compares to the actual in house dataset\n",
    "DatasetCombine[DatasetCombine['sgRNA Context Sequence'] == 'GACGCTAGGGAAAAGCTGGCTCTCTGGGAT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "847376f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7527</th>\n",
       "      <td>CCT7</td>\n",
       "      <td>chr2</td>\n",
       "      <td>GTTGCGCGCTCACCGTGGAT</td>\n",
       "      <td>AACGCTTGCGCGCTCACCGTGGATAGGCGC</td>\n",
       "      <td>73233747.0</td>\n",
       "      <td>+</td>\n",
       "      <td>73234309.0</td>\n",
       "      <td>-562.0</td>\n",
       "      <td>-0.035603</td>\n",
       "      <td>-0.127112</td>\n",
       "      <td>-1.805003</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98609</th>\n",
       "      <td>CCT7</td>\n",
       "      <td>chr2</td>\n",
       "      <td>GCTTGCGCGCTCACCGTGGAT</td>\n",
       "      <td>AACGCTTGCGCGCTCACCGTGGATAGGCGC</td>\n",
       "      <td>73233747.0</td>\n",
       "      <td>+</td>\n",
       "      <td>73234309.0</td>\n",
       "      <td>-562.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>0.015746</td>\n",
       "      <td>0.735152</td>\n",
       "      <td>0.473593</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323572</th>\n",
       "      <td>CCT7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTTGCGCGCTCACCGTGGAT</td>\n",
       "      <td>AACGCTTGCGCGCTCACCGTGGATAGGCGC</td>\n",
       "      <td>73233747.0</td>\n",
       "      <td>+</td>\n",
       "      <td>73234309.0</td>\n",
       "      <td>-562.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.647993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323573</th>\n",
       "      <td>CCT7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTTGCGCGCTCACCGTGGAT</td>\n",
       "      <td>AACGCTTGCGCGCTCACCGTGGATAGGCGC</td>\n",
       "      <td>73233747.0</td>\n",
       "      <td>+</td>\n",
       "      <td>73234309.0</td>\n",
       "      <td>-562.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-6.681559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323574</th>\n",
       "      <td>CCT7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTTGCGCGCTCACCGTGGAT</td>\n",
       "      <td>AACGCTTGCGCGCTCACCGTGGATAGGCGC</td>\n",
       "      <td>73233747.0</td>\n",
       "      <td>+</td>\n",
       "      <td>73234309.0</td>\n",
       "      <td>-562.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.565990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323575</th>\n",
       "      <td>CCT7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTTGCGCGCTCACCGTGGAT</td>\n",
       "      <td>AACGCTTGCGCGCTCACCGTGGATAGGCGC</td>\n",
       "      <td>73233747.0</td>\n",
       "      <td>+</td>\n",
       "      <td>73234309.0</td>\n",
       "      <td>-562.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.258887</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome         sgRNA Sequence  \\\n",
       "7527                 CCT7       chr2   GTTGCGCGCTCACCGTGGAT   \n",
       "98609                CCT7       chr2  GCTTGCGCGCTCACCGTGGAT   \n",
       "323572               CCT7        2.0   CTTGCGCGCTCACCGTGGAT   \n",
       "323573               CCT7        2.0   CTTGCGCGCTCACCGTGGAT   \n",
       "323574               CCT7        2.0   CTTGCGCGCTCACCGTGGAT   \n",
       "323575               CCT7        2.0   CTTGCGCGCTCACCGTGGAT   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "7527    AACGCTTGCGCGCTCACCGTGGATAGGCGC            73233747.0                +   \n",
       "98609   AACGCTTGCGCGCTCACCGTGGATAGGCGC            73233747.0                +   \n",
       "323572  AACGCTTGCGCGCTCACCGTGGATAGGCGC            73233747.0                +   \n",
       "323573  AACGCTTGCGCGCTCACCGTGGATAGGCGC            73233747.0                +   \n",
       "323574  AACGCTTGCGCGCTCACCGTGGATAGGCGC            73233747.0                +   \n",
       "323575  AACGCTTGCGCGCTCACCGTGGATAGGCGC            73233747.0                +   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "7527      73234309.0                       -562.0   \n",
       "98609     73234309.0                       -562.0   \n",
       "323572    73234309.0                       -562.0   \n",
       "323573    73234309.0                       -562.0   \n",
       "323574    73234309.0                       -562.0   \n",
       "323575    73234309.0                       -562.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "7527                         -0.035603       -0.127112   \n",
       "98609                              NaN             NaN   \n",
       "323572                             NaN             NaN   \n",
       "323573                             NaN             NaN   \n",
       "323574                             NaN             NaN   \n",
       "323575                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "7527                 -1.805003      K562   Kox1    Nunez   \n",
       "98609                      NaN      K562   Kox1  Gilbert   \n",
       "323572                     NaN      A549   Kox1  InHouse   \n",
       "323573                     NaN      A549   Zim3  InHouse   \n",
       "323574                     NaN    HCT116   Kox1  InHouse   \n",
       "323575                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "7527                       NaN                      NaN             NaN   \n",
       "98609                 0.015746                 0.735152        0.473593   \n",
       "323572                     NaN                      NaN             NaN   \n",
       "323573                     NaN                      NaN             NaN   \n",
       "323574                     NaN                      NaN             NaN   \n",
       "323575                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "7527                 NaN  \n",
       "98609                NaN  \n",
       "323572         -4.647993  \n",
       "323573         -6.681559  \n",
       "323574         -0.565990  \n",
       "323575          0.258887  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DatasetCombine[DatasetCombine['sgRNA Context Sequence'] == 'AACGCTTGCGCGCTCACCGTGGATAGGCGC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "6bbd0d9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32372</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GCCGCGTTGTGTGTCCATGA</td>\n",
       "      <td>GACACCCGCGTTGTGTGTCCATGACGGTGC</td>\n",
       "      <td>236604217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>299.0</td>\n",
       "      <td>-0.10189</td>\n",
       "      <td>-0.731931</td>\n",
       "      <td>-5.217218</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103561</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GACACCCGCGTTGTGTGTCCATGA</td>\n",
       "      <td>GACACCCGCGTTGTGTGTCCATGACGGTGC</td>\n",
       "      <td>236604217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>299.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>0.010282</td>\n",
       "      <td>-0.400256</td>\n",
       "      <td>0.083416</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255154</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CCCGCGTTGTGTGTCCATGA</td>\n",
       "      <td>GACACCCGCGTTGTGTGTCCATGACGGTGC</td>\n",
       "      <td>236604217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-5.703113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255155</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CCCGCGTTGTGTGTCCATGA</td>\n",
       "      <td>GACACCCGCGTTGTGTGTCCATGACGGTGC</td>\n",
       "      <td>236604217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-7.454270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255156</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CCCGCGTTGTGTGTCCATGA</td>\n",
       "      <td>GACACCCGCGTTGTGTGTCCATGACGGTGC</td>\n",
       "      <td>236604217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-8.943440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255157</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CCCGCGTTGTGTGTCCATGA</td>\n",
       "      <td>GACACCCGCGTTGTGTGTCCATGACGGTGC</td>\n",
       "      <td>236604217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-7.069832</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome            sgRNA Sequence  \\\n",
       "32372              HEATR1       chr1      GCCGCGTTGTGTGTCCATGA   \n",
       "103561             HEATR1       chr1  GACACCCGCGTTGTGTGTCCATGA   \n",
       "255154             HEATR1        1.0      CCCGCGTTGTGTGTCCATGA   \n",
       "255155             HEATR1        1.0      CCCGCGTTGTGTGTCCATGA   \n",
       "255156             HEATR1        1.0      CCCGCGTTGTGTGTCCATGA   \n",
       "255157             HEATR1        1.0      CCCGCGTTGTGTGTCCATGA   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "32372   GACACCCGCGTTGTGTGTCCATGACGGTGC           236604217.0                -   \n",
       "103561  GACACCCGCGTTGTGTGTCCATGACGGTGC           236604217.0                -   \n",
       "255154  GACACCCGCGTTGTGTGTCCATGACGGTGC           236604217.0                -   \n",
       "255155  GACACCCGCGTTGTGTGTCCATGACGGTGC           236604217.0                -   \n",
       "255156  GACACCCGCGTTGTGTGTCCATGACGGTGC           236604217.0                -   \n",
       "255157  GACACCCGCGTTGTGTGTCCATGACGGTGC           236604217.0                -   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "32372    236604516.0                        299.0   \n",
       "103561   236604516.0                        299.0   \n",
       "255154   236604516.0                        300.0   \n",
       "255155   236604516.0                        300.0   \n",
       "255156   236604516.0                        300.0   \n",
       "255157   236604516.0                        300.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "32372                         -0.10189       -0.731931   \n",
       "103561                             NaN             NaN   \n",
       "255154                             NaN             NaN   \n",
       "255155                             NaN             NaN   \n",
       "255156                             NaN             NaN   \n",
       "255157                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "32372                -5.217218      K562   Kox1    Nunez   \n",
       "103561                     NaN      K562   Kox1  Gilbert   \n",
       "255154                     NaN      A549   Kox1  InHouse   \n",
       "255155                     NaN      A549   Zim3  InHouse   \n",
       "255156                     NaN    HCT116   Kox1  InHouse   \n",
       "255157                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "32372                      NaN                      NaN             NaN   \n",
       "103561                0.010282                -0.400256        0.083416   \n",
       "255154                     NaN                      NaN             NaN   \n",
       "255155                     NaN                      NaN             NaN   \n",
       "255156                     NaN                      NaN             NaN   \n",
       "255157                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "32372                NaN  \n",
       "103561               NaN  \n",
       "255154         -5.703113  \n",
       "255155         -7.454270  \n",
       "255156         -8.943440  \n",
       "255157         -7.069832  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DatasetCombine[DatasetCombine['sgRNA Context Sequence'] == 'GACACCCGCGTTGTGTGTCCATGACGGTGC']\n",
    "#Nunez"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "99b4bc5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove overlap\n",
    "overlapping_gene = pd.read_csv('../Data/CRISPRiChallenges/geneOverlap/GW_ensembl_protein_coding_df_1kb.csv')\n",
    "overlapping_gene_2_3 =overlapping_gene[overlapping_gene['overlap type'].isin([2,3])]\n",
    "gene_2_3_unique = set(list(overlapping_gene_2_3['Selected Gene name']) +\\\n",
    "list(overlapping_gene_2_3['Overlapping Gene name']))\n",
    "NewDatasetCombine = deepcopy(DatasetCombine[~DatasetCombine['Target Gene Symbol'].isin(gene_2_3_unique)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "f0556b80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check the overlap issue after removing overlap genes in 2 or 3 \n",
    "conseq_with_varied_gene = NewDatasetCombine.groupby('sgRNA Context Sequence').filter(lambda NewDatasetCombine: NewDatasetCombine['Target Gene Symbol'].nunique() > 1)\n",
    "len(conseq_with_varied_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "b95e9d27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AGTCGTACCCATTTAAAGCCACGTCGGTAC    6\n",
       "GCGGGACCGAGGCCAGTACCGACGTGGCTT    6\n",
       "CTGGGAATTTGGGTATATCTTGGAAGGCAA    6\n",
       "GGAACTGGGAATTTGGGTATATCTTGGAAG    6\n",
       "GCTCTGTGCCGCTTACCTGGAACTGGGAAT    6\n",
       "                                 ..\n",
       "TGGGGGCAAGACAGTGAAAACAGGTGGGCA    2\n",
       "GAATGCCTGTGAAAGATTGTATAAAGGAAC    2\n",
       "GAAAGATTGTATAAAGGAACAAACAGGCCC    2\n",
       "GGTTTGCTTTCAGCTTTTGAGTACTGGGCT    2\n",
       "GCCTTCGCCGCTCGGGCCGCCCGGGGGAAA    2\n",
       "Name: sgRNA Context Sequence, Length: 86955, dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NewDatasetCombine[NewDatasetCombine['sgRNA Context Sequence'].duplicated(False)]['sgRNA Context Sequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "36558b39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32471</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GTGGGAATTTGGGTATATCT</td>\n",
       "      <td>GGAACTGGGAATTTGGGTATATCTTGGAAG</td>\n",
       "      <td>236604445.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>-0.387244</td>\n",
       "      <td>-1.270996</td>\n",
       "      <td>-8.25847</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103722</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GGAACTGGGAATTTGGGTATATCT</td>\n",
       "      <td>GGAACTGGGAATTTGGGTATATCTTGGAAG</td>\n",
       "      <td>236604445.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>0.131272</td>\n",
       "      <td>3.193287</td>\n",
       "      <td>1.318318</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316247</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGGGAATTTGGGTATATCT</td>\n",
       "      <td>GGAACTGGGAATTTGGGTATATCTTGGAAG</td>\n",
       "      <td>236604445.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-8.086635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316248</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGGGAATTTGGGTATATCT</td>\n",
       "      <td>GGAACTGGGAATTTGGGTATATCTTGGAAG</td>\n",
       "      <td>236604445.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-8.911877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316249</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGGGAATTTGGGTATATCT</td>\n",
       "      <td>GGAACTGGGAATTTGGGTATATCTTGGAAG</td>\n",
       "      <td>236604445.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-9.114155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316250</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGGGAATTTGGGTATATCT</td>\n",
       "      <td>GGAACTGGGAATTTGGGTATATCTTGGAAG</td>\n",
       "      <td>236604445.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-9.410548</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome            sgRNA Sequence  \\\n",
       "32471              HEATR1       chr1      GTGGGAATTTGGGTATATCT   \n",
       "103722             HEATR1       chr1  GGAACTGGGAATTTGGGTATATCT   \n",
       "316247             HEATR1        1.0      CTGGGAATTTGGGTATATCT   \n",
       "316248             HEATR1        1.0      CTGGGAATTTGGGTATATCT   \n",
       "316249             HEATR1        1.0      CTGGGAATTTGGGTATATCT   \n",
       "316250             HEATR1        1.0      CTGGGAATTTGGGTATATCT   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "32471   GGAACTGGGAATTTGGGTATATCTTGGAAG           236604445.0                -   \n",
       "103722  GGAACTGGGAATTTGGGTATATCTTGGAAG           236604445.0                -   \n",
       "316247  GGAACTGGGAATTTGGGTATATCTTGGAAG           236604445.0                -   \n",
       "316248  GGAACTGGGAATTTGGGTATATCTTGGAAG           236604445.0                -   \n",
       "316249  GGAACTGGGAATTTGGGTATATCTTGGAAG           236604445.0                -   \n",
       "316250  GGAACTGGGAATTTGGGTATATCTTGGAAG           236604445.0                -   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "32471    236604516.0                         71.0   \n",
       "103722   236604516.0                         71.0   \n",
       "316247   236604516.0                         72.0   \n",
       "316248   236604516.0                         72.0   \n",
       "316249   236604516.0                         72.0   \n",
       "316250   236604516.0                         72.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "32471                        -0.387244       -1.270996   \n",
       "103722                             NaN             NaN   \n",
       "316247                             NaN             NaN   \n",
       "316248                             NaN             NaN   \n",
       "316249                             NaN             NaN   \n",
       "316250                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "32471                 -8.25847      K562   Kox1    Nunez   \n",
       "103722                     NaN      K562   Kox1  Gilbert   \n",
       "316247                     NaN      A549   Kox1  InHouse   \n",
       "316248                     NaN      A549   Zim3  InHouse   \n",
       "316249                     NaN    HCT116   Kox1  InHouse   \n",
       "316250                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "32471                      NaN                      NaN             NaN   \n",
       "103722                0.131272                 3.193287        1.318318   \n",
       "316247                     NaN                      NaN             NaN   \n",
       "316248                     NaN                      NaN             NaN   \n",
       "316249                     NaN                      NaN             NaN   \n",
       "316250                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "32471                NaN  \n",
       "103722               NaN  \n",
       "316247         -8.086635  \n",
       "316248         -8.911877  \n",
       "316249         -9.114155  \n",
       "316250         -9.410548  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NewDatasetCombine[NewDatasetCombine['sgRNA Context Sequence'] == 'GGAACTGGGAATTTGGGTATATCTTGGAAG']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "7096487c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32374</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>0.001632</td>\n",
       "      <td>0.222083</td>\n",
       "      <td>0.165053</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103565</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GCTGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>-0.057501</td>\n",
       "      <td>-0.69649</td>\n",
       "      <td>-0.018383</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500476</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.694686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500477</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.814877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500478</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.792970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500479</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.389959</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome          sgRNA Sequence  \\\n",
       "32374              HEATR1       chr1    GGCCATCGTCTTTGCATCTC   \n",
       "103565             HEATR1       chr1  GCTGCCATCGTCTTTGCATCTC   \n",
       "500476             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "500477             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "500478             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "500479             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "32374   CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "103565  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500476  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500477  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500478  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500479  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "32374    236604516.0                        256.0   \n",
       "103565   236604516.0                        256.0   \n",
       "500476   236604516.0                        257.0   \n",
       "500477   236604516.0                        257.0   \n",
       "500478   236604516.0                        257.0   \n",
       "500479   236604516.0                        257.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "32374                         0.001632        0.222083   \n",
       "103565                             NaN             NaN   \n",
       "500476                             NaN             NaN   \n",
       "500477                             NaN             NaN   \n",
       "500478                             NaN             NaN   \n",
       "500479                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "32374                 0.165053      K562   Kox1    Nunez   \n",
       "103565                     NaN      K562   Kox1  Gilbert   \n",
       "500476                     NaN      A549   Kox1  InHouse   \n",
       "500477                     NaN      A549   Zim3  InHouse   \n",
       "500478                     NaN    HCT116   Kox1  InHouse   \n",
       "500479                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "32374                      NaN                      NaN             NaN   \n",
       "103565               -0.057501                 -0.69649       -0.018383   \n",
       "500476                     NaN                      NaN             NaN   \n",
       "500477                     NaN                      NaN             NaN   \n",
       "500478                     NaN                      NaN             NaN   \n",
       "500479                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "32374                NaN  \n",
       "103565               NaN  \n",
       "500476         -4.694686  \n",
       "500477         -4.814877  \n",
       "500478         -4.792970  \n",
       "500479         -4.389959  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NewDatasetCombine[NewDatasetCombine['sgRNA Context Sequence'] == 'CTGCTGCCATCGTCTTTGCATCTCCGGGGT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f563de56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32469</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GTGTGCCGCTTACCTGGAAC</td>\n",
       "      <td>GGCTCTGTGCCGCTTACCTGGAACTGGGAA</td>\n",
       "      <td>236604426.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>0.001559</td>\n",
       "      <td>0.216603</td>\n",
       "      <td>0.134137</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103719</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GCTCTGTGCCGCTTACCTGGAAC</td>\n",
       "      <td>GGCTCTGTGCCGCTTACCTGGAACTGGGAA</td>\n",
       "      <td>236604426.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>0.037329</td>\n",
       "      <td>1.38622</td>\n",
       "      <td>0.697329</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319086</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGTGCCGCTTACCTGGAAC</td>\n",
       "      <td>GGCTCTGTGCCGCTTACCTGGAACTGGGAA</td>\n",
       "      <td>236604426.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.873853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319087</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGTGCCGCTTACCTGGAAC</td>\n",
       "      <td>GGCTCTGTGCCGCTTACCTGGAACTGGGAA</td>\n",
       "      <td>236604426.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.657091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319088</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGTGCCGCTTACCTGGAAC</td>\n",
       "      <td>GGCTCTGTGCCGCTTACCTGGAACTGGGAA</td>\n",
       "      <td>236604426.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.566195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319089</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTGTGCCGCTTACCTGGAAC</td>\n",
       "      <td>GGCTCTGTGCCGCTTACCTGGAACTGGGAA</td>\n",
       "      <td>236604426.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.330788</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome           sgRNA Sequence  \\\n",
       "32469              HEATR1       chr1     GTGTGCCGCTTACCTGGAAC   \n",
       "103719             HEATR1       chr1  GCTCTGTGCCGCTTACCTGGAAC   \n",
       "319086             HEATR1        1.0     CTGTGCCGCTTACCTGGAAC   \n",
       "319087             HEATR1        1.0     CTGTGCCGCTTACCTGGAAC   \n",
       "319088             HEATR1        1.0     CTGTGCCGCTTACCTGGAAC   \n",
       "319089             HEATR1        1.0     CTGTGCCGCTTACCTGGAAC   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "32469   GGCTCTGTGCCGCTTACCTGGAACTGGGAA           236604426.0                -   \n",
       "103719  GGCTCTGTGCCGCTTACCTGGAACTGGGAA           236604426.0                -   \n",
       "319086  GGCTCTGTGCCGCTTACCTGGAACTGGGAA           236604426.0                -   \n",
       "319087  GGCTCTGTGCCGCTTACCTGGAACTGGGAA           236604426.0                -   \n",
       "319088  GGCTCTGTGCCGCTTACCTGGAACTGGGAA           236604426.0                -   \n",
       "319089  GGCTCTGTGCCGCTTACCTGGAACTGGGAA           236604426.0                -   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "32469    236604516.0                         90.0   \n",
       "103719   236604516.0                         90.0   \n",
       "319086   236604516.0                         91.0   \n",
       "319087   236604516.0                         91.0   \n",
       "319088   236604516.0                         91.0   \n",
       "319089   236604516.0                         91.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "32469                         0.001559        0.216603   \n",
       "103719                             NaN             NaN   \n",
       "319086                             NaN             NaN   \n",
       "319087                             NaN             NaN   \n",
       "319088                             NaN             NaN   \n",
       "319089                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "32469                 0.134137      K562   Kox1    Nunez   \n",
       "103719                     NaN      K562   Kox1  Gilbert   \n",
       "319086                     NaN      A549   Kox1  InHouse   \n",
       "319087                     NaN      A549   Zim3  InHouse   \n",
       "319088                     NaN    HCT116   Kox1  InHouse   \n",
       "319089                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "32469                      NaN                      NaN             NaN   \n",
       "103719                0.037329                  1.38622        0.697329   \n",
       "319086                     NaN                      NaN             NaN   \n",
       "319087                     NaN                      NaN             NaN   \n",
       "319088                     NaN                      NaN             NaN   \n",
       "319089                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "32469                NaN  \n",
       "103719               NaN  \n",
       "319086         -1.873853  \n",
       "319087         -1.657091  \n",
       "319088         -4.566195  \n",
       "319089         -1.330788  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NewDatasetCombine[NewDatasetCombine['sgRNA Context Sequence'] == 'GGCTCTGTGCCGCTTACCTGGAACTGGGAA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "8fa7df9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# temp fix for why crispick is one off for neg target gene\n",
    "NewDatasetCombine.loc[(NewDatasetCombine['DataSet'].isin(['Nunez', 'Gilbert'])) & (NewDatasetCombine['Strand of Target'] == '-'), \n",
    "                      'sgRNA \\'Cut\\' Site TSS Offset'] += 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "c24a79fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Gene Symbol</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>sgRNA Sequence</th>\n",
       "      <th>sgRNA Context Sequence</th>\n",
       "      <th>sgRNA 'Cut' Position</th>\n",
       "      <th>Strand of Target</th>\n",
       "      <th>TSS Position</th>\n",
       "      <th>sgRNA 'Cut' Site TSS Offset</th>\n",
       "      <th>Phenotype scores-ave_Rep1_Rep2</th>\n",
       "      <th>Avg_LFC-Tfinal</th>\n",
       "      <th>z-score relative to NC</th>\n",
       "      <th>condition</th>\n",
       "      <th>Domain</th>\n",
       "      <th>DataSet</th>\n",
       "      <th>rhoSignChangedAndAvged</th>\n",
       "      <th>zscoreRelativeNC_signed</th>\n",
       "      <th>Avg_LFC_signed</th>\n",
       "      <th>z_scored_avg_lfc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32374</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>0.001632</td>\n",
       "      <td>0.222083</td>\n",
       "      <td>0.165053</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Nunez</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103565</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>GCTGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K562</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>Gilbert</td>\n",
       "      <td>-0.057501</td>\n",
       "      <td>-0.69649</td>\n",
       "      <td>-0.018383</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500476</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.694686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500477</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A549</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.814877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500478</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Kox1</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.792970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500479</th>\n",
       "      <td>HEATR1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TGCCATCGTCTTTGCATCTC</td>\n",
       "      <td>CTGCTGCCATCGTCTTTGCATCTCCGGGGT</td>\n",
       "      <td>236604260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>236604516.0</td>\n",
       "      <td>257.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>Zim3</td>\n",
       "      <td>InHouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.389959</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Target Gene Symbol chromosome          sgRNA Sequence  \\\n",
       "32374              HEATR1       chr1    GGCCATCGTCTTTGCATCTC   \n",
       "103565             HEATR1       chr1  GCTGCCATCGTCTTTGCATCTC   \n",
       "500476             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "500477             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "500478             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "500479             HEATR1        1.0    TGCCATCGTCTTTGCATCTC   \n",
       "\n",
       "                sgRNA Context Sequence  sgRNA 'Cut' Position Strand of Target  \\\n",
       "32374   CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "103565  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500476  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500477  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500478  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "500479  CTGCTGCCATCGTCTTTGCATCTCCGGGGT           236604260.0                -   \n",
       "\n",
       "        TSS Position  sgRNA 'Cut' Site TSS Offset  \\\n",
       "32374    236604516.0                        257.0   \n",
       "103565   236604516.0                        257.0   \n",
       "500476   236604516.0                        257.0   \n",
       "500477   236604516.0                        257.0   \n",
       "500478   236604516.0                        257.0   \n",
       "500479   236604516.0                        257.0   \n",
       "\n",
       "        Phenotype scores-ave_Rep1_Rep2  Avg_LFC-Tfinal  \\\n",
       "32374                         0.001632        0.222083   \n",
       "103565                             NaN             NaN   \n",
       "500476                             NaN             NaN   \n",
       "500477                             NaN             NaN   \n",
       "500478                             NaN             NaN   \n",
       "500479                             NaN             NaN   \n",
       "\n",
       "        z-score relative to NC condition Domain  DataSet  \\\n",
       "32374                 0.165053      K562   Kox1    Nunez   \n",
       "103565                     NaN      K562   Kox1  Gilbert   \n",
       "500476                     NaN      A549   Kox1  InHouse   \n",
       "500477                     NaN      A549   Zim3  InHouse   \n",
       "500478                     NaN    HCT116   Kox1  InHouse   \n",
       "500479                     NaN    HCT116   Zim3  InHouse   \n",
       "\n",
       "        rhoSignChangedAndAvged  zscoreRelativeNC_signed  Avg_LFC_signed  \\\n",
       "32374                      NaN                      NaN             NaN   \n",
       "103565               -0.057501                 -0.69649       -0.018383   \n",
       "500476                     NaN                      NaN             NaN   \n",
       "500477                     NaN                      NaN             NaN   \n",
       "500478                     NaN                      NaN             NaN   \n",
       "500479                     NaN                      NaN             NaN   \n",
       "\n",
       "        z_scored_avg_lfc  \n",
       "32374                NaN  \n",
       "103565               NaN  \n",
       "500476         -4.694686  \n",
       "500477         -4.814877  \n",
       "500478         -4.792970  \n",
       "500479         -4.389959  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NewDatasetCombine[NewDatasetCombine['sgRNA Context Sequence'] == 'CTGCTGCCATCGTCTTTGCATCTCCGGGGT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e97fd43f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "166624"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NewDatasetCombine['sgRNA Context Sequence'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "e2723bb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "431095"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(NewDatasetCombine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "889b9a42",
   "metadata": {},
   "outputs": [],
   "source": [
    "NewDatasetCombine['chromosome'] = NewDatasetCombine['chromosome'].apply(standardize_chromosome)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f402197d",
   "metadata": {},
   "outputs": [],
   "source": [
    "outpath = '../Data/CleanedExternalData/'\n",
    "\n",
    "NewDatasetCombine.to_csv(outpath + 'DatasetCombineCleaned_v3.csv', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
